{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "mOAiPuMecwfT",
        "outputId": "89f1819e-3e70-420c-8dab-eab0ef00ea60"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "bin/micromamba\n",
            "\u001b[?25l\u001b[2K\u001b[0G[+] 0.0s\n",
            "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.1s\n",
            "bioconda/linux-64 ..  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[0G\u001b[?25h\u001b[?25l\u001b[2K\u001b[0G[+] 0.0s\n",
            "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.1s\n",
            "bioconda/linux-64     ⣾  \n",
            "bioconda/noarch       ⣾  \n",
            "conda-forge/linux-64   1%\n",
            "conda-forge/noarch    ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.2s\n",
            "bioconda/linux-64     33%\n",
            "bioconda/noarch       24%\n",
            "conda-forge/linux-64   8%\n",
            "conda-forge/noarch     7%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.3s\n",
            "bioconda/linux-64     75%\n",
            "bioconda/noarch       70%\n",
            "conda-forge/linux-64  14%\n",
            "conda-forge/noarch    19%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.4s\n",
            "bioconda/linux-64     93%\n",
            "bioconda/noarch      100%\n",
            "conda-forge/linux-64  16%\n",
            "conda-forge/noarch    27%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gbioconda/linux-64                                 \n",
            "bioconda/noarch                                   \n",
            "[+] 0.5s\n",
            "conda-forge/linux-64  23%\n",
            "conda-forge/noarch    40%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.6s\n",
            "conda-forge/linux-64  32%\n",
            "conda-forge/noarch    58%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.7s\n",
            "conda-forge/linux-64  40%\n",
            "conda-forge/noarch    66%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.8s\n",
            "conda-forge/linux-64  44%\n",
            "conda-forge/noarch    85%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.9s\n",
            "conda-forge/linux-64  49%\n",
            "conda-forge/noarch    97%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.0s\n",
            "conda-forge/linux-64  50%\n",
            "conda-forge/noarch    97%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gconda-forge/noarch                                \n",
            "[+] 1.1s\n",
            "conda-forge/linux-64  65%\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.2s\n",
            "conda-forge/linux-64  71%\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.3s\n",
            "conda-forge/linux-64  79%\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.4s\n",
            "conda-forge/linux-64  88%\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.5s\n",
            "conda-forge/linux-64  96%\u001b[2K\u001b[1A\u001b[2K\u001b[0Gconda-forge/linux-64                              \n",
            "\u001b[?25h\n",
            "\n",
            "Transaction\n",
            "\n",
            "  Prefix: /root/.local/share/mamba/envs/bio\n",
            "\n",
            "  Updating specs:\n",
            "\n",
            "   - fastqc\n",
            "   - trimmomatic\n",
            "   - bwa\n",
            "   - samtools\n",
            "   - bcftools\n",
            "   - fastp\n",
            "   - hisat2\n",
            "   - bowtie2\n",
            "   - bedtools\n",
            "   - seqtk\n",
            "   - multiqc\n",
            "\n",
            "\n",
            "  Package                           Version  Build                 Channel          Size\n",
            "──────────────────────────────────────────────────────────────────────────────────────────\n",
            "  Install:\n",
            "──────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  \u001b[32m+ _libgcc_mutex            \u001b[0m           0.1  conda_forge           conda-forge       3kB\n",
            "  \u001b[32m+ _openmp_mutex            \u001b[0m           4.5  2_gnu                 conda-forge      24kB\n",
            "  \u001b[32m+ alsa-lib                 \u001b[0m        1.2.14  hb9d3cd8_0            conda-forge     567kB\n",
            "  \u001b[32m+ annotated-types          \u001b[0m         0.7.0  pyhd8ed1ab_1          conda-forge      18kB\n",
            "  \u001b[32m+ attrs                    \u001b[0m        25.3.0  pyh71513ae_0          conda-forge      57kB\n",
            "  \u001b[32m+ bcftools                 \u001b[0m          1.21  h3a4d415_1            bioconda          1MB\n",
            "  \u001b[32m+ bedtools                 \u001b[0m        2.31.1  h13024bc_3            bioconda          2MB\n",
            "  \u001b[32m+ bowtie2                  \u001b[0m         2.5.4  he96a11b_5            bioconda         15MB\n",
            "  \u001b[32m+ brotli-python            \u001b[0m         1.1.0  py313h46c70d0_2       conda-forge     350kB\n",
            "  \u001b[32m+ bwa                      \u001b[0m        0.7.19  h577a1d6_0            bioconda        198kB\n",
            "  \u001b[32m+ bzip2                    \u001b[0m         1.0.8  h4bc722e_7            conda-forge     253kB\n",
            "  \u001b[32m+ c-ares                   \u001b[0m        1.34.5  hb9d3cd8_0            conda-forge     207kB\n",
            "  \u001b[32m+ ca-certificates          \u001b[0m     2025.4.26  hbd8a1cb_0            conda-forge     152kB\n",
            "  \u001b[32m+ cairo                    \u001b[0m        1.18.4  h3394656_0            conda-forge     978kB\n",
            "  \u001b[32m+ certifi                  \u001b[0m     2025.4.26  pyhd8ed1ab_0          conda-forge     157kB\n",
            "  \u001b[32m+ cffi                     \u001b[0m        1.17.1  py313hfab6e84_0       conda-forge     296kB\n",
            "  \u001b[32m+ charset-normalizer       \u001b[0m         3.4.2  pyhd8ed1ab_0          conda-forge      50kB\n",
            "  \u001b[32m+ click                    \u001b[0m         8.1.8  pyh707e725_0          conda-forge      85kB\n",
            "  \u001b[32m+ colorama                 \u001b[0m         0.4.6  pyhd8ed1ab_1          conda-forge      27kB\n",
            "  \u001b[32m+ coloredlogs              \u001b[0m        15.0.1  pyhd8ed1ab_4          conda-forge      44kB\n",
            "  \u001b[32m+ colormath                \u001b[0m         3.0.0  pyhd8ed1ab_4          conda-forge      39kB\n",
            "  \u001b[32m+ expat                    \u001b[0m         2.7.0  h5888daf_0            conda-forge     140kB\n",
            "  \u001b[32m+ fastp                    \u001b[0m        0.24.1  heae3180_0            bioconda        264kB\n",
            "  \u001b[32m+ fastqc                   \u001b[0m        0.12.1  hdfd78af_0            bioconda         12MB\n",
            "  \u001b[32m+ font-ttf-dejavu-sans-mono\u001b[0m          2.37  hab24e00_0            conda-forge     397kB\n",
            "  \u001b[32m+ font-ttf-inconsolata     \u001b[0m         3.000  h77eed37_0            conda-forge      97kB\n",
            "  \u001b[32m+ font-ttf-source-code-pro \u001b[0m         2.038  h77eed37_0            conda-forge     701kB\n",
            "  \u001b[32m+ font-ttf-ubuntu          \u001b[0m          0.83  h77eed37_3            conda-forge       2MB\n",
            "  \u001b[32m+ fontconfig               \u001b[0m        2.15.0  h7e30c49_1            conda-forge     266kB\n",
            "  \u001b[32m+ fonts-conda-ecosystem    \u001b[0m             1  0                     conda-forge       4kB\n",
            "  \u001b[32m+ fonts-conda-forge        \u001b[0m             1  0                     conda-forge       4kB\n",
            "  \u001b[32m+ freetype                 \u001b[0m        2.13.3  ha770c72_1            conda-forge     172kB\n",
            "  \u001b[32m+ giflib                   \u001b[0m         5.2.2  hd590300_0            conda-forge      77kB\n",
            "  \u001b[32m+ graphite2                \u001b[0m        1.3.13  h59595ed_1003         conda-forge      97kB\n",
            "  \u001b[32m+ gsl                      \u001b[0m           2.7  he838d99_0            conda-forge       3MB\n",
            "  \u001b[32m+ h2                       \u001b[0m         4.2.0  pyhd8ed1ab_0          conda-forge      54kB\n",
            "  \u001b[32m+ harfbuzz                 \u001b[0m        11.1.0  h3beb420_0            conda-forge       2MB\n",
            "  \u001b[32m+ hisat2                   \u001b[0m         2.2.1  h503566f_8            bioconda         17MB\n",
            "  \u001b[32m+ hpack                    \u001b[0m         4.1.0  pyhd8ed1ab_0          conda-forge      31kB\n",
            "  \u001b[32m+ htslib                   \u001b[0m          1.21  h566b1c6_1            bioconda          3MB\n",
            "  \u001b[32m+ humanfriendly            \u001b[0m          10.0  pyh707e725_8          conda-forge      74kB\n",
            "  \u001b[32m+ humanize                 \u001b[0m        4.12.3  pyhd8ed1ab_0          conda-forge      66kB\n",
            "  \u001b[32m+ hyperframe               \u001b[0m         6.1.0  pyhd8ed1ab_0          conda-forge      17kB\n",
            "  \u001b[32m+ icu                      \u001b[0m          75.1  he02047a_0            conda-forge      12MB\n",
            "  \u001b[32m+ idna                     \u001b[0m          3.10  pyhd8ed1ab_1          conda-forge      50kB\n",
            "  \u001b[32m+ importlib-metadata       \u001b[0m         8.6.1  pyha770c72_0          conda-forge      29kB\n",
            "  \u001b[32m+ importlib_resources      \u001b[0m         6.5.2  pyhd8ed1ab_0          conda-forge      34kB\n",
            "  \u001b[32m+ isa-l                    \u001b[0m        2.31.1  hb9d3cd8_1            conda-forge     157kB\n",
            "  \u001b[32m+ jinja2                   \u001b[0m         3.1.6  pyhd8ed1ab_0          conda-forge     113kB\n",
            "  \u001b[32m+ jsonschema               \u001b[0m        4.23.0  pyhd8ed1ab_1          conda-forge      74kB\n",
            "  \u001b[32m+ jsonschema-specifications\u001b[0m      2025.4.1  pyh29332c3_0          conda-forge      19kB\n",
            "  \u001b[32m+ kaleido-core             \u001b[0m         0.2.1  h3644ca4_0            conda-forge      62MB\n",
            "  \u001b[32m+ keyutils                 \u001b[0m         1.6.1  h166bdaf_0            conda-forge     118kB\n",
            "  \u001b[32m+ krb5                     \u001b[0m        1.21.3  h659f571_0            conda-forge       1MB\n",
            "  \u001b[32m+ lcms2                    \u001b[0m          2.17  h717163a_0            conda-forge     248kB\n",
            "  \u001b[32m+ ld_impl_linux-64         \u001b[0m          2.43  h712a8e2_4            conda-forge     671kB\n",
            "  \u001b[32m+ lerc                     \u001b[0m         4.0.0  h0aef613_1            conda-forge     264kB\n",
            "  \u001b[32m+ libblas                  \u001b[0m         3.9.0  31_h59b9bed_openblas  conda-forge      17kB\n",
            "  \u001b[32m+ libcblas                 \u001b[0m         3.9.0  31_he106b2a_openblas  conda-forge      17kB\n",
            "  \u001b[32m+ libcups                  \u001b[0m         2.3.3  h4637d8d_4            conda-forge       5MB\n",
            "  \u001b[32m+ libcurl                  \u001b[0m        8.13.0  h332b0f4_0            conda-forge     438kB\n",
            "  \u001b[32m+ libdeflate               \u001b[0m          1.22  hb9d3cd8_0            conda-forge      72kB\n",
            "  \u001b[32m+ libedit                  \u001b[0m  3.1.20250104  pl5321h7949ede_0      conda-forge     135kB\n",
            "  \u001b[32m+ libev                    \u001b[0m          4.33  hd590300_2            conda-forge     113kB\n",
            "  \u001b[32m+ libexpat                 \u001b[0m         2.7.0  h5888daf_0            conda-forge      74kB\n",
            "  \u001b[32m+ libffi                   \u001b[0m         3.4.6  h2dba641_1            conda-forge      57kB\n",
            "  \u001b[32m+ libfreetype              \u001b[0m        2.13.3  ha770c72_1            conda-forge       8kB\n",
            "  \u001b[32m+ libfreetype6             \u001b[0m        2.13.3  h48d6fc4_1            conda-forge     380kB\n",
            "  \u001b[32m+ libgcc                   \u001b[0m        14.2.0  h767d61c_2            conda-forge     848kB\n",
            "  \u001b[32m+ libgcc-ng                \u001b[0m        14.2.0  h69a702a_2            conda-forge      54kB\n",
            "  \u001b[32m+ libgfortran              \u001b[0m        14.2.0  h69a702a_2            conda-forge      54kB\n",
            "  \u001b[32m+ libgfortran5             \u001b[0m        14.2.0  hf1ad2bd_2            conda-forge       1MB\n",
            "  \u001b[32m+ libglib                  \u001b[0m        2.84.1  h3618099_1            conda-forge       4MB\n",
            "  \u001b[32m+ libgomp                  \u001b[0m        14.2.0  h767d61c_2            conda-forge     460kB\n",
            "  \u001b[32m+ libiconv                 \u001b[0m          1.18  h4ce23a2_1            conda-forge     713kB\n",
            "  \u001b[32m+ libjpeg-turbo            \u001b[0m         3.1.0  hb9d3cd8_0            conda-forge     629kB\n",
            "  \u001b[32m+ liblapack                \u001b[0m         3.9.0  31_h7ac8fdf_openblas  conda-forge      17kB\n",
            "  \u001b[32m+ liblzma                  \u001b[0m         5.8.1  hb9d3cd8_1            conda-forge     113kB\n",
            "  \u001b[32m+ libmpdec                 \u001b[0m         4.0.0  h4bc722e_0            conda-forge      90kB\n",
            "  \u001b[32m+ libnghttp2               \u001b[0m        1.64.0  h161d5f1_0            conda-forge     648kB\n",
            "  \u001b[32m+ libopenblas              \u001b[0m        0.3.29  pthreads_h94d23a6_0   conda-forge       6MB\n",
            "  \u001b[32m+ libpng                   \u001b[0m        1.6.47  h943b412_0            conda-forge     289kB\n",
            "  \u001b[32m+ libsqlite                \u001b[0m        3.49.1  hee588c1_2            conda-forge     919kB\n",
            "  \u001b[32m+ libssh2                  \u001b[0m        1.11.1  hcf80075_0            conda-forge     305kB\n",
            "  \u001b[32m+ libstdcxx                \u001b[0m        14.2.0  h8f9b012_2            conda-forge       4MB\n",
            "  \u001b[32m+ libstdcxx-ng             \u001b[0m        14.2.0  h4852527_2            conda-forge      54kB\n",
            "  \u001b[32m+ libtiff                  \u001b[0m         4.7.0  hc4654cb_2            conda-forge     429kB\n",
            "  \u001b[32m+ libuuid                  \u001b[0m        2.38.1  h0b41bf4_0            conda-forge      34kB\n",
            "  \u001b[32m+ libwebp-base             \u001b[0m         1.5.0  h851e524_0            conda-forge     430kB\n",
            "  \u001b[32m+ libxcb                   \u001b[0m        1.17.0  h8a09558_0            conda-forge     396kB\n",
            "  \u001b[32m+ libxcrypt                \u001b[0m        4.4.36  hd590300_1            conda-forge     100kB\n",
            "  \u001b[32m+ libzlib                  \u001b[0m         1.3.1  hb9d3cd8_2            conda-forge      61kB\n",
            "  \u001b[32m+ markdown                 \u001b[0m           3.8  pyhd8ed1ab_0          conda-forge      80kB\n",
            "  \u001b[32m+ markdown-it-py           \u001b[0m         3.0.0  pyhd8ed1ab_1          conda-forge      64kB\n",
            "  \u001b[32m+ markupsafe               \u001b[0m         3.0.2  py313h8060acc_1       conda-forge      25kB\n",
            "  \u001b[32m+ mathjax                  \u001b[0m         2.7.7  ha770c72_3            conda-forge      22MB\n",
            "  \u001b[32m+ mdurl                    \u001b[0m         0.1.2  pyhd8ed1ab_1          conda-forge      14kB\n",
            "  \u001b[32m+ multiqc                  \u001b[0m          1.28  pyhdfd78af_0          bioconda          4MB\n",
            "  \u001b[32m+ narwhals                 \u001b[0m        1.38.0  pyhe01879c_0          conda-forge     215kB\n",
            "  \u001b[32m+ natsort                  \u001b[0m         8.4.0  pyh29332c3_1          conda-forge      39kB\n",
            "  \u001b[32m+ ncurses                  \u001b[0m           6.5  h2d0b736_3            conda-forge     892kB\n",
            "  \u001b[32m+ networkx                 \u001b[0m         3.4.2  pyh267e887_2          conda-forge       1MB\n",
            "  \u001b[32m+ nspr                     \u001b[0m          4.36  h5888daf_0            conda-forge     230kB\n",
            "  \u001b[32m+ nss                      \u001b[0m         3.111  h159eef7_0            conda-forge       2MB\n",
            "  \u001b[32m+ numpy                    \u001b[0m         2.2.5  py313h17eae1a_0       conda-forge       9MB\n",
            "  \u001b[32m+ openjdk                  \u001b[0m        23.0.2  h53dfc1b_2            conda-forge     190MB\n",
            "  \u001b[32m+ openjpeg                 \u001b[0m         2.5.3  h5fbd93e_0            conda-forge     343kB\n",
            "  \u001b[32m+ openssl                  \u001b[0m         3.5.0  h7b32b05_1            conda-forge       3MB\n",
            "  \u001b[32m+ packaging                \u001b[0m          25.0  pyh29332c3_1          conda-forge      62kB\n",
            "  \u001b[32m+ pcre2                    \u001b[0m         10.45  hc749103_0            conda-forge       1MB\n",
            "  \u001b[32m+ perl                     \u001b[0m        5.32.1  7_hd590300_perl5      conda-forge      13MB\n",
            "  \u001b[32m+ pillow                   \u001b[0m        11.1.0  py313h8db990d_0       conda-forge      42MB\n",
            "  \u001b[32m+ pip                      \u001b[0m        25.1.1  pyh145f28c_0          conda-forge       1MB\n",
            "  \u001b[32m+ pixman                   \u001b[0m        0.46.0  h29eaf8c_0            conda-forge     399kB\n",
            "  \u001b[32m+ pkgutil-resolve-name     \u001b[0m        1.3.10  pyhd8ed1ab_2          conda-forge      11kB\n",
            "  \u001b[32m+ plotly                   \u001b[0m         6.0.1  pyhd8ed1ab_0          conda-forge       5MB\n",
            "  \u001b[32m+ pthread-stubs            \u001b[0m           0.4  hb9d3cd8_1002         conda-forge       8kB\n",
            "  \u001b[32m+ pyaml-env                \u001b[0m         1.2.2  pyhd8ed1ab_0          conda-forge      15kB\n",
            "  \u001b[32m+ pycparser                \u001b[0m          2.22  pyh29332c3_1          conda-forge     110kB\n",
            "  \u001b[32m+ pydantic                 \u001b[0m        2.11.3  pyh3cfb1c2_0          conda-forge     307kB\n",
            "  \u001b[32m+ pydantic-core            \u001b[0m        2.33.1  py313h6071e0b_0       conda-forge       2MB\n",
            "  \u001b[32m+ pygments                 \u001b[0m        2.19.1  pyhd8ed1ab_0          conda-forge     889kB\n",
            "  \u001b[32m+ pysocks                  \u001b[0m         1.7.1  pyha55dd90_7          conda-forge      21kB\n",
            "  \u001b[32m+ python                   \u001b[0m        3.13.3  hf636f53_101_cp313    conda-forge      33MB\n",
            "  \u001b[32m+ python-dotenv            \u001b[0m         1.1.0  pyh29332c3_1          conda-forge      26kB\n",
            "  \u001b[32m+ python-kaleido           \u001b[0m         0.2.1  pyhd8ed1ab_0          conda-forge      18kB\n",
            "  \u001b[32m+ python_abi               \u001b[0m          3.13  7_cp313               conda-forge       7kB\n",
            "  \u001b[32m+ pyyaml                   \u001b[0m         6.0.2  py313h8060acc_2       conda-forge     206kB\n",
            "  \u001b[32m+ readline                 \u001b[0m           8.2  h8c095d6_2            conda-forge     282kB\n",
            "  \u001b[32m+ referencing              \u001b[0m        0.36.2  pyh29332c3_0          conda-forge      52kB\n",
            "  \u001b[32m+ regex                    \u001b[0m     2024.11.6  py313h536fd9c_0       conda-forge     404kB\n",
            "  \u001b[32m+ requests                 \u001b[0m        2.32.3  pyhd8ed1ab_1          conda-forge      59kB\n",
            "  \u001b[32m+ rich                     \u001b[0m        14.0.0  pyh29332c3_0          conda-forge     200kB\n",
            "  \u001b[32m+ rich-click               \u001b[0m         1.8.8  pyhd8ed1ab_0          conda-forge      34kB\n",
            "  \u001b[32m+ rpds-py                  \u001b[0m        0.24.0  py313h6071e0b_0       conda-forge     393kB\n",
            "  \u001b[32m+ samtools                 \u001b[0m          1.21  h96c455f_1            bioconda        487kB\n",
            "  \u001b[32m+ seqtk                    \u001b[0m           1.4  h577a1d6_3            bioconda         44kB\n",
            "  \u001b[32m+ spectra                  \u001b[0m        0.0.11  pyhd8ed1ab_2          conda-forge      22kB\n",
            "  \u001b[32m+ sqlite                   \u001b[0m        3.49.1  h9eae976_2            conda-forge     860kB\n",
            "  \u001b[32m+ tiktoken                 \u001b[0m         0.9.0  py313h3f234b3_0       conda-forge     969kB\n",
            "  \u001b[32m+ tk                       \u001b[0m        8.6.13  noxft_h4845f30_101    conda-forge       3MB\n",
            "  \u001b[32m+ tqdm                     \u001b[0m        4.67.1  pyhd8ed1ab_1          conda-forge      89kB\n",
            "  \u001b[32m+ trimmomatic              \u001b[0m          0.39  hdfd78af_2            bioconda        147kB\n",
            "  \u001b[32m+ typeguard                \u001b[0m         4.4.2  pyhd8ed1ab_0          conda-forge      35kB\n",
            "  \u001b[32m+ typing-extensions        \u001b[0m        4.13.2  h0e9735f_0            conda-forge      90kB\n",
            "  \u001b[32m+ typing-inspection        \u001b[0m         0.4.0  pyhd8ed1ab_0          conda-forge      18kB\n",
            "  \u001b[32m+ typing_extensions        \u001b[0m        4.13.2  pyh29332c3_0          conda-forge      52kB\n",
            "  \u001b[32m+ tzdata                   \u001b[0m         2025b  h78e105d_0            conda-forge     123kB\n",
            "  \u001b[32m+ urllib3                  \u001b[0m         2.4.0  pyhd8ed1ab_0          conda-forge     101kB\n",
            "  \u001b[32m+ xorg-libice              \u001b[0m         1.1.2  hb9d3cd8_0            conda-forge      59kB\n",
            "  \u001b[32m+ xorg-libsm               \u001b[0m         1.2.6  he73a12e_0            conda-forge      28kB\n",
            "  \u001b[32m+ xorg-libx11              \u001b[0m        1.8.12  h4f16b4b_0            conda-forge     836kB\n",
            "  \u001b[32m+ xorg-libxau              \u001b[0m        1.0.12  hb9d3cd8_0            conda-forge      15kB\n",
            "  \u001b[32m+ xorg-libxdmcp            \u001b[0m         1.1.5  hb9d3cd8_0            conda-forge      20kB\n",
            "  \u001b[32m+ xorg-libxext             \u001b[0m         1.3.6  hb9d3cd8_0            conda-forge      50kB\n",
            "  \u001b[32m+ xorg-libxfixes           \u001b[0m         6.0.1  hb9d3cd8_0            conda-forge      20kB\n",
            "  \u001b[32m+ xorg-libxi               \u001b[0m         1.8.2  hb9d3cd8_0            conda-forge      47kB\n",
            "  \u001b[32m+ xorg-libxrandr           \u001b[0m         1.5.4  hb9d3cd8_0            conda-forge      30kB\n",
            "  \u001b[32m+ xorg-libxrender          \u001b[0m        0.9.12  hb9d3cd8_0            conda-forge      33kB\n",
            "  \u001b[32m+ xorg-libxt               \u001b[0m         1.3.1  hb9d3cd8_0            conda-forge     380kB\n",
            "  \u001b[32m+ xorg-libxtst             \u001b[0m         1.2.5  hb9d3cd8_3            conda-forge      33kB\n",
            "  \u001b[32m+ yaml                     \u001b[0m         0.2.5  h7f98852_2            conda-forge      89kB\n",
            "  \u001b[32m+ zipp                     \u001b[0m        3.21.0  pyhd8ed1ab_1          conda-forge      22kB\n",
            "  \u001b[32m+ zlib                     \u001b[0m         1.3.1  hb9d3cd8_2            conda-forge      92kB\n",
            "  \u001b[32m+ zstandard                \u001b[0m        0.23.0  py313h536fd9c_2       conda-forge     737kB\n",
            "  \u001b[32m+ zstd                     \u001b[0m         1.5.7  hb8e6e7a_2            conda-forge     568kB\n",
            "\n",
            "  Summary:\n",
            "\n",
            "  Install: 166 packages\n",
            "\n",
            "  Total download: 511MB\n",
            "\n",
            "──────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "\n",
            "\n",
            "Transaction starting\n",
            "\u001b[?25l\u001b[2K\u001b[0G[+] 0.0s\n",
            "Downloading        5%\n",
            "Extracting         0%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.1s\n",
            "Downloading  (5)   1%\n",
            "Extracting         0%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.2s\n",
            "Downloading  (5)  10%\n",
            "Extracting         0%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.3s\n",
            "Downloading  (5)  18%\n",
            "Extracting         0%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.4s\n",
            "Downloading  (5)  27%\n",
            "Extracting         0%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gmathjax                                             22.3MB @  54.0MB/s  0.4s\n",
            "[+] 0.5s\n",
            "Downloading  (5)  36%\n",
            "Extracting         0%\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpython                                              33.3MB @  56.9MB/s  0.5s\n",
            "[+] 0.6s\n",
            "Downloading  (5)  42%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.7s\n",
            "Downloading  (5)  49%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpillow                                              41.8MB @  55.9MB/s  0.7s\n",
            "[+] 0.8s\n",
            "Downloading  (5)  52%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Ghisat2                                              16.7MB @  44.5MB/s  0.4s\n",
            "[+] 0.9s\n",
            "Downloading  (5)  55%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.0s\n",
            "Downloading  (5)  58%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gbowtie2                                             15.0MB @  30.5MB/s  0.5s\n",
            "[+] 1.1s\n",
            "Downloading  (5)  59%\n",
            "Extracting   (3)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.2s\n",
            "Downloading  (5)  63%\n",
            "Extracting   (4)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gperl                                                13.3MB @  27.0MB/s  0.5s\n",
            "icu                                                 12.1MB @  25.4MB/s  0.5s\n",
            "[+] 1.3s\n",
            "Downloading  (5)  63%\n",
            "Extracting   (5)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gkaleido-core                                        62.1MB @  46.4MB/s  1.3s\n",
            "[+] 1.4s\n",
            "Downloading  (5)  67%\n",
            "Extracting   (5)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.5s\n",
            "Downloading  (5)  69%\n",
            "Extracting   (7)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Glibopenblas                                          5.9MB @  22.1MB/s  0.2s\n",
            "fastqc                                              11.7MB @  25.7MB/s  0.4s\n",
            "plotly                                               5.1MB @  23.5MB/s  0.2s\n",
            "numpy                                                8.5MB @  28.8MB/s  0.3s\n",
            "[+] 1.6s\n",
            "Downloading  (5)  71%\n",
            "Extracting   (7)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.7s\n",
            "Downloading  (5)  73%\n",
            "Extracting   (9)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Glibglib                                              3.9MB @  20.0MB/s  0.2s\n",
            "libcups                                              4.5MB @  19.2MB/s  0.2s\n",
            "libstdcxx                                            3.9MB @  21.1MB/s  0.2s\n",
            "multiqc                                              3.8MB @  20.2MB/s  0.2s\n",
            "[+] 1.8s\n",
            "Downloading  (5)  74%\n",
            "Extracting  (11)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.9s\n",
            "Downloading  (5)  75%\n",
            "Extracting  (15)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gopenssl                                              3.1MB @  12.6MB/s  0.2s\n",
            "tk                                                   3.3MB @  13.7MB/s  0.2s\n",
            "htslib                                               3.2MB @  13.0MB/s  0.2s\n",
            "gsl                                                  3.4MB @  14.2MB/s  0.2s\n",
            "[+] 2.0s\n",
            "Downloading  (5)  76%\n",
            "Extracting  (16)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gharfbuzz                                             1.7MB @  14.0MB/s  0.1s\n",
            "nss                                                  2.0MB @  10.0MB/s  0.1s\n",
            "font-ttf-ubuntu                                      1.6MB @  12.2MB/s  0.1s\n",
            "pydantic-core                                        1.9MB @  10.8MB/s  0.1s\n",
            "[+] 2.1s\n",
            "Downloading  (5)  78%\n",
            "Extracting  (19)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gnetworkx                                             1.3MB @  15.0MB/s  0.1s\n",
            "libgfortran5                                         1.5MB @  15.8MB/s  0.1s\n",
            "bedtools                                             1.6MB @   7.2MB/s  0.1s\n",
            "krb5                                                 1.4MB @   3.0MB/s  0.1s\n",
            "pip                                                  1.2MB @  21.1MB/s  0.1s\n",
            "[+] 2.2s\n",
            "Downloading  (4)  81%\n",
            "Extracting  (27)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpcre2                                                1.2MB @   1.4MB/s  0.1s\n",
            "bcftools                                             1.0MB @   3.5MB/s  0.1s\n",
            "cairo                                              978.1kB @  ??.?MB/s  0.1s\n",
            "tiktoken                                           968.7kB @  18.8MB/s  0.1s\n",
            "ncurses                                            891.6kB @  ??.?MB/s  0.0s\n",
            "libsqlite                                          918.7kB @  ??.?MB/s  0.1s\n",
            "[+] 2.3s\n",
            "Downloading  (4)  84%\n",
            "Extracting  (32)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpygments                                           888.6kB @  ??.?MB/s  0.1s\n",
            "sqlite                                             859.6kB @   4.3MB/s  0.1s\n",
            "libgcc                                             847.9kB @  14.0MB/s  0.1s\n",
            "xorg-libx11                                        835.9kB @  11.0MB/s  0.1s\n",
            "zstandard                                          736.9kB @   9.4MB/s  0.1s\n",
            "[+] 2.4s\n",
            "Downloading  (5)  84%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gfont-ttf-source-code-pro                           700.8kB @  ??.?MB/s  0.1s\n",
            "libiconv                                           713.1kB @  ??.?MB/s  0.1s\n",
            "ld_impl_linux-64                                   671.2kB @   3.4MB/s  0.1s\n",
            "libnghttp2                                         647.6kB @   6.9MB/s  0.1s\n",
            "libjpeg-turbo                                      628.9kB @  ??.?MB/s  0.0s\n",
            "alsa-lib                                           566.5kB @  ??.?MB/s  0.0s\n",
            "zstd                                               567.6kB @  ??.?MB/s  0.1s\n",
            "samtools                                           487.2kB @   9.5MB/s  0.1s\n",
            "libgomp                                            459.9kB @  ??.?MB/s  0.0s\n",
            "[+] 2.5s\n",
            "Downloading  (4)  87%\n",
            "Extracting  (47)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Glibcurl                                            438.1kB @   1.9MB/s  0.1s\n",
            "libwebp-base                                       430.0kB @   2.2MB/s  0.1s\n",
            "libtiff                                            429.0kB @   2.2MB/s  0.1s\n",
            "regex                                              403.9kB @   2.3MB/s  0.1s\n",
            "[+] 2.6s\n",
            "Downloading  (5)  88%\n",
            "Extracting  (53)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpixman                                             398.7kB @  ??.?MB/s  0.0s\n",
            "rpds-py                                            393.3kB @   5.4MB/s  0.1s\n",
            "font-ttf-dejavu-sans-mono                          397.4kB @   7.3MB/s  0.1s\n",
            "libxcb                                             395.9kB @  ??.?MB/s  0.1s\n",
            "libfreetype6                                       380.1kB @   7.4MB/s  0.1s\n",
            "xorg-libxt                                         379.7kB @  ??.?MB/s  0.0s\n",
            "brotli-python                                      350.4kB @  ??.?MB/s  0.1s\n",
            "openjpeg                                           343.0kB @  ??.?MB/s  0.0s\n",
            "[+] 2.7s\n",
            "Downloading  (5)  91%\n",
            "Extracting  (60)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpydantic                                           306.6kB @  ??.?MB/s  0.0s\n",
            "libssh2                                            304.8kB @   5.9MB/s  0.1s\n",
            "libpng                                             288.7kB @  ??.?MB/s  0.1s\n",
            "cffi                                               295.5kB @   5.9MB/s  0.1s\n",
            "readline                                           282.5kB @   3.3MB/s  0.1s\n",
            "fontconfig                                         265.6kB @  ??.?MB/s  0.0s\n",
            "[+] 2.8s\n",
            "Downloading  (5)  94%\n",
            "Extracting  (66)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Glerc                                               264.2kB @   3.4MB/s  0.1s\n",
            "fastp                                              264.0kB @   1.2MB/s  0.1s\n",
            "bzip2                                              252.8kB @  ??.?MB/s  0.0s\n",
            "lcms2                                              248.0kB @  ??.?MB/s  0.0s\n",
            "nspr                                               230.2kB @  ??.?MB/s  0.0s\n",
            "c-ares                                             206.9kB @  ??.?MB/s  0.0s\n",
            "narwhals                                           215.4kB @   3.0MB/s  0.1s\n",
            "pyyaml                                             205.9kB @  ??.?MB/s  0.0s\n",
            "rich                                               200.3kB @  ??.?MB/s  0.0s\n",
            "bwa                                                197.9kB @  ??.?MB/s  0.0s\n",
            "[+] 2.9s\n",
            "Downloading  (5)  97%\n",
            "Extracting  (75)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gfreetype                                           172.4kB @  ??.?MB/s  0.0s\n",
            "certifi                                            157.2kB @  ??.?MB/s  0.0s\n",
            "isa-l                                              157.3kB @  ??.?MB/s  0.1s\n",
            "ca-certificates                                    152.3kB @   2.9MB/s  0.1s\n",
            "expat                                              140.1kB @  ??.?MB/s  0.0s\n",
            "libedit                                            134.7kB @  ??.?MB/s  0.0s\n",
            "openjdk                                            190.2MB @  63.2MB/s  2.9s\n",
            "tzdata                                             123.0kB @  ??.?MB/s  0.0s\n",
            "[+] 3.0s\n",
            "Downloading  (5) 100%\n",
            "Extracting  (81)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gkeyutils                                           117.8kB @  ??.?MB/s  0.0s\n",
            "liblzma                                            112.8kB @  ??.?MB/s  0.0s\n",
            "libev                                              112.8kB @  ??.?MB/s  0.0s\n",
            "jinja2                                             112.7kB @  ??.?MB/s  0.0s\n",
            "trimmomatic                                        147.4kB @  ??.?MB/s  0.1s\n",
            "pycparser                                          110.1kB @  ??.?MB/s  0.0s\n",
            "libxcrypt                                          100.4kB @  ??.?MB/s  0.0s\n",
            "urllib3                                            100.8kB @  ??.?MB/s  0.0s\n",
            "graphite2                                           96.9kB @  ??.?MB/s  0.0s\n",
            "font-ttf-inconsolata                                96.5kB @  ??.?MB/s  0.0s\n",
            "[+] 3.1s\n",
            "Downloading  (5) 100%\n",
            "Extracting  (92)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gtyping-extensions                                   89.9kB @  ??.?MB/s  0.0s\n",
            "zlib                                                92.3kB @  ??.?MB/s  0.0s\n",
            "libmpdec                                            90.0kB @  ??.?MB/s  0.1s\n",
            "tqdm                                                89.5kB @  ??.?MB/s  0.0s\n",
            "yaml                                                89.1kB @  ??.?MB/s  0.0s\n",
            "giflib                                              77.2kB @  ??.?MB/s  0.0s\n",
            "click                                               84.7kB @   1.7MB/s  0.1s\n",
            "markdown                                            80.0kB @   1.6MB/s  0.1s\n",
            "libexpat                                            74.4kB @  ??.?MB/s  0.0s\n",
            "[+] 3.2s\n",
            "Downloading  (5)  100%\n",
            "Extracting  (102)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gjsonschema                                          74.3kB @  ??.?MB/s  0.1s\n",
            "humanfriendly                                       73.6kB @  ??.?MB/s  0.0s\n",
            "markdown-it-py                                      64.4kB @  ??.?MB/s  0.0s\n",
            "libdeflate                                          72.2kB @  ??.?MB/s  0.0s\n",
            "humanize                                            66.0kB @  ??.?MB/s  0.0s\n",
            "packaging                                           62.5kB @  ??.?MB/s  0.0s\n",
            "libzlib                                             61.0kB @  ??.?MB/s  0.0s\n",
            "requests                                            58.7kB @  ??.?MB/s  0.0s\n",
            "xorg-libice                                         58.6kB @  ??.?MB/s  0.0s\n",
            "libffi                                              57.4kB @  ??.?MB/s  0.0s\n",
            "h2                                                  53.9kB @  ??.?MB/s  0.0s\n",
            "libstdcxx-ng                                        53.8kB @  ??.?MB/s  0.0s\n",
            "[+] 3.3s\n",
            "Downloading  (5)  100%\n",
            "Extracting  (112)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gattrs                                               57.2kB @  ??.?MB/s  0.1s\n",
            "libgcc-ng                                           53.8kB @  ??.?MB/s  0.0s\n",
            "libgfortran                                         53.7kB @ 995.7kB/s  0.1s\n",
            "referencing                                         51.7kB @  ??.?MB/s  0.0s\n",
            "typing_extensions                                   52.2kB @  ??.?MB/s  0.0s\n",
            "charset-normalizer                                  50.5kB @  ??.?MB/s  0.0s\n",
            "xorg-libxext                                        50.1kB @  ??.?MB/s  0.0s\n",
            "idna                                                49.8kB @  ??.?MB/s  0.0s\n",
            "seqtk                                               44.1kB @  ??.?MB/s  0.0s\n",
            "xorg-libxi                                          47.2kB @ 859.0kB/s  0.1s\n",
            "coloredlogs                                         43.8kB @  ??.?MB/s  0.0s\n",
            "[+] 3.4s\n",
            "Downloading  (5)  100%\n",
            "Extracting  (125)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gnatsort                                             39.0kB @  ??.?MB/s  0.0s\n",
            "typeguard                                           35.2kB @  ??.?MB/s  0.0s\n",
            "rich-click                                          34.4kB @  ??.?MB/s  0.0s\n",
            "importlib_resources                                 33.8kB @ 655.6kB/s  0.1s\n",
            "colormath                                           39.3kB @  ??.?MB/s  0.1s\n",
            "xorg-libxrender                                     33.0kB @  ??.?MB/s  0.0s\n",
            "xorg-libxtst                                        32.8kB @  ??.?MB/s  0.0s\n",
            "libuuid                                             33.6kB @  ??.?MB/s  0.1s\n",
            "hpack                                               30.7kB @ 576.7kB/s  0.1s\n",
            "importlib-metadata                                  29.1kB @  ??.?MB/s  0.0s\n",
            "xorg-libsm                                          27.6kB @  ??.?MB/s  0.0s\n",
            "xorg-libxrandr                                      29.6kB @  ??.?MB/s  0.1s\n",
            "[+] 3.5s\n",
            "Downloading  (5)  100%\n",
            "Extracting  (134)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpython-dotenv                                       25.6kB @  ??.?MB/s  0.0s\n",
            "markupsafe                                          24.9kB @  ??.?MB/s  0.0s\n",
            "_openmp_mutex                                       23.6kB @  ??.?MB/s  0.0s\n",
            "spectra                                             22.3kB @  ??.?MB/s  0.0s\n",
            "colorama                                            27.0kB @  ??.?MB/s  0.1s\n",
            "zipp                                                21.8kB @  ??.?MB/s  0.0s\n",
            "pysocks                                             21.1kB @  ??.?MB/s  0.0s\n",
            "xorg-libxdmcp                                       19.9kB @  ??.?MB/s  0.0s\n",
            "[+] 3.6s\n",
            "Downloading  (5)  100%\n",
            "Extracting  (145)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gjsonschema-specifications                           19.2kB @  ??.?MB/s  0.1s\n",
            "typing-inspection                                   18.1kB @  ??.?MB/s  0.0s\n",
            "python-kaleido                                      18.3kB @  ??.?MB/s  0.1s\n",
            "annotated-types                                     18.1kB @  ??.?MB/s  0.0s\n",
            "xorg-libxfixes                                      19.6kB @  ??.?MB/s  0.1s\n",
            "hyperframe                                          17.4kB @  ??.?MB/s  0.0s\n",
            "libblas                                             16.9kB @  ??.?MB/s  0.0s\n",
            "libcblas                                            16.8kB @  ??.?MB/s  0.0s\n",
            "liblapack                                           16.8kB @  ??.?MB/s  0.0s\n",
            "xorg-libxau                                         14.8kB @ 276.5kB/s  0.1s\n",
            "pyaml-env                                           14.6kB @  ??.?MB/s  0.0s\n",
            "[+] 3.7s\n",
            "Downloading  (5)  100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gmdurl                                               14.5kB @ 278.3kB/s  0.1s\n",
            "pkgutil-resolve-name                                10.7kB @ 210.5kB/s  0.1s\n",
            "pthread-stubs                                        8.3kB @  ??.?MB/s  0.1s\n",
            "python_abi                                           7.0kB @  ??.?MB/s  0.0s\n",
            "fonts-conda-forge                                    4.1kB @  ??.?MB/s  0.0s\n",
            "fonts-conda-ecosystem                                3.7kB @  ??.?MB/s  0.0s\n",
            "_libgcc_mutex                                        2.6kB @  ??.?MB/s  0.0s\n",
            "libfreetype                                          7.7kB @  ??.?MB/s  0.1s\n",
            "[+] 3.8s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.9s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.0s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.1s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.2s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.3s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.4s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.5s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.6s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.7s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.8s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.9s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.0s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.1s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.2s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.3s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.4s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.5s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.6s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.7s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.8s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.9s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.0s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.1s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.2s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.3s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.4s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.5s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.6s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.7s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.8s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.9s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.0s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.1s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.2s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.3s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.4s\n",
            "Downloading       100%\n",
            "Extracting  (165)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.5s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.6s\n",
            "Downloading       100%\n",
            "Extracting  (164)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.7s\n",
            "Downloading       100%\n",
            "Extracting  (162)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.8s\n",
            "Downloading       100%\n",
            "Extracting  (161)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.9s\n",
            "Downloading       100%\n",
            "Extracting  (160)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.0s\n",
            "Downloading       100%\n",
            "Extracting  (160)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.1s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.2s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.3s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.4s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.5s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.6s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.7s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.8s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.9s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.0s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.1s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.2s\n",
            "Downloading       100%\n",
            "Extracting  (159)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.3s\n",
            "Downloading       100%\n",
            "Extracting  (156)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.4s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.5s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.6s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.7s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.8s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.9s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.0s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.1s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.2s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.3s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.4s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.5s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.6s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.7s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.8s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.9s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.0s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.1s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.2s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.3s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.4s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.5s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.6s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.7s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.8s\n",
            "Downloading       100%\n",
            "Extracting  (154)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 11.9s\n",
            "Downloading       100%\n",
            "Extracting  (152)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.0s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.1s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.2s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.3s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.4s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.5s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.6s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.7s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.8s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 12.9s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.0s\n",
            "Downloading       100%\n",
            "Extracting  (151)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.1s\n",
            "Downloading       100%\n",
            "Extracting  (150)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.2s\n",
            "Downloading       100%\n",
            "Extracting  (149)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.3s\n",
            "Downloading       100%\n",
            "Extracting  (148)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.4s\n",
            "Downloading       100%\n",
            "Extracting  (146)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.5s\n",
            "Downloading       100%\n",
            "Extracting  (145)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.6s\n",
            "Downloading       100%\n",
            "Extracting  (143)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.7s\n",
            "Downloading       100%\n",
            "Extracting  (141)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.8s\n",
            "Downloading       100%\n",
            "Extracting  (141)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 13.9s\n",
            "Downloading       100%\n",
            "Extracting  (140)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.0s\n",
            "Downloading       100%\n",
            "Extracting  (137)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.1s\n",
            "Downloading       100%\n",
            "Extracting  (136)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.2s\n",
            "Downloading       100%\n",
            "Extracting  (133)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.3s\n",
            "Downloading       100%\n",
            "Extracting  (130)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.4s\n",
            "Downloading       100%\n",
            "Extracting  (129)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.5s\n",
            "Downloading       100%\n",
            "Extracting  (129)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.6s\n",
            "Downloading       100%\n",
            "Extracting  (128)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.7s\n",
            "Downloading       100%\n",
            "Extracting  (125)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.8s\n",
            "Downloading       100%\n",
            "Extracting  (124)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 14.9s\n",
            "Downloading       100%\n",
            "Extracting  (121)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.0s\n",
            "Downloading       100%\n",
            "Extracting  (119)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.1s\n",
            "Downloading       100%\n",
            "Extracting  (117)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.2s\n",
            "Downloading       100%\n",
            "Extracting  (114)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.3s\n",
            "Downloading       100%\n",
            "Extracting  (112)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.4s\n",
            "Downloading       100%\n",
            "Extracting  (110)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.5s\n",
            "Downloading       100%\n",
            "Extracting  (108)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.6s\n",
            "Downloading       100%\n",
            "Extracting  (107)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.7s\n",
            "Downloading       100%\n",
            "Extracting  (106)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.8s\n",
            "Downloading       100%\n",
            "Extracting  (103)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 15.9s\n",
            "Downloading       100%\n",
            "Extracting  (101)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.0s\n",
            "Downloading       100%\n",
            "Extracting  (101)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.1s\n",
            "Downloading       100%\n",
            "Extracting  (101)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.2s\n",
            "Downloading       100%\n",
            "Extracting  (101)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.3s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.4s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.5s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.6s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.7s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.8s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 16.9s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.0s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.1s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.2s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.3s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.4s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.5s\n",
            "Downloading       100%\n",
            "Extracting  (100)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.6s\n",
            "Downloading      100%\n",
            "Extracting  (98)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.7s\n",
            "Downloading      100%\n",
            "Extracting  (96)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.8s\n",
            "Downloading      100%\n",
            "Extracting  (95)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 17.9s\n",
            "Downloading      100%\n",
            "Extracting  (95)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.0s\n",
            "Downloading      100%\n",
            "Extracting  (92)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.1s\n",
            "Downloading      100%\n",
            "Extracting  (90)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.2s\n",
            "Downloading      100%\n",
            "Extracting  (89)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.3s\n",
            "Downloading      100%\n",
            "Extracting  (87)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.4s\n",
            "Downloading      100%\n",
            "Extracting  (87)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.5s\n",
            "Downloading      100%\n",
            "Extracting  (84)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.6s\n",
            "Downloading      100%\n",
            "Extracting  (81)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.7s\n",
            "Downloading      100%\n",
            "Extracting  (81)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.8s\n",
            "Downloading      100%\n",
            "Extracting  (79)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 18.9s\n",
            "Downloading      100%\n",
            "Extracting  (79)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.0s\n",
            "Downloading      100%\n",
            "Extracting  (79)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.1s\n",
            "Downloading      100%\n",
            "Extracting  (79)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.2s\n",
            "Downloading      100%\n",
            "Extracting  (79)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.3s\n",
            "Downloading      100%\n",
            "Extracting  (78)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.4s\n",
            "Downloading      100%\n",
            "Extracting  (78)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.5s\n",
            "Downloading      100%\n",
            "Extracting  (76)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.6s\n",
            "Downloading      100%\n",
            "Extracting  (75)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.7s\n",
            "Downloading      100%\n",
            "Extracting  (74)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.8s\n",
            "Downloading      100%\n",
            "Extracting  (73)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 19.9s\n",
            "Downloading      100%\n",
            "Extracting  (72)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.0s\n",
            "Downloading      100%\n",
            "Extracting  (70)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.1s\n",
            "Downloading      100%\n",
            "Extracting  (70)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.2s\n",
            "Downloading      100%\n",
            "Extracting  (68)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.3s\n",
            "Downloading      100%\n",
            "Extracting  (68)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.4s\n",
            "Downloading      100%\n",
            "Extracting  (67)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.5s\n",
            "Downloading      100%\n",
            "Extracting  (65)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.6s\n",
            "Downloading      100%\n",
            "Extracting  (64)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.7s\n",
            "Downloading      100%\n",
            "Extracting  (63)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.8s\n",
            "Downloading      100%\n",
            "Extracting  (62)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 20.9s\n",
            "Downloading      100%\n",
            "Extracting  (60)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.0s\n",
            "Downloading      100%\n",
            "Extracting  (60)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.1s\n",
            "Downloading      100%\n",
            "Extracting  (60)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.2s\n",
            "Downloading      100%\n",
            "Extracting  (60)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.3s\n",
            "Downloading      100%\n",
            "Extracting  (59)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.4s\n",
            "Downloading      100%\n",
            "Extracting  (57)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.5s\n",
            "Downloading      100%\n",
            "Extracting  (55)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.6s\n",
            "Downloading      100%\n",
            "Extracting  (54)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.7s\n",
            "Downloading      100%\n",
            "Extracting  (53)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.8s\n",
            "Downloading      100%\n",
            "Extracting  (51)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 21.9s\n",
            "Downloading      100%\n",
            "Extracting  (50)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.0s\n",
            "Downloading      100%\n",
            "Extracting  (49)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.1s\n",
            "Downloading      100%\n",
            "Extracting  (48)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.2s\n",
            "Downloading      100%\n",
            "Extracting  (47)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.3s\n",
            "Downloading      100%\n",
            "Extracting  (46)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.4s\n",
            "Downloading      100%\n",
            "Extracting  (44)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.5s\n",
            "Downloading      100%\n",
            "Extracting  (43)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.6s\n",
            "Downloading      100%\n",
            "Extracting  (41)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.7s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.8s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 22.9s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.0s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.1s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.2s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.3s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.4s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.5s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.6s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.7s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.8s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 23.9s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.0s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.1s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.2s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.3s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.4s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.5s\n",
            "Downloading      100%\n",
            "Extracting  (40)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.6s\n",
            "Downloading      100%\n",
            "Extracting  (38)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.7s\n",
            "Downloading      100%\n",
            "Extracting  (36)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.8s\n",
            "Downloading      100%\n",
            "Extracting  (33)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 24.9s\n",
            "Downloading      100%\n",
            "Extracting  (33)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.0s\n",
            "Downloading      100%\n",
            "Extracting  (33)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.1s\n",
            "Downloading      100%\n",
            "Extracting  (33)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.2s\n",
            "Downloading      100%\n",
            "Extracting  (33)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.3s\n",
            "Downloading      100%\n",
            "Extracting  (32)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.4s\n",
            "Downloading      100%\n",
            "Extracting  (32)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.5s\n",
            "Downloading      100%\n",
            "Extracting  (32)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.6s\n",
            "Downloading      100%\n",
            "Extracting  (32)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.7s\n",
            "Downloading      100%\n",
            "Extracting  (29)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.8s\n",
            "Downloading      100%\n",
            "Extracting  (28)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 25.9s\n",
            "Downloading      100%\n",
            "Extracting  (28)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.0s\n",
            "Downloading      100%\n",
            "Extracting  (28)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.1s\n",
            "Downloading      100%\n",
            "Extracting  (26)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.2s\n",
            "Downloading      100%\n",
            "Extracting  (23)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.3s\n",
            "Downloading      100%\n",
            "Extracting  (23)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.4s\n",
            "Downloading      100%\n",
            "Extracting  (20)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.5s\n",
            "Downloading      100%\n",
            "Extracting  (18)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.6s\n",
            "Downloading      100%\n",
            "Extracting  (15)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.7s\n",
            "Downloading      100%\n",
            "Extracting  (12)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.8s\n",
            "Downloading      100%\n",
            "Extracting  (10)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 26.9s\n",
            "Downloading      100%\n",
            "Extracting   (8)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.0s\n",
            "Downloading      100%\n",
            "Extracting   (7)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.1s\n",
            "Downloading      100%\n",
            "Extracting   (4)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.2s\n",
            "Downloading      100%\n",
            "Extracting   (3)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.3s\n",
            "Downloading      100%\n",
            "Extracting   (2)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.4s\n",
            "Downloading      100%\n",
            "Extracting   (2)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.5s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.6s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.7s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.8s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 27.9s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 28.0s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 28.1s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 28.2s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 28.3s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 28.4s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 28.5s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 28.6s\n",
            "Downloading      100%\n",
            "Extracting   (1)  ⣾  \u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G\u001b[?25hLinking font-ttf-dejavu-sans-mono-2.37-hab24e00_0\n",
            "Linking font-ttf-ubuntu-0.83-h77eed37_3\n",
            "Linking font-ttf-inconsolata-3.000-h77eed37_0\n",
            "Linking font-ttf-source-code-pro-2.038-h77eed37_0\n",
            "Linking python_abi-3.13-7_cp313\n",
            "Linking tzdata-2025b-h78e105d_0\n",
            "Linking ca-certificates-2025.4.26-hbd8a1cb_0\n",
            "Linking fonts-conda-forge-1-0\n",
            "Linking fonts-conda-ecosystem-1-0\n",
            "Linking libgomp-14.2.0-h767d61c_2\n",
            "Linking mathjax-2.7.7-ha770c72_3\n",
            "Linking ld_impl_linux-64-2.43-h712a8e2_4\n",
            "Linking _libgcc_mutex-0.1-conda_forge\n",
            "Linking _openmp_mutex-4.5-2_gnu\n",
            "Linking libgcc-14.2.0-h767d61c_2\n",
            "Linking libiconv-1.18-h4ce23a2_1\n",
            "Linking c-ares-1.34.5-hb9d3cd8_0\n",
            "Linking libgfortran5-14.2.0-hf1ad2bd_2\n",
            "Linking xorg-libxdmcp-1.1.5-hb9d3cd8_0\n",
            "Linking xorg-libxau-1.0.12-hb9d3cd8_0\n",
            "Linking pthread-stubs-0.4-hb9d3cd8_1002\n",
            "Linking alsa-lib-1.2.14-hb9d3cd8_0\n",
            "Linking libjpeg-turbo-3.1.0-hb9d3cd8_0\n",
            "Linking xorg-libice-1.1.2-hb9d3cd8_0\n",
            "Linking libwebp-base-1.5.0-h851e524_0\n",
            "Linking openssl-3.5.0-h7b32b05_1\n",
            "Linking liblzma-5.8.1-hb9d3cd8_1\n",
            "Linking libffi-3.4.6-h2dba641_1\n",
            "Linking ncurses-6.5-h2d0b736_3\n",
            "Linking libzlib-1.3.1-hb9d3cd8_2\n",
            "Linking libstdcxx-14.2.0-h8f9b012_2\n",
            "Linking libgcc-ng-14.2.0-h69a702a_2\n",
            "Linking libexpat-2.7.0-h5888daf_0\n",
            "Linking libdeflate-1.22-hb9d3cd8_0\n",
            "Linking isa-l-2.31.1-hb9d3cd8_1\n",
            "Linking libgfortran-14.2.0-h69a702a_2\n",
            "Linking libxcb-1.17.0-h8a09558_0\n",
            "Linking libedit-3.1.20250104-pl5321h7949ede_0\n",
            "Linking readline-8.2-h8c095d6_2\n",
            "Linking libssh2-1.11.1-hcf80075_0\n",
            "Linking libpng-1.6.47-h943b412_0\n",
            "Linking zlib-1.3.1-hb9d3cd8_2\n",
            "Linking libsqlite-3.49.1-hee588c1_2\n",
            "Linking pixman-0.46.0-h29eaf8c_0\n",
            "Linking libstdcxx-ng-14.2.0-h4852527_2\n",
            "Linking lerc-4.0.0-h0aef613_1\n",
            "Linking zstd-1.5.7-hb8e6e7a_2\n",
            "Linking nspr-4.36-h5888daf_0\n",
            "Linking libev-4.33-hd590300_2\n",
            "Linking libxcrypt-4.4.36-hd590300_1\n",
            "Linking yaml-0.2.5-h7f98852_2\n",
            "Linking keyutils-1.6.1-h166bdaf_0\n",
            "Linking giflib-5.2.2-hd590300_0\n",
            "Linking tk-8.6.13-noxft_h4845f30_101\n",
            "Linking bzip2-1.0.8-h4bc722e_7\n",
            "Linking libuuid-2.38.1-h0b41bf4_0\n",
            "Linking libmpdec-4.0.0-h4bc722e_0\n",
            "Linking expat-2.7.0-h5888daf_0\n",
            "Linking libopenblas-0.3.29-pthreads_h94d23a6_0\n",
            "Linking xorg-libx11-1.8.12-h4f16b4b_0\n",
            "Linking libfreetype6-2.13.3-h48d6fc4_1\n",
            "Linking sqlite-3.49.1-h9eae976_2\n",
            "Linking graphite2-1.3.13-h59595ed_1003\n",
            "Linking icu-75.1-he02047a_0\n",
            "Linking libtiff-4.7.0-hc4654cb_2\n",
            "Linking nss-3.111-h159eef7_0\n",
            "Linking libnghttp2-1.64.0-h161d5f1_0\n",
            "Linking perl-5.32.1-7_hd590300_perl5\n",
            "Linking krb5-1.21.3-h659f571_0\n",
            "Linking pcre2-10.45-hc749103_0\n",
            "Linking xorg-libsm-1.2.6-he73a12e_0\n",
            "Linking python-3.13.3-hf636f53_101_cp313\n",
            "Linking libblas-3.9.0-31_h59b9bed_openblas\n",
            "Linking xorg-libxfixes-6.0.1-hb9d3cd8_0\n",
            "Linking xorg-libxrender-0.9.12-hb9d3cd8_0\n",
            "Linking xorg-libxext-1.3.6-hb9d3cd8_0\n",
            "Linking libfreetype-2.13.3-ha770c72_1\n",
            "Linking openjpeg-2.5.3-h5fbd93e_0\n",
            "Linking lcms2-2.17-h717163a_0\n",
            "Linking libcups-2.3.3-h4637d8d_4\n",
            "Linking libcurl-8.13.0-h332b0f4_0\n",
            "Linking libglib-2.84.1-h3618099_1\n",
            "Linking xorg-libxt-1.3.1-hb9d3cd8_0\n",
            "Linking liblapack-3.9.0-31_h7ac8fdf_openblas\n",
            "Linking libcblas-3.9.0-31_he106b2a_openblas\n",
            "Linking xorg-libxrandr-1.5.4-hb9d3cd8_0\n",
            "Linking xorg-libxi-1.8.2-hb9d3cd8_0\n",
            "Linking freetype-2.13.3-ha770c72_1\n",
            "Linking gsl-2.7-he838d99_0\n",
            "Linking xorg-libxtst-1.2.5-hb9d3cd8_3\n",
            "Linking fontconfig-2.15.0-h7e30c49_1\n",
            "Linking cairo-1.18.4-h3394656_0\n",
            "Linking kaleido-core-0.2.1-h3644ca4_0\n",
            "Linking harfbuzz-11.1.0-h3beb420_0\n",
            "Linking openjdk-23.0.2-h53dfc1b_2\n",
            "Linking fastp-0.24.1-heae3180_0\n",
            "Linking seqtk-1.4-h577a1d6_3\n",
            "Linking bedtools-2.31.1-h13024bc_3\n",
            "Linking bwa-0.7.19-h577a1d6_0\n",
            "Linking htslib-1.21-h566b1c6_1\n",
            "Linking samtools-1.21-h96c455f_1\n",
            "Linking bcftools-1.21-h3a4d415_1\n",
            "Linking pip-25.1.1-pyh145f28c_0\n",
            "Linking pycparser-2.22-pyh29332c3_1\n",
            "Linking pysocks-1.7.1-pyha55dd90_7\n",
            "Linking hyperframe-6.1.0-pyhd8ed1ab_0\n",
            "Linking hpack-4.1.0-pyhd8ed1ab_0\n",
            "Linking humanfriendly-10.0-pyh707e725_8\n",
            "Linking networkx-3.4.2-pyh267e887_2\n",
            "Linking zipp-3.21.0-pyhd8ed1ab_1\n",
            "Linking pkgutil-resolve-name-1.3.10-pyhd8ed1ab_2\n",
            "Linking attrs-25.3.0-pyh71513ae_0\n",
            "Linking mdurl-0.1.2-pyhd8ed1ab_1\n",
            "Linking narwhals-1.38.0-pyhe01879c_0\n",
            "Linking charset-normalizer-3.4.2-pyhd8ed1ab_0\n",
            "Linking idna-3.10-pyhd8ed1ab_1\n",
            "Linking certifi-2025.4.26-pyhd8ed1ab_0\n",
            "Linking pygments-2.19.1-pyhd8ed1ab_0\n",
            "Linking colorama-0.4.6-pyhd8ed1ab_1\n",
            "Linking typing_extensions-4.13.2-pyh29332c3_0\n",
            "Linking python-dotenv-1.1.0-pyh29332c3_1\n",
            "Linking packaging-25.0-pyh29332c3_1\n",
            "Linking natsort-8.4.0-pyh29332c3_1\n",
            "Linking humanize-4.12.3-pyhd8ed1ab_0\n",
            "Linking click-8.1.8-pyh707e725_0\n",
            "Linking python-kaleido-0.2.1-pyhd8ed1ab_0\n",
            "Linking h2-4.2.0-pyhd8ed1ab_0\n",
            "Linking coloredlogs-15.0.1-pyhd8ed1ab_4\n",
            "Linking importlib_resources-6.5.2-pyhd8ed1ab_0\n",
            "Linking importlib-metadata-8.6.1-pyha770c72_0\n",
            "Linking markdown-it-py-3.0.0-pyhd8ed1ab_1\n",
            "Linking tqdm-4.67.1-pyhd8ed1ab_1\n",
            "Linking typing-extensions-4.13.2-h0e9735f_0\n",
            "Linking typing-inspection-0.4.0-pyhd8ed1ab_0\n",
            "Linking plotly-6.0.1-pyhd8ed1ab_0\n",
            "Linking markdown-3.8-pyhd8ed1ab_0\n",
            "Linking rich-14.0.0-pyh29332c3_0\n",
            "Linking annotated-types-0.7.0-pyhd8ed1ab_1\n",
            "Linking typeguard-4.4.2-pyhd8ed1ab_0\n",
            "Linking rich-click-1.8.8-pyhd8ed1ab_0\n",
            "Linking fastqc-0.12.1-hdfd78af_0\n",
            "Linking trimmomatic-0.39-hdfd78af_2\n",
            "Linking regex-2024.11.6-py313h536fd9c_0\n",
            "Linking markupsafe-3.0.2-py313h8060acc_1\n",
            "Linking rpds-py-0.24.0-py313h6071e0b_0\n",
            "Linking brotli-python-1.1.0-py313h46c70d0_2\n",
            "Linking pyyaml-6.0.2-py313h8060acc_2\n",
            "Linking pillow-11.1.0-py313h8db990d_0\n",
            "Linking numpy-2.2.5-py313h17eae1a_0\n",
            "Linking cffi-1.17.1-py313hfab6e84_0\n",
            "Linking pydantic-core-2.33.1-py313h6071e0b_0\n",
            "Linking zstandard-0.23.0-py313h536fd9c_2\n",
            "Linking hisat2-2.2.1-h503566f_8\n",
            "Linking bowtie2-2.5.4-he96a11b_5\n",
            "Linking jinja2-3.1.6-pyhd8ed1ab_0\n",
            "Linking referencing-0.36.2-pyh29332c3_0\n",
            "Linking pyaml-env-1.2.2-pyhd8ed1ab_0\n",
            "Linking colormath-3.0.0-pyhd8ed1ab_4\n",
            "Linking pydantic-2.11.3-pyh3cfb1c2_0\n",
            "Linking urllib3-2.4.0-pyhd8ed1ab_0\n",
            "Linking jsonschema-specifications-2025.4.1-pyh29332c3_0\n",
            "Linking spectra-0.0.11-pyhd8ed1ab_2\n",
            "Linking requests-2.32.3-pyhd8ed1ab_1\n",
            "Linking jsonschema-4.23.0-pyhd8ed1ab_1\n",
            "Linking tiktoken-0.9.0-py313h3f234b3_0\n",
            "Linking multiqc-1.28-pyhdfd78af_0\n",
            "\n",
            "Transaction finished\n",
            "\n",
            "\n",
            "To activate this environment, use:\n",
            "\n",
            "    micromamba activate bio\n",
            "\n",
            "Or to execute a single command in this environment, use:\n",
            "\n",
            "    micromamba run -n bio mycommand\n",
            "\n",
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created. If --delete is \n",
            "                    also specified then the zip file will be removed after the \n",
            "                    contents are unzipped. \n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "\n",
            "    --dup_length    Sets a length to which the sequences will be truncated when \n",
            "                    defining them to be duplicates, affecting the duplication and\n",
            "                    overrepresented sequences plot.  This can be useful if you have\n",
            "                    long reads with higher levels of miscalls, or contamination with\n",
            "                    adapter dimers containing UMI sequences.\n",
            "\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "\n",
            "    --memory        Sets the base amount of memory, in Megabytes, used to process \n",
            "                    each file.  Defaults to 512MB.  You may need to increase this if\n",
            "                    you have a file with very long sequences in it.\n",
            "                \n",
            "    --svg           Save the graphs in the report in SVG format.\n",
            "\n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Suppress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n",
            "\n",
            "Program: bwa (alignment via Burrows-Wheeler transformation)\n",
            "Version: 0.7.19-r1273\n",
            "Contact: Heng Li <hli@ds.dfci.harvard.edu>\n",
            "\n",
            "Usage:   bwa <command> [options]\n",
            "\n",
            "Command: index         index sequences in the FASTA format\n",
            "         mem           BWA-MEM algorithm\n",
            "         fastmap       identify super-maximal exact matches\n",
            "         pemerge       merge overlapping paired ends (EXPERIMENTAL)\n",
            "         aln           gapped/ungapped alignment\n",
            "         samse         generate alignment (single ended)\n",
            "         sampe         generate alignment (paired ended)\n",
            "         bwasw         BWA-SW for long queries (DEPRECATED)\n",
            "\n",
            "         shm           manage indices in shared memory\n",
            "         fa2pac        convert FASTA to PAC format\n",
            "         pac2bwt       generate BWT from PAC\n",
            "         pac2bwtgen    alternative algorithm for generating BWT\n",
            "         bwtupdate     update .bwt to the new format\n",
            "         bwt2sa        generate SA from BWT and Occ\n",
            "\n",
            "Note: To use BWA, you need to first index the genome with `bwa index'.\n",
            "      There are three alignment algorithms in BWA: `mem', `bwasw', and\n",
            "      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'\n",
            "      first. Please `man ./bwa.1' for the manual.\n",
            "\n",
            "samtools 1.21\n",
            "Using htslib 1.21\n",
            "Copyright (C) 2024 Genome Research Ltd.\n",
            "\n",
            "Samtools compilation details:\n",
            "    Features:       build=configure curses=yes \n",
            "    CC:             /opt/conda/conda-bld/samtools_1734249775320/_build_env/bin/x86_64-conda-linux-gnu-cc\n",
            "    CPPFLAGS:       -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /root/.local/share/mamba/envs/bio/include\n",
            "    CFLAGS:         -Wall -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /root/.local/share/mamba/envs/bio/include -fdebug-prefix-map=/opt/conda/conda-bld/samtools_1734249775320/work=/usr/local/src/conda/samtools-1.21 -fdebug-prefix-map=/root/.local/share/mamba/envs/bio=/usr/local/src/conda-prefix\n",
            "    LDFLAGS:        -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/root/.local/share/mamba/envs/bio/lib -Wl,-rpath-link,/root/.local/share/mamba/envs/bio/lib -L/root/.local/share/mamba/envs/bio/lib\n",
            "    HTSDIR:         \n",
            "    LIBS:           \n",
            "    CURSES_LIB:     -ltinfow -lncursesw\n",
            "\n",
            "HTSlib compilation details:\n",
            "    Features:       build=configure libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes plugins=yes plugin-path=/root/.local/share/mamba/envs/bio/libexec/htslib htscodecs=1.6.1\n",
            "    CC:             /opt/conda/conda-bld/htslib_1734145038589/_build_env/bin/x86_64-conda-linux-gnu-cc\n",
            "    CPPFLAGS:       -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /root/.local/share/mamba/envs/bio/include\n",
            "    CFLAGS:         -Wall -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /root/.local/share/mamba/envs/bio/include -fdebug-prefix-map=/opt/conda/conda-bld/htslib_1734145038589/work=/usr/local/src/conda/htslib-1.21 -fdebug-prefix-map=/root/.local/share/mamba/envs/bio=/usr/local/src/conda-prefix -fvisibility=hidden\n",
            "    LDFLAGS:        -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/root/.local/share/mamba/envs/bio/lib -Wl,-rpath-link,/root/.local/share/mamba/envs/bio/lib -L/root/.local/share/mamba/envs/bio/lib -fvisibility=hidden -rdynamic\n",
            "\n",
            "HTSlib URL scheme handlers present:\n",
            "    built-in:\t file, preload, data\n",
            "    Amazon S3:\t s3+https, s3, s3+http\n",
            "    libcurl:\t gophers, smtp, wss, smb, rtsp, tftp, pop3, smbs, imaps, pop3s, ws, ftps, https, ftp, gopher, sftp, imap, http, smtps, scp, dict, mqtt, telnet\n",
            "    Google Cloud Storage:\t gs+http, gs+https, gs\n",
            "    S3 Multipart Upload:\t s3w+https, s3w+http, s3w\n",
            "    crypt4gh-needed:\t crypt4gh\n",
            "    mem:\t mem\n"
          ]
        }
      ],
      "source": [
        "# 1. Установка micromamba\n",
        "!wget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba\n",
        "\n",
        "# 2. Создание окружения и установка инструментов\n",
        "!./bin/micromamba create -y -n bio \\\n",
        "  -c bioconda -c conda-forge \\\n",
        "  fastqc trimmomatic bwa samtools bcftools fastp hisat2 bowtie2 bedtools seqtk multiqc\n",
        "\n",
        "# 3. Активация окружения и проверка (для примера)\n",
        "!./bin/micromamba run -n bio fastqc --help\n",
        "!./bin/micromamba run -n bio bwa\n",
        "!./bin/micromamba run -n bio samtools --version"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "OH4DAqBIr--i"
      },
      "source": [
        "# СКАЧИВАЕМ РЕФЕРЕНСНЫЙ ГЕНОМ ( **Escherichia coli**) через NCBI Datasets\n",
        "datasets accession GCF_000005845.2 --include gff3,rna,cds,protein,genome,seq-report\n",
        "\n",
        "https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000005845.2/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "2LHjCmFbeAt0",
        "outputId": "32822410-6e60-4553-c0ff-89fb52a6ebec"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2025-05-07 13:40:13--  https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Escherichia_coli/reference/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.11, 130.14.250.12, 130.14.250.13, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.11|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1379902 (1.3M) [application/x-gzip]\n",
            "Saving to: ‘GCF_000005845.2_ASM584v2_genomic.fna.gz’\n",
            "\n",
            "GCF_000005845.2_ASM 100%[===================>]   1.32M  3.80MB/s    in 0.3s    \n",
            "\n",
            "2025-05-07 13:40:13 (3.80 MB/s) - ‘GCF_000005845.2_ASM584v2_genomic.fna.gz’ saved [1379902/1379902]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Escherichia_coli/reference/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz\n",
        "!gunzip GCF_000005845.2_ASM584v2_genomic.fna.gz\n",
        "!mv GCF_000005845.2_ASM584v2_genomic.fna ecoli.fna\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0l2bPuv6wELR"
      },
      "source": [
        "# Cкачиваем риды (NGS-данные) для Escherichia coli из NCBI\n",
        "\n",
        "https://www.ncbi.nlm.nih.gov/sra/SRX1317390[accn]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "4SSvMl5SwJ6P",
        "outputId": "6a23892f-08b5-47f6-e610-c1bd13487c2d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "Suggested packages:\n",
            "  blends-doc menu-l10n gksu | kde-runtime | ktsuss\n",
            "The following NEW packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "  sra-toolkit\n",
            "0 upgraded, 7 newly installed, 0 to remove and 34 not upgraded.\n",
            "Need to get 8,290 kB of archives.\n",
            "After this operation, 23.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n",
            "Fetched 8,290 kB in 2s (4,395 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package menu.\n",
            "(Reading database ... 126102 files and directories currently installed.)\n",
            "Preparing to unpack .../0-menu_2.1.47ubuntu4_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu4) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../1-blends-common_0.7.4ubuntu1_all.deb ...\n",
            "Unpacking blends-common (0.7.4ubuntu1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../2-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../3-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../4-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../5-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../6-med-config_3.7.1_all.deb ...\n",
            "Unpacking med-config (3.7.1) ...\n",
            "Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up menu (2.1.47ubuntu4) ...\n",
            "Setting up blends-common (0.7.4ubuntu1) ...\n",
            "Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Setting up med-config (3.7.1) ...\n",
            "Adding group `med' (GID 107) ...\n",
            "Done.\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "fastq-dump was killed (signal 2 SIGINT)\n"
          ]
        }
      ],
      "source": [
        "!apt-get install -y sra-toolkit\n",
        "!fastq-dump --split-files SRR2584863"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3-oexcbCy6kR"
      },
      "source": [
        "#Проверяем качество ридов триммируем"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "n7KEBXX0yYy5",
        "outputId": "acb736b3-b555-4191-db6b-0d4dcbae19a5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "FastQC уже установлен\n",
            "--2025-05-07 14:00:36--  http://opengene.org/fastp/fastp.0.23.4\n",
            "Resolving opengene.org (opengene.org)... 8.210.133.117\n",
            "Connecting to opengene.org (opengene.org)|8.210.133.117|:80... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://opengene.org/fastp/fastp.0.23.4 [following]\n",
            "--2025-05-07 14:00:37--  https://opengene.org/fastp/fastp.0.23.4\n",
            "Connecting to opengene.org (opengene.org)|8.210.133.117|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10946176 (10M) [application/octet-stream]\n",
            "Saving to: ‘fastp’\n",
            "\n",
            "fastp               100%[===================>]  10.44M  4.55MB/s    in 2.3s    \n",
            "\n",
            "2025-05-07 14:00:40 (4.55 MB/s) - ‘fastp’ saved [10946176/10946176]\n",
            "\n",
            "FastQC v0.12.1\n",
            "fastp 0.23.4\n",
            "null\n",
            "null\n",
            "Started analysis of SRR2584863_1.fastq\n",
            "Approx 5% complete for SRR2584863_1.fastq\n",
            "Approx 10% complete for SRR2584863_1.fastq\n",
            "Approx 15% complete for SRR2584863_1.fastq\n",
            "Approx 20% complete for SRR2584863_1.fastq\n",
            "Approx 25% complete for SRR2584863_1.fastq\n",
            "Approx 30% complete for SRR2584863_1.fastq\n",
            "Approx 35% complete for SRR2584863_1.fastq\n",
            "Approx 40% complete for SRR2584863_1.fastq\n",
            "Approx 45% complete for SRR2584863_1.fastq\n",
            "Approx 50% complete for SRR2584863_1.fastq\n",
            "Approx 55% complete for SRR2584863_1.fastq\n",
            "Approx 60% complete for SRR2584863_1.fastq\n",
            "Approx 65% complete for SRR2584863_1.fastq\n",
            "Approx 70% complete for SRR2584863_1.fastq\n",
            "Approx 75% complete for SRR2584863_1.fastq\n",
            "Approx 80% complete for SRR2584863_1.fastq\n",
            "Approx 85% complete for SRR2584863_1.fastq\n",
            "Approx 90% complete for SRR2584863_1.fastq\n",
            "Approx 95% complete for SRR2584863_1.fastq\n",
            "Analysis complete for SRR2584863_1.fastq\n",
            "Started analysis of SRR2584863_2.fastq\n",
            "Approx 5% complete for SRR2584863_2.fastq\n",
            "Approx 10% complete for SRR2584863_2.fastq\n",
            "Approx 15% complete for SRR2584863_2.fastq\n",
            "Approx 20% complete for SRR2584863_2.fastq\n",
            "Approx 25% complete for SRR2584863_2.fastq\n",
            "Approx 30% complete for SRR2584863_2.fastq\n",
            "Approx 35% complete for SRR2584863_2.fastq\n",
            "Approx 40% complete for SRR2584863_2.fastq\n",
            "Approx 45% complete for SRR2584863_2.fastq\n",
            "Approx 50% complete for SRR2584863_2.fastq\n",
            "Approx 55% complete for SRR2584863_2.fastq\n",
            "Approx 60% complete for SRR2584863_2.fastq\n",
            "Approx 65% complete for SRR2584863_2.fastq\n",
            "Approx 70% complete for SRR2584863_2.fastq\n",
            "Approx 75% complete for SRR2584863_2.fastq\n",
            "Approx 80% complete for SRR2584863_2.fastq\n",
            "Approx 85% complete for SRR2584863_2.fastq\n",
            "Approx 90% complete for SRR2584863_2.fastq\n",
            "Approx 95% complete for SRR2584863_2.fastq\n",
            "Analysis complete for SRR2584863_2.fastq\n",
            "Read1 before filtering:\n",
            "total reads: 1553259\n",
            "total bases: 232988850\n",
            "Q20 bases: 222136776(95.3422%)\n",
            "Q30 bases: 208265689(89.3887%)\n",
            "\n",
            "Read2 before filtering:\n",
            "total reads: 1553259\n",
            "total bases: 232988850\n",
            "Q20 bases: 193356778(82.9897%)\n",
            "Q30 bases: 172128273(73.8783%)\n",
            "\n",
            "Read1 after filtering:\n",
            "total reads: 1301326\n",
            "total bases: 186018280\n",
            "Q20 bases: 182719034(98.2264%)\n",
            "Q30 bases: 174020955(93.5505%)\n",
            "\n",
            "Read2 after filtering:\n",
            "total reads: 1301326\n",
            "total bases: 186018280\n",
            "Q20 bases: 175211377(94.1904%)\n",
            "Q30 bases: 157598267(84.7219%)\n",
            "\n",
            "Filtering result:\n",
            "reads passed filter: 2602652\n",
            "reads failed due to low quality: 502702\n",
            "reads failed due to too many N: 1164\n",
            "reads failed due to too short: 0\n",
            "reads with adapter trimmed: 319910\n",
            "bases trimmed due to adapters: 18679052\n",
            "\n",
            "Duplication rate: 0.369739%\n",
            "\n",
            "Insert size peak (evaluated by paired-end reads): 100\n",
            "\n",
            "JSON report: fastp_report.json\n",
            "HTML report: fastp_report.html\n",
            "\n",
            "fastp -i SRR2584863_1.fastq -I SRR2584863_2.fastq -o clean_1.fastq -O clean_2.fastq -h fastp_report.html -j fastp_report.json \n",
            "fastp v0.23.4, time used: 31 seconds\n"
          ]
        }
      ],
      "source": [
        "# Установка FastQC (если ещё не установлен)\n",
        "!if [ ! -f \"/usr/local/bin/fastqc\" ]; then \\\n",
        "    wget -O fastqc.zip https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.12.1.zip && \\\n",
        "    unzip -o fastqc.zip && \\\n",
        "    chmod +x FastQC/fastqc && \\\n",
        "    ln -s $(pwd)/FastQC/fastqc /usr/local/bin/fastqc ; \\\n",
        "else echo \"FastQC уже установлен\"; fi\n",
        "\n",
        "# Установка fastp v0.23.4\n",
        "!wget http://opengene.org/fastp/fastp.0.23.4 -O fastp\n",
        "!chmod +x fastp\n",
        "!sudo mv fastp /usr/local/bin/\n",
        "\n",
        "# Проверка версий\n",
        "!fastqc --version\n",
        "!fastp --version\n",
        "\n",
        "# Запуск FastQC\n",
        "!fastqc SRR2584863_1.fastq SRR2584863_2.fastq\n",
        "\n",
        "# Тримминг fastp\n",
        "!fastp \\\n",
        "  -i SRR2584863_1.fastq \\\n",
        "  -I SRR2584863_2.fastq \\\n",
        "  -o clean_1.fastq \\\n",
        "  -O clean_2.fastq \\\n",
        "  -h fastp_report.html \\\n",
        "  -j fastp_report.json\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "U1xotzVKAoHK",
        "outputId": "c424028f-110f-436d-f0a2-63ce08363578"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "fastp 0.23.4\n"
          ]
        }
      ],
      "source": [
        "!fastp --version\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "i-XGoll-LYzD",
        "outputId": "ae9cb00e-6d7d-4d99-e5be-48aa40de8c7f"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "-rw-r--r-- 1 root root 663M May  7 13:41 SRR2584863_1.fastq\n",
            "-rw-r--r-- 1 root root 663M May  7 13:41 SRR2584863_2.fastq\n"
          ]
        }
      ],
      "source": [
        "#Проверяем создались ли файлы ридов\n",
        "!ls -lh SRR2584863_*.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "D_OlbeUWLe3o",
        "outputId": "16a8d66d-7622-447f-e1dd-8419b8cfdead"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "-rw-r--r-- 1 root root 538M May  7 14:01 clean_1.fastq\n",
            "-rw-r--r-- 1 root root 538M May  7 14:01 clean_2.fastq\n"
          ]
        }
      ],
      "source": [
        "#Проверяем создались ли очищенные после тримменга файлы ридов\n",
        "!ls -lh clean_*.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "WfRKOd7H4s8U"
      },
      "source": [
        "# Выравнивание ридов по генам в формате .bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "ROhHb8jT43T9",
        "outputId": "fcd947e7-a764-4b08-a03b-a73738000540"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "bwa is already the newest version (0.7.17-6).\n",
            "samtools is already the newest version (1.13-4).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 34 not upgraded.\n",
            "[bwa_index] Pack FASTA... 0.05 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[bwa_index] 1.48 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.03 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.03 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 0.72 sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa index ecoli.fna\n",
            "[main] Real time: 2.464 sec; CPU: 2.313 sec\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 70692 sequences (10000074 bp)...\n",
            "[M::process] read 70688 sequences (10000202 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (117, 31403, 11, 113)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (462, 901, 2491)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6549)\n",
            "[M::mem_pestat] mean and std.dev: (1462.01, 1473.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8578)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (230, 496, 706)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1658)\n",
            "[M::mem_pestat] mean and std.dev: (485.94, 282.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2134)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (175, 475, 654)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1612)\n",
            "[M::mem_pestat] mean and std.dev: (382.20, 242.07)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2091)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (397, 857, 2468)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6610)\n",
            "[M::mem_pestat] mean and std.dev: (1329.07, 1405.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8681)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70692 reads in 5.055 CPU sec, 5.068 real sec\n",
            "[M::process] read 70396 sequences (10000208 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (105, 31502, 14, 109)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (542, 1009, 3063)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8105)\n",
            "[M::mem_pestat] mean and std.dev: (1884.21, 1988.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10626)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (226, 484, 702)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1654)\n",
            "[M::mem_pestat] mean and std.dev: (481.53, 281.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2130)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (42, 388, 3031)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9009)\n",
            "[M::mem_pestat] mean and std.dev: (1316.00, 1852.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11998)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (441, 810, 2187)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5679)\n",
            "[M::mem_pestat] mean and std.dev: (1218.30, 1225.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7425)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70688 reads in 4.587 CPU sec, 4.439 real sec\n",
            "[M::process] read 70676 sequences (10000130 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (117, 31223, 12, 114)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (437, 803, 2340)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6146)\n",
            "[M::mem_pestat] mean and std.dev: (1374.11, 1420.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8049)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (239, 504, 710)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1652)\n",
            "[M::mem_pestat] mean and std.dev: (493.29, 282.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2123)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (173, 480, 1314)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3596)\n",
            "[M::mem_pestat] mean and std.dev: (462.90, 399.08)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4737)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (392, 691, 2226)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5894)\n",
            "[M::mem_pestat] mean and std.dev: (1390.41, 1493.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7728)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70396 reads in 5.968 CPU sec, 5.930 real sec\n",
            "[M::process] read 70438 sequences (10000212 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (102, 31394, 7, 121)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (458, 895, 2745)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7319)\n",
            "[M::mem_pestat] mean and std.dev: (1603.70, 1554.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9606)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (230, 489, 708)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1664)\n",
            "[M::mem_pestat] mean and std.dev: (484.78, 281.57)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2142)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (498, 1234, 3153)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8463)\n",
            "[M::mem_pestat] mean and std.dev: (2008.66, 2080.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11118)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70676 reads in 4.610 CPU sec, 4.491 real sec\n",
            "[M::process] read 70224 sequences (10000270 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (115, 31378, 11, 125)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (361, 756, 1689)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4345)\n",
            "[M::mem_pestat] mean and std.dev: (956.79, 841.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5673)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (237, 504, 720)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1686)\n",
            "[M::mem_pestat] mean and std.dev: (496.03, 286.92)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2169)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (419, 1051, 2189)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5729)\n",
            "[M::mem_pestat] mean and std.dev: (1090.70, 934.95)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7499)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (510, 1330, 3194)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8562)\n",
            "[M::mem_pestat] mean and std.dev: (2038.78, 1952.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11246)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70438 reads in 4.688 CPU sec, 4.541 real sec\n",
            "[M::process] read 70572 sequences (10000188 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (120, 31198, 4, 119)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (358, 707, 1929)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5071)\n",
            "[M::mem_pestat] mean and std.dev: (1129.11, 1134.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6642)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (257, 535, 744)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1718)\n",
            "[M::mem_pestat] mean and std.dev: (518.36, 292.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2205)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (527, 1068, 2399)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6143)\n",
            "[M::mem_pestat] mean and std.dev: (1499.56, 1442.12)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8015)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70224 reads in 6.151 CPU sec, 6.175 real sec\n",
            "[M::process] read 69924 sequences (10000178 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (112, 31275, 4, 89)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (445, 1087, 2458)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6484)\n",
            "[M::mem_pestat] mean and std.dev: (1586.22, 1506.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8497)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (232, 497, 716)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1684)\n",
            "[M::mem_pestat] mean and std.dev: (491.55, 286.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2168)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (423, 885, 2480)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6594)\n",
            "[M::mem_pestat] mean and std.dev: (1629.93, 1599.66)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8651)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70572 reads in 4.603 CPU sec, 4.465 real sec\n",
            "[M::process] read 70266 sequences (10000122 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (115, 31003, 13, 94)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (495, 923, 3064)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8202)\n",
            "[M::mem_pestat] mean and std.dev: (1896.51, 2054.79)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10771)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (271, 551, 758)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1732)\n",
            "[M::mem_pestat] mean and std.dev: (530.61, 292.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2219)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (136, 396, 1203)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3337)\n",
            "[M::mem_pestat] mean and std.dev: (669.08, 751.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4404)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (497, 1155, 2578)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6740)\n",
            "[M::mem_pestat] mean and std.dev: (1518.62, 1343.50)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8821)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69924 reads in 4.883 CPU sec, 4.751 real sec\n",
            "[M::process] read 70084 sequences (10000094 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (118, 31167, 11, 113)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (486, 994, 1862)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4614)\n",
            "[M::mem_pestat] mean and std.dev: (1169.27, 1037.78)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5990)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 516, 732)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1702)\n",
            "[M::mem_pestat] mean and std.dev: (506.52, 289.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2187)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (83, 415, 646)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1772)\n",
            "[M::mem_pestat] mean and std.dev: (395.60, 430.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2335)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (509, 778, 2451)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6335)\n",
            "[M::mem_pestat] mean and std.dev: (1472.01, 1489.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8277)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70266 reads in 5.798 CPU sec, 5.880 real sec\n",
            "[M::process] read 69722 sequences (10000156 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (148, 31046, 9, 126)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (403, 870, 2500)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6694)\n",
            "[M::mem_pestat] mean and std.dev: (1372.47, 1476.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8791)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (279, 575, 775)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1767)\n",
            "[M::mem_pestat] mean and std.dev: (544.99, 300.12)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2263)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (409, 879, 1983)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5131)\n",
            "[M::mem_pestat] mean and std.dev: (1230.36, 1142.00)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6705)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70084 reads in 4.701 CPU sec, 4.566 real sec\n",
            "[M::process] read 69888 sequences (10000064 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (107, 30934, 7, 141)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (538, 989, 2398)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6118)\n",
            "[M::mem_pestat] mean and std.dev: (1505.50, 1458.00)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7978)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (335, 629, 816)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1778)\n",
            "[M::mem_pestat] mean and std.dev: (586.55, 303.79)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2259)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (547, 1089, 3415)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9151)\n",
            "[M::mem_pestat] mean and std.dev: (2048.62, 2120.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12019)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69722 reads in 5.320 CPU sec, 5.210 real sec\n",
            "[M::process] read 69764 sequences (10000068 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (130, 30993, 13, 138)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (476, 1086, 2803)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7457)\n",
            "[M::mem_pestat] mean and std.dev: (1909.84, 1989.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9869)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (327, 628, 821)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1809)\n",
            "[M::mem_pestat] mean and std.dev: (588.27, 310.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2303)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (119, 888, 2860)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8342)\n",
            "[M::mem_pestat] mean and std.dev: (1745.85, 2250.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11083)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (498, 821, 2148)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5448)\n",
            "[M::mem_pestat] mean and std.dev: (1182.45, 1051.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7098)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69888 reads in 5.456 CPU sec, 5.471 real sec\n",
            "[M::process] read 69866 sequences (10000102 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (135, 31012, 8, 125)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (484, 1017, 2481)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6475)\n",
            "[M::mem_pestat] mean and std.dev: (1530.82, 1519.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8472)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (334, 642, 829)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1819)\n",
            "[M::mem_pestat] mean and std.dev: (595.71, 311.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2314)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (514, 1004, 2694)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7054)\n",
            "[M::mem_pestat] mean and std.dev: (1527.78, 1566.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9234)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69764 reads in 4.639 CPU sec, 4.525 real sec\n",
            "[M::process] read 70002 sequences (10000152 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (148, 30925, 6, 127)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (431, 989, 2034)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5240)\n",
            "[M::mem_pestat] mean and std.dev: (1198.68, 1059.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6843)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 626, 819)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1817)\n",
            "[M::mem_pestat] mean and std.dev: (584.40, 308.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2316)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (475, 832, 2326)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6028)\n",
            "[M::mem_pestat] mean and std.dev: (1368.49, 1399.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7879)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69866 reads in 5.817 CPU sec, 5.775 real sec\n",
            "[M::process] read 70166 sequences (10000008 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (132, 31067, 18, 141)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (485, 925, 2367)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6131)\n",
            "[M::mem_pestat] mean and std.dev: (1335.79, 1350.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8013)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (314, 620, 809)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1799)\n",
            "[M::mem_pestat] mean and std.dev: (577.26, 306.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2294)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (66, 442, 1023)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2937)\n",
            "[M::mem_pestat] mean and std.dev: (590.50, 759.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3894)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (454, 861, 2178)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5626)\n",
            "[M::mem_pestat] mean and std.dev: (1256.44, 1167.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7350)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70002 reads in 4.896 CPU sec, 4.957 real sec\n",
            "[M::process] read 70020 sequences (10000114 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (104, 31232, 9, 125)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (520, 1083, 2660)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6940)\n",
            "[M::mem_pestat] mean and std.dev: (1575.17, 1504.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9080)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (286, 594, 793)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1807)\n",
            "[M::mem_pestat] mean and std.dev: (557.98, 304.66)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2314)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (446, 806, 2026)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5186)\n",
            "[M::mem_pestat] mean and std.dev: (1206.64, 1222.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6766)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70166 reads in 4.848 CPU sec, 4.703 real sec\n",
            "[M::process] read 69660 sequences (10000232 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (117, 31067, 12, 125)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (510, 1067, 2810)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7410)\n",
            "[M::mem_pestat] mean and std.dev: (1789.89, 1833.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9710)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (302, 600, 790)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1766)\n",
            "[M::mem_pestat] mean and std.dev: (561.27, 300.17)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2254)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 375, 4300)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12640)\n",
            "[M::mem_pestat] mean and std.dev: (1752.08, 2109.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16810)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (527, 1206, 2631)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6839)\n",
            "[M::mem_pestat] mean and std.dev: (1715.03, 1690.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8943)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70020 reads in 6.133 CPU sec, 6.127 real sec\n",
            "[M::process] read 69770 sequences (10000132 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (134, 30823, 10, 123)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (505, 1052, 2792)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7366)\n",
            "[M::mem_pestat] mean and std.dev: (1807.79, 1791.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9653)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (329, 632, 821)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1805)\n",
            "[M::mem_pestat] mean and std.dev: (589.71, 307.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2297)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (739, 1585, 4157)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10993)\n",
            "[M::mem_pestat] mean and std.dev: (2563.00, 2482.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14411)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (469, 817, 2575)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6787)\n",
            "[M::mem_pestat] mean and std.dev: (1495.74, 1606.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8893)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69660 reads in 4.684 CPU sec, 4.541 real sec\n",
            "[M::process] read 69832 sequences (10000018 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (136, 31098, 7, 132)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (474, 966, 2585)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6807)\n",
            "[M::mem_pestat] mean and std.dev: (1602.96, 1531.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8918)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (336, 639, 828)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1812)\n",
            "[M::mem_pestat] mean and std.dev: (594.94, 310.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2304)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (416, 751, 2189)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5735)\n",
            "[M::mem_pestat] mean and std.dev: (1288.80, 1286.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7508)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69770 reads in 4.706 CPU sec, 4.559 real sec\n",
            "[M::process] read 69942 sequences (10000174 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (139, 31011, 11, 116)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (563, 866, 2183)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5423)\n",
            "[M::mem_pestat] mean and std.dev: (1275.74, 1182.40)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7043)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (331, 629, 819)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1795)\n",
            "[M::mem_pestat] mean and std.dev: (588.28, 309.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2283)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (65, 223, 1137)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3281)\n",
            "[M::mem_pestat] mean and std.dev: (488.45, 510.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4353)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (540, 1058, 2414)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6162)\n",
            "[M::mem_pestat] mean and std.dev: (1468.50, 1323.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8036)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69832 reads in 6.226 CPU sec, 6.211 real sec\n",
            "[M::process] read 70126 sequences (10000150 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (117, 31131, 9, 116)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (508, 1057, 2594)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6766)\n",
            "[M::mem_pestat] mean and std.dev: (1519.62, 1496.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8852)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (321, 629, 819)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1815)\n",
            "[M::mem_pestat] mean and std.dev: (584.96, 309.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2313)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (507, 904, 3134)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8388)\n",
            "[M::mem_pestat] mean and std.dev: (1882.86, 1918.78)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11015)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69942 reads in 4.710 CPU sec, 4.570 real sec\n",
            "[M::process] read 69976 sequences (10000122 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (119, 31077, 12, 119)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (566, 882, 2213)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5507)\n",
            "[M::mem_pestat] mean and std.dev: (1397.09, 1323.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7154)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (301, 616, 808)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1822)\n",
            "[M::mem_pestat] mean and std.dev: (573.14, 307.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2329)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (90, 254, 3073)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9039)\n",
            "[M::mem_pestat] mean and std.dev: (1022.58, 1523.02)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12022)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (431, 978, 2864)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7730)\n",
            "[M::mem_pestat] mean and std.dev: (1649.27, 1785.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10163)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70126 reads in 5.183 CPU sec, 5.091 real sec\n",
            "[M::process] read 69554 sequences (10000214 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (143, 31026, 8, 105)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (421, 788, 2350)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6208)\n",
            "[M::mem_pestat] mean and std.dev: (1276.80, 1301.09)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8137)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (307, 608, 802)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1792)\n",
            "[M::mem_pestat] mean and std.dev: (569.03, 304.02)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2287)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (464, 965, 2123)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5441)\n",
            "[M::mem_pestat] mean and std.dev: (1228.24, 1119.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7100)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69976 reads in 5.861 CPU sec, 5.973 real sec\n",
            "[M::process] read 69548 sequences (10000258 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (139, 30845, 9, 134)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (436, 808, 2114)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5470)\n",
            "[M::mem_pestat] mean and std.dev: (1123.18, 1053.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7148)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (353, 648, 839)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1811)\n",
            "[M::mem_pestat] mean and std.dev: (607.69, 314.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2297)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (504, 1035, 2883)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7641)\n",
            "[M::mem_pestat] mean and std.dev: (1704.36, 1673.00)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10020)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69554 reads in 4.690 CPU sec, 4.545 real sec\n",
            "[M::process] read 69338 sequences (10000090 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (122, 30907, 7, 144)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (532, 1021, 2586)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6694)\n",
            "[M::mem_pestat] mean and std.dev: (1575.80, 1515.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8748)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (362, 666, 855)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1841)\n",
            "[M::mem_pestat] mean and std.dev: (619.94, 318.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2334)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (521, 1065, 2281)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5801)\n",
            "[M::mem_pestat] mean and std.dev: (1405.37, 1329.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7561)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69548 reads in 5.630 CPU sec, 5.549 real sec\n",
            "[M::process] read 69562 sequences (10000190 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (153, 30654, 9, 151)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (539, 965, 2258)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5696)\n",
            "[M::mem_pestat] mean and std.dev: (1380.62, 1281.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7415)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (376, 667, 860)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1828)\n",
            "[M::mem_pestat] mean and std.dev: (626.67, 319.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2312)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (502, 945, 2798)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7390)\n",
            "[M::mem_pestat] mean and std.dev: (1617.69, 1597.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9686)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69338 reads in 5.227 CPU sec, 5.276 real sec\n",
            "[M::process] read 69710 sequences (10000188 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (158, 30782, 11, 128)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (478, 1116, 2546)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6682)\n",
            "[M::mem_pestat] mean and std.dev: (1654.39, 1657.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8750)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (348, 655, 848)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1848)\n",
            "[M::mem_pestat] mean and std.dev: (612.84, 320.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2348)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (269, 350, 773)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1781)\n",
            "[M::mem_pestat] mean and std.dev: (441.90, 453.07)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2285)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (548, 1383, 2845)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7439)\n",
            "[M::mem_pestat] mean and std.dev: (1765.45, 1680.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9736)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69562 reads in 4.886 CPU sec, 4.750 real sec\n",
            "[M::process] read 69786 sequences (10000012 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (140, 30936, 14, 146)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (532, 969, 3054)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8098)\n",
            "[M::mem_pestat] mean and std.dev: (1933.06, 1967.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10620)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (333, 642, 838)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1848)\n",
            "[M::mem_pestat] mean and std.dev: (601.27, 317.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2353)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (142, 422, 2338)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6730)\n",
            "[M::mem_pestat] mean and std.dev: (1042.93, 1323.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8926)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (536, 1133, 2401)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6131)\n",
            "[M::mem_pestat] mean and std.dev: (1521.45, 1400.59)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7996)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69710 reads in 6.225 CPU sec, 6.197 real sec\n",
            "[M::process] read 70080 sequences (10000038 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (128, 30994, 9, 116)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (659, 1186, 2418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5936)\n",
            "[M::mem_pestat] mean and std.dev: (1555.34, 1333.67)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7695)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (337, 640, 832)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1822)\n",
            "[M::mem_pestat] mean and std.dev: (597.29, 313.87)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2317)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (495, 1041, 2085)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5265)\n",
            "[M::mem_pestat] mean and std.dev: (1307.54, 1183.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6855)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69786 reads in 4.826 CPU sec, 4.780 real sec\n",
            "[M::process] read 69714 sequences (10000022 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (119, 31072, 12, 128)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (558, 1175, 2502)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6390)\n",
            "[M::mem_pestat] mean and std.dev: (1426.51, 1327.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8334)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (307, 616, 813)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1825)\n",
            "[M::mem_pestat] mean and std.dev: (577.06, 310.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2331)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (214, 610, 2301)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6475)\n",
            "[M::mem_pestat] mean and std.dev: (1188.33, 1315.87)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8562)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (558, 1207, 3161)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8367)\n",
            "[M::mem_pestat] mean and std.dev: (1925.90, 1879.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10970)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70080 reads in 4.754 CPU sec, 4.612 real sec\n",
            "[M::process] read 69426 sequences (10000050 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (131, 30914, 6, 122)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (555, 1091, 2257)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5661)\n",
            "[M::mem_pestat] mean and std.dev: (1308.57, 1139.08)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7363)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (335, 632, 821)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1793)\n",
            "[M::mem_pestat] mean and std.dev: (591.16, 308.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2279)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (442, 1000, 2692)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7192)\n",
            "[M::mem_pestat] mean and std.dev: (1622.56, 1697.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9442)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69714 reads in 6.492 CPU sec, 6.456 real sec\n",
            "[M::process] read 69474 sequences (10000180 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (139, 30768, 10, 142)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (533, 984, 2601)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6737)\n",
            "[M::mem_pestat] mean and std.dev: (1607.86, 1597.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8805)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (370, 662, 846)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1798)\n",
            "[M::mem_pestat] mean and std.dev: (617.17, 312.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2274)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (59, 543, 1198)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3476)\n",
            "[M::mem_pestat] mean and std.dev: (443.25, 463.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4615)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (612, 966, 2354)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5838)\n",
            "[M::mem_pestat] mean and std.dev: (1471.27, 1351.91)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7580)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69426 reads in 4.821 CPU sec, 4.703 real sec\n",
            "[M::process] read 69492 sequences (10000108 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (126, 30831, 16, 126)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (458, 949, 2730)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7274)\n",
            "[M::mem_pestat] mean and std.dev: (1638.17, 1645.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9546)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (372, 661, 854)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1818)\n",
            "[M::mem_pestat] mean and std.dev: (619.26, 315.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2300)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (192, 532, 1801)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5019)\n",
            "[M::mem_pestat] mean and std.dev: (780.60, 912.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6628)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (512, 934, 2375)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6101)\n",
            "[M::mem_pestat] mean and std.dev: (1317.27, 1269.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7964)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69474 reads in 5.141 CPU sec, 5.017 real sec\n",
            "[M::process] read 69776 sequences (10000286 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (123, 30911, 11, 141)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (481, 925, 2063)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5227)\n",
            "[M::mem_pestat] mean and std.dev: (1207.97, 1038.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6809)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (365, 662, 857)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1841)\n",
            "[M::mem_pestat] mean and std.dev: (619.56, 318.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2333)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (303, 745, 2085)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5649)\n",
            "[M::mem_pestat] mean and std.dev: (1395.00, 1542.50)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7565)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (452, 931, 2817)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7547)\n",
            "[M::mem_pestat] mean and std.dev: (1622.97, 1613.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9912)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69492 reads in 6.099 CPU sec, 6.217 real sec\n",
            "[M::process] read 69880 sequences (10000190 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (131, 30878, 10, 144)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (449, 999, 2981)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8045)\n",
            "[M::mem_pestat] mean and std.dev: (1855.64, 2054.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10577)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (332, 641, 835)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1841)\n",
            "[M::mem_pestat] mean and std.dev: (598.94, 316.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2344)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (187, 387, 4008)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11650)\n",
            "[M::mem_pestat] mean and std.dev: (2010.00, 2794.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15471)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (598, 1341, 3185)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8359)\n",
            "[M::mem_pestat] mean and std.dev: (1997.52, 1951.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10946)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69776 reads in 4.759 CPU sec, 4.616 real sec\n",
            "[M::process] read 70346 sequences (10000130 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (116, 30937, 5, 130)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (536, 1402, 3110)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8258)\n",
            "[M::mem_pestat] mean and std.dev: (2069.54, 2109.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10832)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (318, 629, 822)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1830)\n",
            "[M::mem_pestat] mean and std.dev: (587.41, 312.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2334)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (487, 1170, 2948)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7870)\n",
            "[M::mem_pestat] mean and std.dev: (1737.18, 1752.61)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10331)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 69880 reads in 5.615 CPU sec, 5.564 real sec\n",
            "[M::process] read 14272 sequences (2031434 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (126, 31186, 7, 109)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (500, 957, 2780)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7340)\n",
            "[M::mem_pestat] mean and std.dev: (1667.20, 1720.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9620)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (292, 604, 799)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1813)\n",
            "[M::mem_pestat] mean and std.dev: (564.67, 308.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2320)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (516, 1168, 2537)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6579)\n",
            "[M::mem_pestat] mean and std.dev: (1678.42, 1529.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8600)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 70346 reads in 5.599 CPU sec, 5.663 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (26, 6295, 1, 24)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (340, 628, 1682)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4366)\n",
            "[M::mem_pestat] mean and std.dev: (907.58, 861.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5708)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (260, 574, 777)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1811)\n",
            "[M::mem_pestat] mean and std.dev: (540.24, 299.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2328)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (707, 1111, 2191)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5159)\n",
            "[M::mem_pestat] mean and std.dev: (1469.26, 1197.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6643)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 14272 reads in 1.095 CPU sec, 1.028 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem ecoli.fna clean_1.fastq clean_2.fastq\n",
            "[main] Real time: 193.347 sec; CPU: 195.538 sec\n",
            "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "#Установка bwa и samtools\n",
        "!apt-get install -y bwa samtools\n",
        "#Индексация референсного генома\n",
        "!bwa index ecoli.fna\n",
        "#Выравнивание ридов dв файл aligned.sam\n",
        "!bwa mem ecoli.fna clean_1.fastq clean_2.fastq > aligned.sam\n",
        "#Преобразование SAM → BAM → сортировка\n",
        "!samtools view -S -b aligned.sam > aligned.bam\n",
        "!samtools sort aligned.bam -o aligned_sorted.bam\n",
        "!samtools index aligned_sorted.bam\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "S9iTYBB76EQH",
        "outputId": "3a08e961-5a58-4126-b135-096c80d6131b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "2623584 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "2602652 + 0 primary\n",
            "0 + 0 secondary\n",
            "20932 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "0 + 0 primary duplicates\n",
            "2475989 + 0 mapped (94.37% : N/A)\n",
            "2455057 + 0 primary mapped (94.33% : N/A)\n",
            "2602652 + 0 paired in sequencing\n",
            "1301326 + 0 read1\n",
            "1301326 + 0 read2\n",
            "2375808 + 0 properly paired (91.28% : N/A)\n",
            "2437740 + 0 with itself and mate mapped\n",
            "17317 + 0 singletons (0.67% : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ],
      "source": [
        "#Проверка. Покажет количество выровненных ридов, процент покрытия и др\n",
        "!samtools flagstat aligned_sorted.bam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "gXSunYbryZ6q"
      },
      "source": [
        "Очень высокий процент выравнивания (~94%) — отличное качество данных!\n",
        "\n",
        " Более 91% ридов выровнены в парах правильно — тоже отлично.\n",
        "\n",
        " Singletons — один рид из пары выровнен, другой — нет. У нас таких всего ~0.67%.\n",
        "\n",
        " Данные качественные:\n",
        "\n",
        "Почти все риды выровнены.\n",
        "\n",
        "Высокий процент правильно спаренных ридов.\n",
        "\n",
        "Нет дубликатов или химерных пар."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mutYALC4H9FD",
        "outputId": "af94383b-6cc0-45dc-9362-d4b15191c429"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SRR2584863.72312\t2147\tNC_000913.3\t1\t60\t116H34M\t=\t4641543\t4641652\tAGCTTTTCATTCTGACTGCAACGGGCAATATGTC\tDDDADDDDDEDDEEEDDDDDCDDBDB<?CCDDDD\tNM:i:0\tMD:Z:34\tMC:Z:110M40H\tAS:i:34\tXS:i:0\tSA:Z:NC_000913.3,4641537,+,116M34S,60,1;\n",
            "SRR2584863.236239\t99\tNC_000913.3\t1\t60\t55S95M\t=\t305\t454\tGCATGATATTGAAAAAAATATCACCAAATAAAAAACGCCTTAGTAAGTATTTTTCAGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGCTTCTGAACTGGTTACCTGCCGTGAG\t@BCFFFFFHHHHHJJJJJIIJJIJJJJIIJJJJJJIGJIJIIIAHGHCFGIJJJIIIHHHHHGHFFFFFBDECACCDDDB=BBCDDEDDEDCDCDDDB>@@>CDDDDBBDCDDACC@@CCCCC:<A@CCCD@@>C>CC@C>A4<8099?B\tNM:i:0\tMD:Z:95\tMC:Z:150M\tAS:i:95\tXS:i:0\tSA:Z:NC_000913.3,4641598,+,55M95S,60,0;\n",
            "SRR2584863.275624\t99\tNC_000913.3\t1\t60\t14S136M\t=\t145\t294\tAGTAAGTATTTTTCAGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGCTTCTGAACTGGTTACCTGCCGTGAGTAAATTAAAATTTTATTGACTTAGGTCACTAAATACTTTAA\t@B@FFFBDHHHHHJJIGJJJJJJJJJJJJIJJJJJJIJJJIJIGIGJJJJJJIJFGIHIJJJJJIJHFFFDFEEEEEEEDDDDDDDDDDEDDDDDCDDDDDCBDDBDDDCDDDEEDDDDEDDDEDDDDDDDDDDDDDDDDCDCDDDDEDD\tNM:i:0\tMD:Z:136\tMC:Z:150M\tAS:i:136\tXS:i:0\n",
            "SRR2584863.404450\t2209\tNC_000913.3\t1\t60\t91H59M\t=\t333\t482\tAGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGT\tDDDDDDCDDEDDEEEDDDDDDCDDDBDDDDCEEDDCCDDDDDDCDCDDCDDDDBDDCDC\tNM:i:0\tMD:Z:59\tMC:Z:150M\tAS:i:59\tXS:i:0\tSA:Z:NC_000913.3,4641562,+,91M59S,60,0;\n",
            "SRR2584863.596115\t2209\tNC_000913.3\t1\t60\t116H34M\t=\t514\t663\tAGCTTTTCATTCTGACTGCAACGGGCAATATGTC\tCDCCCCD<ACDCCDCCCDCCA3055@B8?CCABA\tNM:i:0\tMD:Z:34\tMC:Z:150M\tAS:i:34\tXS:i:0\tSA:Z:NC_000913.3,4641537,+,116M34S,60,1;\n",
            "SRR2584863.651714\t2145\tNC_000913.3\t1\t60\t95H55M\t=\t26\t175\tAGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGA\tDDDDDDACDEEEEEEDDDDDCDBBD@BDCCDEDDEDDDCD@BDDDDCCDCCB@B?\tNM:i:0\tMD:Z:55\tMC:Z:150M\tAS:i:55\tXS:i:0\tSA:Z:NC_000913.3,4641558,+,95M55S,60,0;\n",
            "SRR2584863.702752\t163\tNC_000913.3\t1\t60\t28S122M\t=\t654\t803\tATAAAAAACGCCTTAGTAAGTATTTTTCAGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGCTTCTGAACTGGTTACCTGCCGTGAGTAAATTAAAATTTTATTGACTTAGGTC\t<@BFFFFFHHHHHJIJIJJJHIIJJJJJIJJJJJJHHIIJJJJJJIGIJIJIIJJJJJJJHGHHHHHFDFFFEDECEDDDDBBCACDDDDDEDACDDDDDDDDEDDDCDDDDDDDDBDDDBDDDDDCDEDDDDDCDDDDDCDCDDDDCCD\tNM:i:0\tMD:Z:122\tMC:Z:150M\tAS:i:122\tXS:i:0\n",
            "SRR2584863.729928\t2145\tNC_000913.3\t1\t60\t119H31M\t=\t84\t233\tAGCTTTTCATTCTGACTGCAACGGGCAATAT\tDDDDCDDCDEEDDEEDDDCCDDD>@<BDC@C\tNM:i:0\tMD:Z:31\tMC:Z:150M\tAS:i:31\tXS:i:0\tSA:Z:NC_000913.3,4641534,+,119M31S,60,1;\n",
            "SRR2584863.1021044\t2209\tNC_000913.3\t1\t60\t77H73M\t=\t279\t429\tAGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGCTTC\tDDBDCDDDDEFEEEEDCCCCDDDDD@B<CCDDDEDDCDCDCDDCCDDDDDDDDDBDCDDDDDDEDCCCCDDCD\tNM:i:0\tMD:Z:73\tMC:Z:23M1D127M\tAS:i:73\tXS:i:0\tSA:Z:NC_000913.3,4641576,+,77M73S,60,0;\n",
            "SRR2584863.1040257\t99\tNC_000913.3\t1\t60\t32S118M\t=\t674\t823\tCCAAATAAAAAACGCCTTAGTAAGTATTTTTCAGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGCTTCTGAACTGGTTACCTGCCGTGAGTAAATTAAAATTTTATTGACTTA\tBC@FFFFFHGHHHJJJJJJJGHIJ?GIIIIJJJJJJJJJJJJDIJJJJIIIGHIGHIGIIHFHHEEHDFF;B?ADDEECDCDDDD@?C>@CCDDCCDDDC>?CCCDCDDDD>C@CCDDDCDBB@BBDCDDC@DAACDD:>C:>:A@C>C@\tNM:i:0\tMD:Z:118\tMC:Z:150M\tAS:i:118\tXS:i:0\tSA:Z:NC_000913.3,4641621,+,32M118S,60,0;\n"
          ]
        }
      ],
      "source": [
        "!samtools view aligned_sorted.bam | head"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "ojpPLsJSKgLY"
      },
      "outputs": [],
      "source": [
        "# Более подробная статистика (глубина, GC-контент, распределение длин и т.д.).\n",
        "!samtools stats aligned_sorted.bam > alignment_stats.txt\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MfYdmgbrK4kh",
        "outputId": "82cfd1b6-8d65-49a1-a371-05c548d2deb8"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "NC_000913.3\t4641652\t2475989\t17317\n",
            "*\t0\t0\t130278\n"
          ]
        }
      ],
      "source": [
        "#Cтатистика по каждому контигу (или хромосоме): число выровненных и невыравненных ридов.\n",
        "!samtools idxstats aligned_sorted.bam\n"
      ]
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "gpuType": "T4",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
