{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "mOAiPuMecwfT",
        "outputId": "2ad31f26-38da-41f5-ff15-8a4b31ffcbf8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "bin/micromamba\n",
            "bioconda/linux-64                                           Using cache\n",
            "bioconda/noarch                                             Using cache\n",
            "conda-forge/linux-64                                        Using cache\n",
            "conda-forge/noarch                                          Using cache\n",
            "\u001b[?25l\u001b[2K\u001b[0G\u001b[?25h\u001b[?25l\u001b[2K\u001b[0G\u001b[?25h\n",
            "\n",
            "Transaction\n",
            "\n",
            "  Prefix: /root/.local/share/mamba/envs/bio\n",
            "\n",
            "  Updating specs:\n",
            "\n",
            "   - fastqc\n",
            "   - trimmomatic\n",
            "   - bwa\n",
            "   - samtools\n",
            "   - bcftools\n",
            "   - fastp\n",
            "   - hisat2\n",
            "   - bowtie2\n",
            "   - bedtools\n",
            "   - seqtk\n",
            "   - multiqc\n",
            "\n",
            "\n",
            "  Package                           Version  Build                 Channel           Size\n",
            "───────────────────────────────────────────────────────────────────────────────────────────\n",
            "  Install:\n",
            "───────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  \u001b[32m+ _libgcc_mutex            \u001b[0m           0.1  conda_forge           conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ _openmp_mutex            \u001b[0m           4.5  2_gnu                 conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ alsa-lib                 \u001b[0m        1.2.14  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ annotated-types          \u001b[0m         0.7.0  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ attrs                    \u001b[0m        25.3.0  pyh71513ae_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ bcftools                 \u001b[0m          1.21  h3a4d415_1            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ bedtools                 \u001b[0m        2.31.1  h13024bc_3            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ bowtie2                  \u001b[0m         2.5.4  he96a11b_5            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ brotli-python            \u001b[0m         1.1.0  py313h46c70d0_2       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ bwa                      \u001b[0m        0.7.19  h577a1d6_0            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ bzip2                    \u001b[0m         1.0.8  h4bc722e_7            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ c-ares                   \u001b[0m        1.34.5  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ ca-certificates          \u001b[0m     2025.4.26  hbd8a1cb_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ cairo                    \u001b[0m        1.18.4  h3394656_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ certifi                  \u001b[0m     2025.4.26  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ cffi                     \u001b[0m        1.17.1  py313hfab6e84_0       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ charset-normalizer       \u001b[0m         3.4.2  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ click                    \u001b[0m         8.1.8  pyh707e725_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ colorama                 \u001b[0m         0.4.6  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ coloredlogs              \u001b[0m        15.0.1  pyhd8ed1ab_4          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ colormath                \u001b[0m         3.0.0  pyhd8ed1ab_4          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ expat                    \u001b[0m         2.7.0  h5888daf_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ fastp                    \u001b[0m        0.24.1  heae3180_0            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ fastqc                   \u001b[0m        0.12.1  hdfd78af_0            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ font-ttf-dejavu-sans-mono\u001b[0m          2.37  hab24e00_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ font-ttf-inconsolata     \u001b[0m         3.000  h77eed37_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ font-ttf-source-code-pro \u001b[0m         2.038  h77eed37_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ font-ttf-ubuntu          \u001b[0m          0.83  h77eed37_3            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ fontconfig               \u001b[0m        2.15.0  h7e30c49_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ fonts-conda-ecosystem    \u001b[0m             1  0                     conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ fonts-conda-forge        \u001b[0m             1  0                     conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ freetype                 \u001b[0m        2.13.3  ha770c72_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ giflib                   \u001b[0m         5.2.2  hd590300_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ graphite2                \u001b[0m        1.3.13  h59595ed_1003         conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ gsl                      \u001b[0m           2.7  he838d99_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ h2                       \u001b[0m         4.2.0  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ harfbuzz                 \u001b[0m        11.1.0  h3beb420_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ hisat2                   \u001b[0m         2.2.1  h503566f_8            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ hpack                    \u001b[0m         4.1.0  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ htslib                   \u001b[0m          1.21  h566b1c6_1            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ humanfriendly            \u001b[0m          10.0  pyh707e725_8          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ humanize                 \u001b[0m        4.12.3  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ hyperframe               \u001b[0m         6.1.0  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ icu                      \u001b[0m          75.1  he02047a_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ idna                     \u001b[0m          3.10  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ importlib-metadata       \u001b[0m         8.6.1  pyha770c72_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ importlib_resources      \u001b[0m         6.5.2  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ isa-l                    \u001b[0m        2.31.1  hb9d3cd8_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ jinja2                   \u001b[0m         3.1.6  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ jsonschema               \u001b[0m        4.23.0  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ jsonschema-specifications\u001b[0m      2025.4.1  pyh29332c3_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ kaleido-core             \u001b[0m         0.2.1  h3644ca4_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ keyutils                 \u001b[0m         1.6.1  h166bdaf_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ krb5                     \u001b[0m        1.21.3  h659f571_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ lcms2                    \u001b[0m          2.17  h717163a_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ ld_impl_linux-64         \u001b[0m          2.43  h712a8e2_4            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ lerc                     \u001b[0m         4.0.0  h0aef613_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libblas                  \u001b[0m         3.9.0  31_h59b9bed_openblas  conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libcblas                 \u001b[0m         3.9.0  31_he106b2a_openblas  conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libcups                  \u001b[0m         2.3.3  h4637d8d_4            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libcurl                  \u001b[0m        8.13.0  h332b0f4_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libdeflate               \u001b[0m          1.22  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libedit                  \u001b[0m  3.1.20250104  pl5321h7949ede_0      conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libev                    \u001b[0m          4.33  hd590300_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libexpat                 \u001b[0m         2.7.0  h5888daf_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libffi                   \u001b[0m         3.4.6  h2dba641_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libfreetype              \u001b[0m        2.13.3  ha770c72_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libfreetype6             \u001b[0m        2.13.3  h48d6fc4_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libgcc                   \u001b[0m        15.1.0  h767d61c_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libgcc-ng                \u001b[0m        15.1.0  h69a702a_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libgfortran              \u001b[0m        15.1.0  h69a702a_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libgfortran5             \u001b[0m        15.1.0  hcea5267_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libglib                  \u001b[0m        2.84.1  h3618099_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libgomp                  \u001b[0m        15.1.0  h767d61c_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libiconv                 \u001b[0m          1.18  h4ce23a2_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libjpeg-turbo            \u001b[0m         3.1.0  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ liblapack                \u001b[0m         3.9.0  31_h7ac8fdf_openblas  conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ liblzma                  \u001b[0m         5.8.1  hb9d3cd8_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libmpdec                 \u001b[0m         4.0.0  h4bc722e_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libnghttp2               \u001b[0m        1.64.0  h161d5f1_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libopenblas              \u001b[0m        0.3.29  pthreads_h94d23a6_0   conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libpng                   \u001b[0m        1.6.47  h943b412_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libsqlite                \u001b[0m        3.49.2  hee588c1_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libssh2                  \u001b[0m        1.11.1  hcf80075_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libstdcxx                \u001b[0m        15.1.0  h8f9b012_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libstdcxx-ng             \u001b[0m        15.1.0  h4852527_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libtiff                  \u001b[0m         4.7.0  hc4654cb_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libuuid                  \u001b[0m        2.38.1  h0b41bf4_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libwebp-base             \u001b[0m         1.5.0  h851e524_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libxcb                   \u001b[0m        1.17.0  h8a09558_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libxcrypt                \u001b[0m        4.4.36  hd590300_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ libzlib                  \u001b[0m         1.3.1  hb9d3cd8_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ markdown                 \u001b[0m           3.8  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ markdown-it-py           \u001b[0m         3.0.0  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ markupsafe               \u001b[0m         3.0.2  py313h8060acc_1       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ mathjax                  \u001b[0m         2.7.7  ha770c72_3            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ mdurl                    \u001b[0m         0.1.2  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ multiqc                  \u001b[0m          1.28  pyhdfd78af_0          bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ narwhals                 \u001b[0m        1.38.2  pyhe01879c_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ natsort                  \u001b[0m         8.4.0  pyh29332c3_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ ncurses                  \u001b[0m           6.5  h2d0b736_3            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ networkx                 \u001b[0m         3.4.2  pyh267e887_2          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ nspr                     \u001b[0m          4.36  h5888daf_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ nss                      \u001b[0m         3.111  h159eef7_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ numpy                    \u001b[0m         2.2.5  py313h17eae1a_0       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ openjdk                  \u001b[0m        23.0.2  h53dfc1b_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ openjpeg                 \u001b[0m         2.5.3  h5fbd93e_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ openssl                  \u001b[0m         3.5.0  h7b32b05_1            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ packaging                \u001b[0m          25.0  pyh29332c3_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pcre2                    \u001b[0m         10.45  hc749103_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ perl                     \u001b[0m        5.32.1  7_hd590300_perl5      conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pillow                   \u001b[0m        11.2.1  py313h8db990d_0       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pip                      \u001b[0m        25.1.1  pyh145f28c_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pixman                   \u001b[0m        0.46.0  h29eaf8c_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pkgutil-resolve-name     \u001b[0m        1.3.10  pyhd8ed1ab_2          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ plotly                   \u001b[0m         6.0.1  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pthread-stubs            \u001b[0m           0.4  hb9d3cd8_1002         conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pyaml-env                \u001b[0m         1.2.2  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pycparser                \u001b[0m          2.22  pyh29332c3_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pydantic                 \u001b[0m        2.11.4  pyh3cfb1c2_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pydantic-core            \u001b[0m        2.33.2  py313h4b2b08d_0       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pygments                 \u001b[0m        2.19.1  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pysocks                  \u001b[0m         1.7.1  pyha55dd90_7          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ python                   \u001b[0m        3.13.3  hf636f53_101_cp313    conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ python-dotenv            \u001b[0m         1.1.0  pyh29332c3_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ python-kaleido           \u001b[0m         0.2.1  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ python_abi               \u001b[0m          3.13  7_cp313               conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ pyyaml                   \u001b[0m         6.0.2  py313h8060acc_2       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ readline                 \u001b[0m           8.2  h8c095d6_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ referencing              \u001b[0m        0.36.2  pyh29332c3_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ regex                    \u001b[0m     2024.11.6  py313h536fd9c_0       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ requests                 \u001b[0m        2.32.3  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ rich                     \u001b[0m        14.0.0  pyh29332c3_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ rich-click               \u001b[0m         1.8.8  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ rpds-py                  \u001b[0m        0.24.0  py313h6071e0b_0       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ samtools                 \u001b[0m          1.21  h96c455f_1            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ seqtk                    \u001b[0m           1.4  h577a1d6_3            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ spectra                  \u001b[0m        0.0.11  pyhd8ed1ab_2          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ sqlite                   \u001b[0m        3.49.2  h9eae976_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ tiktoken                 \u001b[0m         0.9.0  py313h3f234b3_0       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ tk                       \u001b[0m        8.6.13  noxft_h4845f30_101    conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ tqdm                     \u001b[0m        4.67.1  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ trimmomatic              \u001b[0m          0.39  hdfd78af_2            bioconda   \u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ typeguard                \u001b[0m         4.4.2  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ typing-extensions        \u001b[0m        4.13.2  h0e9735f_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ typing-inspection        \u001b[0m         0.4.0  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ typing_extensions        \u001b[0m        4.13.2  pyh29332c3_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ tzdata                   \u001b[0m         2025b  h78e105d_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ urllib3                  \u001b[0m         2.4.0  pyhd8ed1ab_0          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libice              \u001b[0m         1.1.2  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libsm               \u001b[0m         1.2.6  he73a12e_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libx11              \u001b[0m        1.8.12  h4f16b4b_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxau              \u001b[0m        1.0.12  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxdmcp            \u001b[0m         1.1.5  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxext             \u001b[0m         1.3.6  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxfixes           \u001b[0m         6.0.1  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxi               \u001b[0m         1.8.2  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxrandr           \u001b[0m         1.5.4  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxrender          \u001b[0m        0.9.12  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxt               \u001b[0m         1.3.1  hb9d3cd8_0            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ xorg-libxtst             \u001b[0m         1.2.5  hb9d3cd8_3            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ yaml                     \u001b[0m         0.2.5  h7f98852_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ zipp                     \u001b[0m        3.21.0  pyhd8ed1ab_1          conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ zlib                     \u001b[0m         1.3.1  hb9d3cd8_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ zstandard                \u001b[0m        0.23.0  py313h536fd9c_2       conda-forge\u001b[32m     Cached\u001b[0m\n",
            "  \u001b[32m+ zstd                     \u001b[0m         1.5.7  hb8e6e7a_2            conda-forge\u001b[32m     Cached\u001b[0m\n",
            "\n",
            "  Summary:\n",
            "\n",
            "  Install: 166 packages\n",
            "\n",
            "  Total download: 0 B\n",
            "\n",
            "───────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "\n",
            "\n",
            "Transaction starting\n",
            "\u001b[?25l\u001b[2K\u001b[0G\u001b[?25hLinking font-ttf-dejavu-sans-mono-2.37-hab24e00_0\n",
            "Linking font-ttf-ubuntu-0.83-h77eed37_3\n",
            "Linking font-ttf-inconsolata-3.000-h77eed37_0\n",
            "Linking font-ttf-source-code-pro-2.038-h77eed37_0\n",
            "Linking python_abi-3.13-7_cp313\n",
            "Linking tzdata-2025b-h78e105d_0\n",
            "Linking ca-certificates-2025.4.26-hbd8a1cb_0\n",
            "Linking fonts-conda-forge-1-0\n",
            "Linking fonts-conda-ecosystem-1-0\n",
            "Linking libgomp-15.1.0-h767d61c_2\n",
            "Linking mathjax-2.7.7-ha770c72_3\n",
            "Linking ld_impl_linux-64-2.43-h712a8e2_4\n",
            "Linking _libgcc_mutex-0.1-conda_forge\n",
            "Linking _openmp_mutex-4.5-2_gnu\n",
            "Linking libgcc-15.1.0-h767d61c_2\n",
            "Linking libiconv-1.18-h4ce23a2_1\n",
            "Linking c-ares-1.34.5-hb9d3cd8_0\n",
            "Linking libgfortran5-15.1.0-hcea5267_2\n",
            "Linking xorg-libxdmcp-1.1.5-hb9d3cd8_0\n",
            "Linking xorg-libxau-1.0.12-hb9d3cd8_0\n",
            "Linking pthread-stubs-0.4-hb9d3cd8_1002\n",
            "Linking alsa-lib-1.2.14-hb9d3cd8_0\n",
            "Linking xorg-libice-1.1.2-hb9d3cd8_0\n",
            "Linking libwebp-base-1.5.0-h851e524_0\n",
            "Linking libjpeg-turbo-3.1.0-hb9d3cd8_0\n",
            "Linking openssl-3.5.0-h7b32b05_1\n",
            "Linking liblzma-5.8.1-hb9d3cd8_1\n",
            "Linking libffi-3.4.6-h2dba641_1\n",
            "Linking ncurses-6.5-h2d0b736_3\n",
            "Linking libzlib-1.3.1-hb9d3cd8_2\n",
            "Linking libstdcxx-15.1.0-h8f9b012_2\n",
            "Linking libgcc-ng-15.1.0-h69a702a_2\n",
            "Linking libexpat-2.7.0-h5888daf_0\n",
            "Linking libdeflate-1.22-hb9d3cd8_0\n",
            "Linking isa-l-2.31.1-hb9d3cd8_1\n",
            "Linking libgfortran-15.1.0-h69a702a_2\n",
            "Linking libxcb-1.17.0-h8a09558_0\n",
            "Linking libedit-3.1.20250104-pl5321h7949ede_0\n",
            "Linking readline-8.2-h8c095d6_2\n",
            "Linking libssh2-1.11.1-hcf80075_0\n",
            "Linking libpng-1.6.47-h943b412_0\n",
            "Linking zlib-1.3.1-hb9d3cd8_2\n",
            "Linking libsqlite-3.49.2-hee588c1_0\n",
            "Linking pixman-0.46.0-h29eaf8c_0\n",
            "Linking libstdcxx-ng-15.1.0-h4852527_2\n",
            "Linking lerc-4.0.0-h0aef613_1\n",
            "Linking zstd-1.5.7-hb8e6e7a_2\n",
            "Linking nspr-4.36-h5888daf_0\n",
            "Linking libev-4.33-hd590300_2\n",
            "Linking libxcrypt-4.4.36-hd590300_1\n",
            "Linking yaml-0.2.5-h7f98852_2\n",
            "Linking keyutils-1.6.1-h166bdaf_0\n",
            "Linking giflib-5.2.2-hd590300_0\n",
            "Linking tk-8.6.13-noxft_h4845f30_101\n",
            "Linking bzip2-1.0.8-h4bc722e_7\n",
            "Linking libuuid-2.38.1-h0b41bf4_0\n",
            "Linking libmpdec-4.0.0-h4bc722e_0\n",
            "Linking expat-2.7.0-h5888daf_0\n",
            "Linking libopenblas-0.3.29-pthreads_h94d23a6_0\n",
            "Linking xorg-libx11-1.8.12-h4f16b4b_0\n",
            "Linking libfreetype6-2.13.3-h48d6fc4_1\n",
            "Linking sqlite-3.49.2-h9eae976_0\n",
            "Linking graphite2-1.3.13-h59595ed_1003\n",
            "Linking icu-75.1-he02047a_0\n",
            "Linking libtiff-4.7.0-hc4654cb_2\n",
            "Linking nss-3.111-h159eef7_0\n",
            "Linking libnghttp2-1.64.0-h161d5f1_0\n",
            "Linking perl-5.32.1-7_hd590300_perl5\n",
            "Linking krb5-1.21.3-h659f571_0\n",
            "Linking pcre2-10.45-hc749103_0\n",
            "Linking xorg-libsm-1.2.6-he73a12e_0\n",
            "Linking python-3.13.3-hf636f53_101_cp313\n",
            "Linking libblas-3.9.0-31_h59b9bed_openblas\n",
            "Linking xorg-libxfixes-6.0.1-hb9d3cd8_0\n",
            "Linking xorg-libxrender-0.9.12-hb9d3cd8_0\n",
            "Linking xorg-libxext-1.3.6-hb9d3cd8_0\n",
            "Linking libfreetype-2.13.3-ha770c72_1\n",
            "Linking openjpeg-2.5.3-h5fbd93e_0\n",
            "Linking lcms2-2.17-h717163a_0\n",
            "Linking libcups-2.3.3-h4637d8d_4\n",
            "Linking libcurl-8.13.0-h332b0f4_0\n",
            "Linking libglib-2.84.1-h3618099_1\n",
            "Linking xorg-libxt-1.3.1-hb9d3cd8_0\n",
            "Linking liblapack-3.9.0-31_h7ac8fdf_openblas\n",
            "Linking libcblas-3.9.0-31_he106b2a_openblas\n",
            "Linking xorg-libxrandr-1.5.4-hb9d3cd8_0\n",
            "Linking xorg-libxi-1.8.2-hb9d3cd8_0\n",
            "Linking freetype-2.13.3-ha770c72_1\n",
            "Linking gsl-2.7-he838d99_0\n",
            "Linking xorg-libxtst-1.2.5-hb9d3cd8_3\n",
            "Linking fontconfig-2.15.0-h7e30c49_1\n",
            "Linking cairo-1.18.4-h3394656_0\n",
            "Linking kaleido-core-0.2.1-h3644ca4_0\n",
            "Linking harfbuzz-11.1.0-h3beb420_0\n",
            "Linking openjdk-23.0.2-h53dfc1b_2\n",
            "Linking fastp-0.24.1-heae3180_0\n",
            "Linking seqtk-1.4-h577a1d6_3\n",
            "Linking bedtools-2.31.1-h13024bc_3\n",
            "Linking bwa-0.7.19-h577a1d6_0\n",
            "Linking htslib-1.21-h566b1c6_1\n",
            "Linking samtools-1.21-h96c455f_1\n",
            "Linking bcftools-1.21-h3a4d415_1\n",
            "Linking pip-25.1.1-pyh145f28c_0\n",
            "Linking pycparser-2.22-pyh29332c3_1\n",
            "Linking pysocks-1.7.1-pyha55dd90_7\n",
            "Linking hyperframe-6.1.0-pyhd8ed1ab_0\n",
            "Linking hpack-4.1.0-pyhd8ed1ab_0\n",
            "Linking humanfriendly-10.0-pyh707e725_8\n",
            "Linking networkx-3.4.2-pyh267e887_2\n",
            "Linking zipp-3.21.0-pyhd8ed1ab_1\n",
            "Linking pkgutil-resolve-name-1.3.10-pyhd8ed1ab_2\n",
            "Linking attrs-25.3.0-pyh71513ae_0\n",
            "Linking mdurl-0.1.2-pyhd8ed1ab_1\n",
            "Linking narwhals-1.38.2-pyhe01879c_0\n",
            "Linking charset-normalizer-3.4.2-pyhd8ed1ab_0\n",
            "Linking idna-3.10-pyhd8ed1ab_1\n",
            "Linking certifi-2025.4.26-pyhd8ed1ab_0\n",
            "Linking pygments-2.19.1-pyhd8ed1ab_0\n",
            "Linking colorama-0.4.6-pyhd8ed1ab_1\n",
            "Linking typing_extensions-4.13.2-pyh29332c3_0\n",
            "Linking python-dotenv-1.1.0-pyh29332c3_1\n",
            "Linking packaging-25.0-pyh29332c3_1\n",
            "Linking natsort-8.4.0-pyh29332c3_1\n",
            "Linking humanize-4.12.3-pyhd8ed1ab_0\n",
            "Linking click-8.1.8-pyh707e725_0\n",
            "Linking python-kaleido-0.2.1-pyhd8ed1ab_0\n",
            "Linking h2-4.2.0-pyhd8ed1ab_0\n",
            "Linking coloredlogs-15.0.1-pyhd8ed1ab_4\n",
            "Linking importlib_resources-6.5.2-pyhd8ed1ab_0\n",
            "Linking importlib-metadata-8.6.1-pyha770c72_0\n",
            "Linking markdown-it-py-3.0.0-pyhd8ed1ab_1\n",
            "Linking tqdm-4.67.1-pyhd8ed1ab_1\n",
            "Linking typing-extensions-4.13.2-h0e9735f_0\n",
            "Linking typing-inspection-0.4.0-pyhd8ed1ab_0\n",
            "Linking plotly-6.0.1-pyhd8ed1ab_0\n",
            "Linking markdown-3.8-pyhd8ed1ab_0\n",
            "Linking rich-14.0.0-pyh29332c3_0\n",
            "Linking annotated-types-0.7.0-pyhd8ed1ab_1\n",
            "Linking typeguard-4.4.2-pyhd8ed1ab_0\n",
            "Linking rich-click-1.8.8-pyhd8ed1ab_0\n",
            "Linking fastqc-0.12.1-hdfd78af_0\n",
            "Linking trimmomatic-0.39-hdfd78af_2\n",
            "Linking regex-2024.11.6-py313h536fd9c_0\n",
            "Linking markupsafe-3.0.2-py313h8060acc_1\n",
            "Linking rpds-py-0.24.0-py313h6071e0b_0\n",
            "Linking brotli-python-1.1.0-py313h46c70d0_2\n",
            "Linking pyyaml-6.0.2-py313h8060acc_2\n",
            "Linking pillow-11.2.1-py313h8db990d_0\n",
            "Linking numpy-2.2.5-py313h17eae1a_0\n",
            "Linking cffi-1.17.1-py313hfab6e84_0\n",
            "Linking pydantic-core-2.33.2-py313h4b2b08d_0\n",
            "Linking zstandard-0.23.0-py313h536fd9c_2\n",
            "Linking hisat2-2.2.1-h503566f_8\n",
            "Linking bowtie2-2.5.4-he96a11b_5\n",
            "Linking jinja2-3.1.6-pyhd8ed1ab_0\n",
            "Linking referencing-0.36.2-pyh29332c3_0\n",
            "Linking pyaml-env-1.2.2-pyhd8ed1ab_0\n",
            "Linking colormath-3.0.0-pyhd8ed1ab_4\n",
            "Linking pydantic-2.11.4-pyh3cfb1c2_0\n",
            "Linking urllib3-2.4.0-pyhd8ed1ab_0\n",
            "Linking jsonschema-specifications-2025.4.1-pyh29332c3_0\n",
            "Linking spectra-0.0.11-pyhd8ed1ab_2\n",
            "Linking requests-2.32.3-pyhd8ed1ab_1\n",
            "Linking jsonschema-4.23.0-pyhd8ed1ab_1\n",
            "Linking tiktoken-0.9.0-py313h3f234b3_0\n",
            "Linking multiqc-1.28-pyhdfd78af_0\n",
            "\n",
            "Transaction finished\n",
            "\n",
            "\n",
            "To activate this environment, use:\n",
            "\n",
            "    micromamba activate bio\n",
            "\n",
            "Or to execute a single command in this environment, use:\n",
            "\n",
            "    micromamba run -n bio mycommand\n",
            "\n",
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created. If --delete is \n",
            "                    also specified then the zip file will be removed after the \n",
            "                    contents are unzipped. \n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "\n",
            "    --dup_length    Sets a length to which the sequences will be truncated when \n",
            "                    defining them to be duplicates, affecting the duplication and\n",
            "                    overrepresented sequences plot.  This can be useful if you have\n",
            "                    long reads with higher levels of miscalls, or contamination with\n",
            "                    adapter dimers containing UMI sequences.\n",
            "\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "\n",
            "    --memory        Sets the base amount of memory, in Megabytes, used to process \n",
            "                    each file.  Defaults to 512MB.  You may need to increase this if\n",
            "                    you have a file with very long sequences in it.\n",
            "                \n",
            "    --svg           Save the graphs in the report in SVG format.\n",
            "\n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Suppress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n",
            "\n",
            "Program: bwa (alignment via Burrows-Wheeler transformation)\n",
            "Version: 0.7.19-r1273\n",
            "Contact: Heng Li <hli@ds.dfci.harvard.edu>\n",
            "\n",
            "Usage:   bwa <command> [options]\n",
            "\n",
            "Command: index         index sequences in the FASTA format\n",
            "         mem           BWA-MEM algorithm\n",
            "         fastmap       identify super-maximal exact matches\n",
            "         pemerge       merge overlapping paired ends (EXPERIMENTAL)\n",
            "         aln           gapped/ungapped alignment\n",
            "         samse         generate alignment (single ended)\n",
            "         sampe         generate alignment (paired ended)\n",
            "         bwasw         BWA-SW for long queries (DEPRECATED)\n",
            "\n",
            "         shm           manage indices in shared memory\n",
            "         fa2pac        convert FASTA to PAC format\n",
            "         pac2bwt       generate BWT from PAC\n",
            "         pac2bwtgen    alternative algorithm for generating BWT\n",
            "         bwtupdate     update .bwt to the new format\n",
            "         bwt2sa        generate SA from BWT and Occ\n",
            "\n",
            "Note: To use BWA, you need to first index the genome with `bwa index'.\n",
            "      There are three alignment algorithms in BWA: `mem', `bwasw', and\n",
            "      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'\n",
            "      first. Please `man ./bwa.1' for the manual.\n",
            "\n",
            "samtools 1.21\n",
            "Using htslib 1.21\n",
            "Copyright (C) 2024 Genome Research Ltd.\n",
            "\n",
            "Samtools compilation details:\n",
            "    Features:       build=configure curses=yes \n",
            "    CC:             /opt/conda/conda-bld/samtools_1734249775320/_build_env/bin/x86_64-conda-linux-gnu-cc\n",
            "    CPPFLAGS:       -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /root/.local/share/mamba/envs/bio/include\n",
            "    CFLAGS:         -Wall -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /root/.local/share/mamba/envs/bio/include -fdebug-prefix-map=/opt/conda/conda-bld/samtools_1734249775320/work=/usr/local/src/conda/samtools-1.21 -fdebug-prefix-map=/root/.local/share/mamba/envs/bio=/usr/local/src/conda-prefix\n",
            "    LDFLAGS:        -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/root/.local/share/mamba/envs/bio/lib -Wl,-rpath-link,/root/.local/share/mamba/envs/bio/lib -L/root/.local/share/mamba/envs/bio/lib\n",
            "    HTSDIR:         \n",
            "    LIBS:           \n",
            "    CURSES_LIB:     -ltinfow -lncursesw\n",
            "\n",
            "HTSlib compilation details:\n",
            "    Features:       build=configure libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes plugins=yes plugin-path=/root/.local/share/mamba/envs/bio/libexec/htslib htscodecs=1.6.1\n",
            "    CC:             /opt/conda/conda-bld/htslib_1734145038589/_build_env/bin/x86_64-conda-linux-gnu-cc\n",
            "    CPPFLAGS:       -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /root/.local/share/mamba/envs/bio/include\n",
            "    CFLAGS:         -Wall -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /root/.local/share/mamba/envs/bio/include -fdebug-prefix-map=/opt/conda/conda-bld/htslib_1734145038589/work=/usr/local/src/conda/htslib-1.21 -fdebug-prefix-map=/root/.local/share/mamba/envs/bio=/usr/local/src/conda-prefix -fvisibility=hidden\n",
            "    LDFLAGS:        -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/root/.local/share/mamba/envs/bio/lib -Wl,-rpath-link,/root/.local/share/mamba/envs/bio/lib -L/root/.local/share/mamba/envs/bio/lib -fvisibility=hidden -rdynamic\n",
            "\n",
            "HTSlib URL scheme handlers present:\n",
            "    built-in:\t file, preload, data\n",
            "    Amazon S3:\t s3+https, s3, s3+http\n",
            "    S3 Multipart Upload:\t s3w+https, s3w+http, s3w\n",
            "    Google Cloud Storage:\t gs+http, gs+https, gs\n",
            "    libcurl:\t gophers, smtp, wss, smb, rtsp, tftp, pop3, smbs, imaps, pop3s, ws, ftps, https, http, ftp, gopher, imap, sftp, smtps, scp, dict, mqtt, telnet\n",
            "    crypt4gh-needed:\t crypt4gh\n",
            "    mem:\t mem\n"
          ]
        }
      ],
      "source": [
        "# 1. Установка micromamba\n",
        "!wget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba\n",
        "\n",
        "# 2. Создание окружения и установка инструментов\n",
        "!./bin/micromamba create -y -n bio \\\n",
        "  -c bioconda -c conda-forge \\\n",
        "  fastqc trimmomatic bwa samtools bcftools fastp hisat2 bowtie2 bedtools seqtk multiqc\n",
        "\n",
        "# 3. Активация окружения и проверка (для примера)\n",
        "!./bin/micromamba run -n bio fastqc --help\n",
        "!./bin/micromamba run -n bio bwa\n",
        "!./bin/micromamba run -n bio samtools --version"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "OH4DAqBIr--i"
      },
      "source": [
        "# СКАЧИВАЕМ РЕФЕРЕНСНЫЙ ГЕНОМ ( **Pseudomonas aeruginosa**) через NCBI Datasets\n",
        "datasets accession GCF_000006765.1 --include gff3,rna,cds,protein,genome,seq-report\n",
        "\n",
        "https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000006765.1/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "2LHjCmFbeAt0",
        "outputId": "da38347f-b8a1-485a-bf3e-908c9ac037ea"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-05-09 18:23:13--  https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Pseudomonas_aeruginosa/reference/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.31, 130.14.250.7, 130.14.250.10, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.31|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1787220 (1.7M) [application/x-gzip]\n",
            "Saving to: ‘GCF_000006765.1_ASM676v1_genomic.fna.gz’\n",
            "\n",
            "GCF_000006765.1_ASM 100%[===================>]   1.70M  4.18MB/s    in 0.4s    \n",
            "\n",
            "2025-05-09 18:23:14 (4.18 MB/s) - ‘GCF_000006765.1_ASM676v1_genomic.fna.gz’ saved [1787220/1787220]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Pseudomonas_aeruginosa/reference/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
        "\n",
        "!gunzip GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
        "!mv GCF_000006765.1_ASM676v1_genomic.fna aeruginosa.fna\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0l2bPuv6wELR"
      },
      "source": [
        "# Cкачиваем риды (NGS-данные) для Pseudomonas aeruginosa из NCBI\n",
        "\n",
        "https://www.ncbi.nlm.nih.gov/sra/SRR12905311[accn]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "4SSvMl5SwJ6P",
        "outputId": "0d1a3699-1b70-4e68-b819-c932b81c758c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "sra-toolkit is already the newest version (2.11.3+dfsg-1ubuntu1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 34 not upgraded.\n",
            "Read 6371441 spots for SRR12905311\n",
            "Written 6371441 spots for SRR12905311\n"
          ]
        }
      ],
      "source": [
        "!apt-get install -y sra-toolkit\n",
        "!fastq-dump --split-files SRR12905311"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3-oexcbCy6kR"
      },
      "source": [
        "#Проверяем качество ридов триммируем"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "n7KEBXX0yYy5",
        "outputId": "23a4877d-f620-4de2-b5bd-5ef159394d77"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "FastQC уже установлен\n",
            "--2025-05-09 18:27:21--  http://opengene.org/fastp/fastp.0.23.4\n",
            "Resolving opengene.org (opengene.org)... 8.210.133.117\n",
            "Connecting to opengene.org (opengene.org)|8.210.133.117|:80... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://opengene.org/fastp/fastp.0.23.4 [following]\n",
            "--2025-05-09 18:27:22--  https://opengene.org/fastp/fastp.0.23.4\n",
            "Connecting to opengene.org (opengene.org)|8.210.133.117|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10946176 (10M) [application/octet-stream]\n",
            "Saving to: ‘fastp’\n",
            "\n",
            "fastp               100%[===================>]  10.44M  4.27MB/s    in 2.4s    \n",
            "\n",
            "2025-05-09 18:27:25 (4.27 MB/s) - ‘fastp’ saved [10946176/10946176]\n",
            "\n",
            "FastQC v0.12.1\n",
            "fastp 0.23.4\n",
            "null\n",
            "null\n",
            "Started analysis of SRR12905311_1.fastq\n",
            "Approx 5% complete for SRR12905311_1.fastq\n",
            "Approx 10% complete for SRR12905311_1.fastq\n",
            "Approx 15% complete for SRR12905311_1.fastq\n",
            "Approx 20% complete for SRR12905311_1.fastq\n",
            "Approx 25% complete for SRR12905311_1.fastq\n",
            "Approx 30% complete for SRR12905311_1.fastq\n",
            "Approx 35% complete for SRR12905311_1.fastq\n",
            "Approx 40% complete for SRR12905311_1.fastq\n",
            "Approx 45% complete for SRR12905311_1.fastq\n",
            "Approx 50% complete for SRR12905311_1.fastq\n",
            "Approx 55% complete for SRR12905311_1.fastq\n",
            "Approx 60% complete for SRR12905311_1.fastq\n",
            "Approx 65% complete for SRR12905311_1.fastq\n",
            "Approx 70% complete for SRR12905311_1.fastq\n",
            "Approx 75% complete for SRR12905311_1.fastq\n",
            "Approx 80% complete for SRR12905311_1.fastq\n",
            "Approx 85% complete for SRR12905311_1.fastq\n",
            "Approx 90% complete for SRR12905311_1.fastq\n",
            "Approx 95% complete for SRR12905311_1.fastq\n",
            "Analysis complete for SRR12905311_1.fastq\n",
            "Started analysis of SRR12905311_2.fastq\n",
            "Approx 5% complete for SRR12905311_2.fastq\n",
            "Approx 10% complete for SRR12905311_2.fastq\n",
            "Approx 15% complete for SRR12905311_2.fastq\n",
            "Approx 20% complete for SRR12905311_2.fastq\n",
            "Approx 25% complete for SRR12905311_2.fastq\n",
            "Approx 30% complete for SRR12905311_2.fastq\n",
            "Approx 35% complete for SRR12905311_2.fastq\n",
            "Approx 40% complete for SRR12905311_2.fastq\n",
            "Approx 45% complete for SRR12905311_2.fastq\n",
            "Approx 50% complete for SRR12905311_2.fastq\n",
            "Approx 55% complete for SRR12905311_2.fastq\n",
            "Approx 60% complete for SRR12905311_2.fastq\n",
            "Approx 65% complete for SRR12905311_2.fastq\n",
            "Approx 70% complete for SRR12905311_2.fastq\n",
            "Approx 75% complete for SRR12905311_2.fastq\n",
            "Approx 80% complete for SRR12905311_2.fastq\n",
            "Approx 85% complete for SRR12905311_2.fastq\n",
            "Approx 90% complete for SRR12905311_2.fastq\n",
            "Approx 95% complete for SRR12905311_2.fastq\n",
            "Analysis complete for SRR12905311_2.fastq\n",
            "Read1 before filtering:\n",
            "total reads: 6371441\n",
            "total bases: 506337238\n",
            "Q20 bases: 489048202(96.5855%)\n",
            "Q30 bases: 481766970(95.1474%)\n",
            "\n",
            "Read2 before filtering:\n",
            "total reads: 6371441\n",
            "total bases: 506091161\n",
            "Q20 bases: 483530535(95.5422%)\n",
            "Q30 bases: 476834184(94.219%)\n",
            "\n",
            "Read1 after filtering:\n",
            "total reads: 6218031\n",
            "total bases: 493884182\n",
            "Q20 bases: 479013522(96.989%)\n",
            "Q30 bases: 472157788(95.6009%)\n",
            "\n",
            "Read2 after filtering:\n",
            "total reads: 6218031\n",
            "total bases: 493890247\n",
            "Q20 bases: 477490738(96.6795%)\n",
            "Q30 bases: 471357879(95.4378%)\n",
            "\n",
            "Filtering result:\n",
            "reads passed filter: 12436062\n",
            "reads failed due to low quality: 300116\n",
            "reads failed due to too many N: 6704\n",
            "reads failed due to too short: 0\n",
            "reads with adapter trimmed: 29540\n",
            "bases trimmed due to adapters: 537300\n",
            "\n",
            "Duplication rate: 1.96095%\n",
            "\n",
            "Insert size peak (evaluated by paired-end reads): 125\n",
            "\n",
            "JSON report: fastp_report.json\n",
            "HTML report: fastp_report.html\n",
            "\n",
            "fastp -i SRR12905311_1.fastq -I SRR12905311_2.fastq -o clean_1.fastq -O clean_2.fastq -h fastp_report.html -j fastp_report.json \n",
            "fastp v0.23.4, time used: 94 seconds\n"
          ]
        }
      ],
      "source": [
        "# Установка FastQC (если ещё не установлен)\n",
        "!if [ ! -f \"/usr/local/bin/fastqc\" ]; then \\\n",
        "    wget -O fastqc.zip https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.12.1.zip && \\\n",
        "    unzip -o fastqc.zip && \\\n",
        "    chmod +x FastQC/fastqc && \\\n",
        "    ln -s $(pwd)/FastQC/fastqc /usr/local/bin/fastqc ; \\\n",
        "else echo \"FastQC уже установлен\"; fi\n",
        "\n",
        "# Установка fastp v0.23.4\n",
        "!wget http://opengene.org/fastp/fastp.0.23.4 -O fastp\n",
        "!chmod +x fastp\n",
        "!sudo mv fastp /usr/local/bin/\n",
        "\n",
        "# Проверка версий\n",
        "!fastqc --version\n",
        "!fastp --version\n",
        "\n",
        "# Запуск FastQC\n",
        "!fastqc SRR12905311_1.fastq SRR12905311_2.fastq\n",
        "\n",
        "# Тримминг fastp\n",
        "!fastp \\\n",
        "  -i SRR12905311_1.fastq \\\n",
        "  -I SRR12905311_2.fastq \\\n",
        "  -o clean_1.fastq \\\n",
        "  -O clean_2.fastq \\\n",
        "  -h fastp_report.html \\\n",
        "  -j fastp_report.json\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "U1xotzVKAoHK",
        "outputId": "687ee30d-9644-4063-c265-e031c2051443"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "fastp 0.23.4\n"
          ]
        }
      ],
      "source": [
        "!fastp --version\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "i-XGoll-LYzD",
        "outputId": "3091d6b1-805c-460c-f47e-e1fe8e51ae14"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "-rw-r--r-- 1 root root 1.5G May  9 18:27 SRR12905311_1.fastq\n",
            "-rw-r--r-- 1 root root 1.5G May  9 18:27 SRR12905311_2.fastq\n"
          ]
        }
      ],
      "source": [
        "#Проверяем создались ли файлы ридов\n",
        "!ls -lh SRR12905311_*.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "D_OlbeUWLe3o",
        "outputId": "63767d4a-1649-4a33-8cec-ffa3848e48fc"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "-rw-r--r-- 1 root root 1.4G May  9 18:30 clean_1.fastq\n",
            "-rw-r--r-- 1 root root 1.4G May  9 18:30 clean_2.fastq\n"
          ]
        }
      ],
      "source": [
        "#Проверяем создались ли очищенные после тримменга файлы ридов\n",
        "!ls -lh clean_*.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "WfRKOd7H4s8U"
      },
      "source": [
        "# Выравнивание ридов по генам в формате .bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "ROhHb8jT43T9",
        "outputId": "52080bc2-77e8-43a2-edef-a4bcfb764731"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "bwa is already the newest version (0.7.17-6).\n",
            "samtools is already the newest version (1.13-4).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 34 not upgraded.\n",
            "[bwa_index] Pack FASTA... 0.07 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[bwa_index] 4.15 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.04 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.03 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 1.41 sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa index aeruginosa.fna\n",
            "[main] Real time: 5.977 sec; CPU: 5.698 sec\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 125898 sequences (10000018 bp)...\n",
            "[M::process] read 125904 sequences (10000083 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (210, 52114, 6, 218)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (140, 186, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (189.39, 76.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.63, 84.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (144, 189, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 453)\n",
            "[M::mem_pestat] mean and std.dev: (194.40, 75.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 8.431 CPU sec, 8.338 real sec\n",
            "[M::process] read 125912 sequences (10000068 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (241, 52248, 9, 238)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 183, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 478)\n",
            "[M::mem_pestat] mean and std.dev: (188.38, 82.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (301.99, 84.79)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 184, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 457)\n",
            "[M::mem_pestat] mean and std.dev: (181.96, 80.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 10.331 CPU sec, 10.567 real sec\n",
            "[M::process] read 125896 sequences (10000047 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (240, 52377, 11, 224)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 176, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 480)\n",
            "[M::mem_pestat] mean and std.dev: (184.36, 82.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 598)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (301.68, 84.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (365, 947, 1440)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3590)\n",
            "[M::mem_pestat] mean and std.dev: (793.00, 599.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4665)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 172, 232)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 430)\n",
            "[M::mem_pestat] mean and std.dev: (180.38, 71.74)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 529)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125912 reads in 7.927 CPU sec, 7.717 real sec\n",
            "[M::process] read 125892 sequences (10000154 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (216, 51972, 7, 246)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 190, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 486)\n",
            "[M::mem_pestat] mean and std.dev: (191.96, 87.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.81, 85.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 174, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
            "[M::mem_pestat] mean and std.dev: (186.76, 87.62)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125896 reads in 10.710 CPU sec, 11.035 real sec\n",
            "[M::process] read 125886 sequences (10000154 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (263, 52271, 4, 241)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 178, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
            "[M::mem_pestat] mean and std.dev: (184.60, 80.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 300, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 577)\n",
            "[M::mem_pestat] mean and std.dev: (303.98, 84.50)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 687)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (114, 173, 228)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
            "[M::mem_pestat] mean and std.dev: (174.99, 76.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125892 reads in 8.521 CPU sec, 8.411 real sec\n",
            "[M::process] read 125914 sequences (10000022 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (233, 52279, 8, 233)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 172, 228)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 436)\n",
            "[M::mem_pestat] mean and std.dev: (178.32, 76.49)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (302.33, 84.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 176, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 472)\n",
            "[M::mem_pestat] mean and std.dev: (183.52, 85.96)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125886 reads in 12.076 CPU sec, 13.092 real sec\n",
            "[M::process] read 125886 sequences (10000046 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (249, 52233, 6, 251)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 171, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 460)\n",
            "[M::mem_pestat] mean and std.dev: (181.29, 80.91)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.19, 84.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (139, 194, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 451)\n",
            "[M::mem_pestat] mean and std.dev: (192.24, 72.95)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 555)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125914 reads in 8.774 CPU sec, 8.856 real sec\n",
            "[M::process] read 125902 sequences (10000139 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (235, 52301, 6, 232)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 177, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 453)\n",
            "[M::mem_pestat] mean and std.dev: (185.72, 79.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 561)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.04, 85.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (146, 194, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 416)\n",
            "[M::mem_pestat] mean and std.dev: (194.61, 77.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 506)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125886 reads in 9.606 CPU sec, 9.509 real sec\n",
            "[M::process] read 125904 sequences (10000108 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (235, 52199, 9, 228)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 181, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 486)\n",
            "[M::mem_pestat] mean and std.dev: (191.60, 80.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 577)\n",
            "[M::mem_pestat] mean and std.dev: (303.26, 84.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 687)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 190, 260)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 516)\n",
            "[M::mem_pestat] mean and std.dev: (191.70, 85.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 644)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125902 reads in 9.053 CPU sec, 9.167 real sec\n",
            "[M::process] read 125900 sequences (10000004 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (260, 52263, 11, 249)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 173, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 492)\n",
            "[M::mem_pestat] mean and std.dev: (185.87, 86.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 613)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.83, 84.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (1523, 1819, 7747)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 20195)\n",
            "[M::mem_pestat] mean and std.dev: (3400.27, 3193.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 26419)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 186, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 484)\n",
            "[M::mem_pestat] mean and std.dev: (192.02, 81.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 9.365 CPU sec, 9.388 real sec\n",
            "[M::process] read 125896 sequences (10000067 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (265, 52259, 11, 244)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (120, 177, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 492)\n",
            "[M::mem_pestat] mean and std.dev: (181.05, 79.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 616)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.67, 84.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (494, 890, 2000)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5012)\n",
            "[M::mem_pestat] mean and std.dev: (1139.60, 1063.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6518)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 176, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 485)\n",
            "[M::mem_pestat] mean and std.dev: (185.07, 84.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125900 reads in 9.269 CPU sec, 9.340 real sec\n",
            "[M::process] read 125902 sequences (10000099 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (246, 52136, 8, 255)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 172, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 463)\n",
            "[M::mem_pestat] mean and std.dev: (179.87, 81.95)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.67, 84.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 185, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (193.11, 85.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 589)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125896 reads in 8.908 CPU sec, 8.814 real sec\n",
            "[M::process] read 125908 sequences (10000119 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (226, 52301, 7, 218)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 177, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 478)\n",
            "[M::mem_pestat] mean and std.dev: (182.48, 78.77)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 596)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.27, 84.86)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 190, 251)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 495)\n",
            "[M::mem_pestat] mean and std.dev: (191.00, 85.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 617)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125902 reads in 9.738 CPU sec, 9.855 real sec\n",
            "[M::process] read 125906 sequences (10000087 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (242, 52131, 8, 241)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 187, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
            "[M::mem_pestat] mean and std.dev: (191.25, 80.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 563)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.56, 84.93)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (111, 172, 232)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 474)\n",
            "[M::mem_pestat] mean and std.dev: (176.31, 85.96)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 8.644 CPU sec, 8.487 real sec\n",
            "[M::process] read 125888 sequences (10000139 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (237, 52277, 6, 258)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 177, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 484)\n",
            "[M::mem_pestat] mean and std.dev: (182.53, 83.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (300.61, 84.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 171, 227)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 437)\n",
            "[M::mem_pestat] mean and std.dev: (177.49, 76.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 542)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125906 reads in 10.207 CPU sec, 10.292 real sec\n",
            "[M::process] read 125904 sequences (10000138 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (244, 52000, 10, 257)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (142, 202, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 442)\n",
            "[M::mem_pestat] mean and std.dev: (193.10, 76.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 542)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.16, 85.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (386, 1880, 1984)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5180)\n",
            "[M::mem_pestat] mean and std.dev: (968.25, 785.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6778)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (119, 170, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
            "[M::mem_pestat] mean and std.dev: (178.64, 81.12)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125888 reads in 8.416 CPU sec, 8.227 real sec\n",
            "[M::process] read 125900 sequences (10000094 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (253, 51913, 1, 244)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 185, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 448)\n",
            "[M::mem_pestat] mean and std.dev: (181.36, 77.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 555)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 580)\n",
            "[M::mem_pestat] mean and std.dev: (303.65, 85.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 186, 251)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 519)\n",
            "[M::mem_pestat] mean and std.dev: (191.65, 90.49)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 653)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 10.510 CPU sec, 10.724 real sec\n",
            "[M::process] read 125898 sequences (10000085 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (290, 52151, 6, 213)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 189, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 453)\n",
            "[M::mem_pestat] mean and std.dev: (187.71, 82.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.34, 84.02)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 185, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 482)\n",
            "[M::mem_pestat] mean and std.dev: (192.86, 83.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 599)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125900 reads in 7.720 CPU sec, 7.497 real sec\n",
            "[M::process] read 125894 sequences (10000064 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (270, 52354, 6, 254)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (114, 173, 226)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 450)\n",
            "[M::mem_pestat] mean and std.dev: (170.72, 74.49)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.76, 84.07)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (137, 190, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (193.33, 81.55)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 11.134 CPU sec, 11.341 real sec\n",
            "[M::process] read 125902 sequences (10000044 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (243, 52253, 8, 228)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 178, 232)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 446)\n",
            "[M::mem_pestat] mean and std.dev: (180.82, 77.55)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 553)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.42, 84.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (119, 188, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 494)\n",
            "[M::mem_pestat] mean and std.dev: (185.62, 84.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 619)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125894 reads in 7.758 CPU sec, 7.543 real sec\n",
            "[M::process] read 125896 sequences (10000002 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (239, 52107, 12, 249)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 180, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (191.23, 86.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 566)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 580)\n",
            "[M::mem_pestat] mean and std.dev: (303.56, 84.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (588, 1342, 1458)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3198)\n",
            "[M::mem_pestat] mean and std.dev: (1006.00, 769.55)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4084)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 180, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 454)\n",
            "[M::mem_pestat] mean and std.dev: (183.57, 79.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125902 reads in 11.092 CPU sec, 11.181 real sec\n",
            "[M::process] read 125888 sequences (10000042 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (229, 52044, 9, 246)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 186, 255)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 501)\n",
            "[M::mem_pestat] mean and std.dev: (192.67, 90.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 624)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.32, 85.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 175, 223)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 425)\n",
            "[M::mem_pestat] mean and std.dev: (177.07, 75.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 526)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125896 reads in 7.734 CPU sec, 7.496 real sec\n",
            "[M::process] read 125898 sequences (10000105 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (249, 52283, 9, 240)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 190, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (190.81, 75.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 571)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (301.34, 84.12)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 176, 254)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 518)\n",
            "[M::mem_pestat] mean and std.dev: (190.40, 92.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 650)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125888 reads in 10.675 CPU sec, 10.750 real sec\n",
            "[M::process] read 125896 sequences (10000138 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (296, 52039, 9, 240)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 187, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 453)\n",
            "[M::mem_pestat] mean and std.dev: (186.80, 77.77)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 559)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.38, 84.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 175, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 480)\n",
            "[M::mem_pestat] mean and std.dev: (184.90, 84.61)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 8.067 CPU sec, 8.141 real sec\n",
            "[M::process] read 125914 sequences (10000065 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (253, 52155, 7, 232)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 181, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 478)\n",
            "[M::mem_pestat] mean and std.dev: (190.35, 80.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.89, 84.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 178, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
            "[M::mem_pestat] mean and std.dev: (182.37, 79.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125896 reads in 9.926 CPU sec, 9.830 real sec\n",
            "[M::process] read 125910 sequences (10000060 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (262, 52263, 3, 229)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 182, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 476)\n",
            "[M::mem_pestat] mean and std.dev: (186.20, 84.78)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 594)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (301.25, 84.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 185, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (186.76, 78.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125914 reads in 8.888 CPU sec, 9.027 real sec\n",
            "[M::process] read 125896 sequences (10000025 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (240, 52117, 10, 241)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 181, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 496)\n",
            "[M::mem_pestat] mean and std.dev: (183.53, 85.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 621)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.84, 84.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (196, 1295, 4629)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13495)\n",
            "[M::mem_pestat] mean and std.dev: (2312.80, 2990.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 17928)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 183, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 460)\n",
            "[M::mem_pestat] mean and std.dev: (185.26, 74.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 571)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125910 reads in 9.200 CPU sec, 9.084 real sec\n",
            "[M::process] read 125880 sequences (10000095 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (257, 52331, 10, 224)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 175, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 483)\n",
            "[M::mem_pestat] mean and std.dev: (185.03, 84.59)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 603)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (302.61, 84.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (168, 733, 1476)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4092)\n",
            "[M::mem_pestat] mean and std.dev: (804.80, 684.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5400)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 177, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 514)\n",
            "[M::mem_pestat] mean and std.dev: (181.98, 81.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 646)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125896 reads in 9.538 CPU sec, 9.660 real sec\n",
            "[M::process] read 125884 sequences (10000038 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (254, 52005, 12, 264)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 196, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
            "[M::mem_pestat] mean and std.dev: (191.34, 80.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (303.25, 84.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (285, 1910, 5307)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15351)\n",
            "[M::mem_pestat] mean and std.dev: (3150.17, 3186.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 20373)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (106, 171, 232)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 484)\n",
            "[M::mem_pestat] mean and std.dev: (174.25, 80.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 610)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125880 reads in 8.525 CPU sec, 8.424 real sec\n",
            "[M::process] read 125898 sequences (10000087 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (219, 51995, 11, 221)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 183, 258)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 524)\n",
            "[M::mem_pestat] mean and std.dev: (189.74, 84.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 657)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (302.07, 84.66)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (719, 1306, 1392)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2738)\n",
            "[M::mem_pestat] mean and std.dev: (894.40, 565.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3411)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 186, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 483)\n",
            "[M::mem_pestat] mean and std.dev: (183.91, 78.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 603)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125884 reads in 12.040 CPU sec, 13.097 real sec\n",
            "[M::process] read 125902 sequences (10000069 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (259, 52317, 8, 232)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (112, 169, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 487)\n",
            "[M::mem_pestat] mean and std.dev: (181.16, 84.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.60, 84.49)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 179, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 470)\n",
            "[M::mem_pestat] mean and std.dev: (183.42, 81.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 585)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 9.124 CPU sec, 9.266 real sec\n",
            "[M::process] read 125898 sequences (10000142 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (244, 52320, 8, 228)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 171, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 484)\n",
            "[M::mem_pestat] mean and std.dev: (177.89, 86.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.53, 85.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (116, 172, 230)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (173.90, 81.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 572)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125902 reads in 11.107 CPU sec, 11.212 real sec\n",
            "[M::process] read 125898 sequences (10000134 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (234, 51889, 8, 225)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 177, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 474)\n",
            "[M::mem_pestat] mean and std.dev: (186.26, 87.00)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 589)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.74, 84.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 190, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
            "[M::mem_pestat] mean and std.dev: (189.52, 82.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 587)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 10.665 CPU sec, 11.315 real sec\n",
            "[M::process] read 125898 sequences (10000044 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (247, 52156, 13, 243)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 176, 232)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 430)\n",
            "[M::mem_pestat] mean and std.dev: (179.37, 70.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 529)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (301.55, 85.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (59, 288, 1422)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4148)\n",
            "[M::mem_pestat] mean and std.dev: (423.25, 487.00)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5511)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 184, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 485)\n",
            "[M::mem_pestat] mean and std.dev: (185.70, 84.79)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 12.150 CPU sec, 13.275 real sec\n",
            "[M::process] read 125890 sequences (10000039 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (241, 52337, 11, 265)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 163, 229)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 451)\n",
            "[M::mem_pestat] mean and std.dev: (175.72, 76.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.17, 84.92)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (377, 726, 1410)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3476)\n",
            "[M::mem_pestat] mean and std.dev: (643.22, 458.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4509)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 184, 252)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 520)\n",
            "[M::mem_pestat] mean and std.dev: (188.12, 88.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 654)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 8.310 CPU sec, 8.360 real sec\n",
            "[M::process] read 125890 sequences (10000094 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (262, 52288, 6, 253)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 182, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 463)\n",
            "[M::mem_pestat] mean and std.dev: (180.19, 80.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 577)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.85, 84.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 172, 254)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 514)\n",
            "[M::mem_pestat] mean and std.dev: (188.38, 89.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 644)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125890 reads in 10.408 CPU sec, 10.340 real sec\n",
            "[M::process] read 125906 sequences (10000101 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (228, 52146, 8, 222)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 182, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 481)\n",
            "[M::mem_pestat] mean and std.dev: (184.49, 83.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.81, 84.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 182, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 495)\n",
            "[M::mem_pestat] mean and std.dev: (189.21, 88.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 619)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125890 reads in 8.950 CPU sec, 9.037 real sec\n",
            "[M::process] read 125906 sequences (10000149 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (227, 51981, 7, 244)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 182, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 459)\n",
            "[M::mem_pestat] mean and std.dev: (190.26, 80.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 567)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.19, 84.55)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 183, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 480)\n",
            "[M::mem_pestat] mean and std.dev: (190.61, 91.57)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 599)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125906 reads in 9.915 CPU sec, 9.800 real sec\n",
            "[M::process] read 125910 sequences (10000075 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (245, 52227, 4, 185)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 184, 229)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 437)\n",
            "[M::mem_pestat] mean and std.dev: (183.41, 78.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 541)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.96, 84.09)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (116, 181, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 485)\n",
            "[M::mem_pestat] mean and std.dev: (185.27, 89.55)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125906 reads in 9.758 CPU sec, 9.864 real sec\n",
            "[M::process] read 125900 sequences (10000091 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (271, 52220, 12, 255)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 183, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 449)\n",
            "[M::mem_pestat] mean and std.dev: (185.58, 80.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 554)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.09, 84.91)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (66, 936, 3327)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9849)\n",
            "[M::mem_pestat] mean and std.dev: (1758.92, 2426.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13110)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 168, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
            "[M::mem_pestat] mean and std.dev: (180.31, 79.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125910 reads in 8.992 CPU sec, 8.846 real sec\n",
            "[M::process] read 125894 sequences (10000049 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (224, 52105, 9, 235)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 169, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (181.89, 78.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 577)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.96, 84.91)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 186, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
            "[M::mem_pestat] mean and std.dev: (187.60, 78.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 550)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125900 reads in 10.587 CPU sec, 10.757 real sec\n",
            "[M::process] read 125908 sequences (10000062 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (267, 52069, 11, 242)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 180, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 474)\n",
            "[M::mem_pestat] mean and std.dev: (185.68, 77.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 588)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (302.84, 85.12)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (151, 489, 1379)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3835)\n",
            "[M::mem_pestat] mean and std.dev: (537.67, 520.77)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5063)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 177, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 477)\n",
            "[M::mem_pestat] mean and std.dev: (185.93, 80.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125894 reads in 8.699 CPU sec, 8.507 real sec\n",
            "[M::process] read 125890 sequences (10000014 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (279, 52302, 9, 215)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (120, 175, 229)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 447)\n",
            "[M::mem_pestat] mean and std.dev: (179.41, 75.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (301.21, 84.09)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 192, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 483)\n",
            "[M::mem_pestat] mean and std.dev: (192.26, 84.08)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 11.210 CPU sec, 11.606 real sec\n",
            "[M::process] read 125892 sequences (10000053 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (207, 52168, 9, 244)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (136, 189, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
            "[M::mem_pestat] mean and std.dev: (195.27, 82.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.93, 84.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 192, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 484)\n",
            "[M::mem_pestat] mean and std.dev: (193.26, 85.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125890 reads in 8.106 CPU sec, 7.858 real sec\n",
            "[M::process] read 125908 sequences (10000121 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (247, 52251, 10, 248)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (136, 188, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 460)\n",
            "[M::mem_pestat] mean and std.dev: (189.84, 78.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.55, 84.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (198, 1008, 1059)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2781)\n",
            "[M::mem_pestat] mean and std.dev: (633.67, 477.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3642)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 182, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (182.43, 76.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125892 reads in 12.001 CPU sec, 12.103 real sec\n",
            "[M::process] read 125906 sequences (10000118 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (240, 51961, 14, 225)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 188, 261)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 529)\n",
            "[M::mem_pestat] mean and std.dev: (194.87, 88.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 663)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 300, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 577)\n",
            "[M::mem_pestat] mean and std.dev: (302.80, 83.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 687)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (216, 955, 1975)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5493)\n",
            "[M::mem_pestat] mean and std.dev: (1483.43, 1684.92)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8223)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 172, 226)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 434)\n",
            "[M::mem_pestat] mean and std.dev: (175.66, 73.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 538)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 8.164 CPU sec, 7.922 real sec\n",
            "[M::process] read 125894 sequences (10000074 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (289, 52178, 11, 255)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 175, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 480)\n",
            "[M::mem_pestat] mean and std.dev: (182.82, 85.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (300.87, 84.40)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (295, 808, 1447)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3751)\n",
            "[M::mem_pestat] mean and std.dev: (613.33, 507.07)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4903)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (143, 187, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 437)\n",
            "[M::mem_pestat] mean and std.dev: (189.86, 76.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 535)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125906 reads in 11.847 CPU sec, 11.818 real sec\n",
            "[M::process] read 125898 sequences (10000127 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (236, 52097, 10, 256)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 169, 226)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 444)\n",
            "[M::mem_pestat] mean and std.dev: (173.99, 78.59)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 553)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.44, 84.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (91, 972, 1427)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4099)\n",
            "[M::mem_pestat] mean and std.dev: (848.56, 638.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5435)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 190, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 489)\n",
            "[M::mem_pestat] mean and std.dev: (188.60, 84.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 611)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125894 reads in 8.677 CPU sec, 8.672 real sec\n",
            "[M::process] read 125896 sequences (10000142 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (232, 51924, 8, 204)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 173, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (183.55, 84.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (301.92, 84.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 190, 254)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 508)\n",
            "[M::mem_pestat] mean and std.dev: (193.71, 89.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 635)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 11.847 CPU sec, 11.818 real sec\n",
            "[M::process] read 125920 sequences (10000097 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (249, 52147, 4, 241)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 180, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 477)\n",
            "[M::mem_pestat] mean and std.dev: (185.70, 85.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.02, 85.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 176, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (185.80, 82.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 575)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125896 reads in 8.254 CPU sec, 8.308 real sec\n",
            "[M::process] read 125902 sequences (10000142 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (235, 52332, 5, 216)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 179, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 442)\n",
            "[M::mem_pestat] mean and std.dev: (185.68, 81.57)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 546)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.93, 84.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 169, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
            "[M::mem_pestat] mean and std.dev: (177.69, 79.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 582)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125920 reads in 10.599 CPU sec, 10.531 real sec\n",
            "[M::process] read 125892 sequences (10000142 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (224, 52342, 6, 231)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 188, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
            "[M::mem_pestat] mean and std.dev: (187.88, 77.93)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.80, 84.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 182, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 486)\n",
            "[M::mem_pestat] mean and std.dev: (191.47, 86.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125902 reads in 10.778 CPU sec, 11.036 real sec\n",
            "[M::process] read 125908 sequences (10000028 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (232, 52081, 5, 231)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (116, 174, 232)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
            "[M::mem_pestat] mean and std.dev: (178.69, 84.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.01, 84.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (147, 194, 252)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (195.05, 76.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 567)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125892 reads in 12.148 CPU sec, 13.083 real sec\n",
            "[M::process] read 125900 sequences (10000038 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (262, 51936, 11, 241)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 180, 251)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 497)\n",
            "[M::mem_pestat] mean and std.dev: (187.91, 79.79)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 620)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.44, 85.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (347, 726, 1488)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3770)\n",
            "[M::mem_pestat] mean and std.dev: (860.82, 585.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4911)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 178, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 455)\n",
            "[M::mem_pestat] mean and std.dev: (181.60, 76.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 563)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 9.293 CPU sec, 9.391 real sec\n",
            "[M::process] read 125900 sequences (10000026 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (256, 52283, 5, 242)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 178, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 469)\n",
            "[M::mem_pestat] mean and std.dev: (188.01, 81.55)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 582)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.43, 84.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 178, 230)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 420)\n",
            "[M::mem_pestat] mean and std.dev: (181.25, 72.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 515)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125900 reads in 9.460 CPU sec, 9.442 real sec\n",
            "[M::process] read 125886 sequences (10000136 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (242, 52476, 8, 230)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (114, 166, 228)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
            "[M::mem_pestat] mean and std.dev: (173.15, 79.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 354)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 574)\n",
            "[M::mem_pestat] mean and std.dev: (300.40, 84.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 684)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 182, 230)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 446)\n",
            "[M::mem_pestat] mean and std.dev: (182.35, 75.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 554)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125900 reads in 10.131 CPU sec, 10.287 real sec\n",
            "[M::process] read 125906 sequences (10000154 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (193, 52054, 10, 213)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (146, 191, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 428)\n",
            "[M::mem_pestat] mean and std.dev: (187.96, 74.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 522)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.72, 85.01)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (972, 1521, 1929)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3843)\n",
            "[M::mem_pestat] mean and std.dev: (1140.25, 678.68)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4800)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 182, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 470)\n",
            "[M::mem_pestat] mean and std.dev: (189.09, 78.96)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125886 reads in 8.696 CPU sec, 8.506 real sec\n",
            "[M::process] read 125918 sequences (10000057 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (243, 52240, 8, 263)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 176, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
            "[M::mem_pestat] mean and std.dev: (184.62, 82.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 550)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (302.25, 84.68)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 181, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
            "[M::mem_pestat] mean and std.dev: (183.85, 81.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125906 reads in 11.182 CPU sec, 11.467 real sec\n",
            "[M::process] read 125908 sequences (10000140 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (262, 52268, 12, 241)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 183, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 477)\n",
            "[M::mem_pestat] mean and std.dev: (185.88, 82.87)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (243, 298, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 579)\n",
            "[M::mem_pestat] mean and std.dev: (301.02, 84.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 689, 1516)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4280)\n",
            "[M::mem_pestat] mean and std.dev: (963.92, 943.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5662)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (115, 174, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 502)\n",
            "[M::mem_pestat] mean and std.dev: (185.09, 94.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 631)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125918 reads in 8.046 CPU sec, 7.796 real sec\n",
            "[M::process] read 125888 sequences (10000115 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (254, 52381, 8, 291)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (145, 191, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 430)\n",
            "[M::mem_pestat] mean and std.dev: (193.83, 79.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 525)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (243, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 579)\n",
            "[M::mem_pestat] mean and std.dev: (300.36, 84.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (140, 187, 241)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 443)\n",
            "[M::mem_pestat] mean and std.dev: (188.27, 74.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 11.867 CPU sec, 12.010 real sec\n",
            "[M::process] read 125898 sequences (10000047 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (216, 52304, 8, 243)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 179, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (187.90, 79.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.28, 84.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 180, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
            "[M::mem_pestat] mean and std.dev: (186.47, 79.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125888 reads in 8.103 CPU sec, 7.860 real sec\n",
            "[M::process] read 125912 sequences (10000097 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (251, 52233, 8, 222)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 177, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 502)\n",
            "[M::mem_pestat] mean and std.dev: (189.51, 89.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 628)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.70, 84.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 187, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
            "[M::mem_pestat] mean and std.dev: (183.61, 77.92)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 11.114 CPU sec, 11.076 real sec\n",
            "[M::process] read 125916 sequences (10000146 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (216, 52128, 5, 251)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 181, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 454)\n",
            "[M::mem_pestat] mean and std.dev: (188.34, 78.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 561)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.91, 84.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 180, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 448)\n",
            "[M::mem_pestat] mean and std.dev: (185.46, 74.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 554)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125912 reads in 9.157 CPU sec, 9.215 real sec\n",
            "[M::process] read 125898 sequences (10000025 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (254, 52436, 7, 245)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (115, 179, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 499)\n",
            "[M::mem_pestat] mean and std.dev: (185.73, 84.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 627)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 354)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 574)\n",
            "[M::mem_pestat] mean and std.dev: (300.16, 84.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 684)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 183, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 491)\n",
            "[M::mem_pestat] mean and std.dev: (188.32, 84.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 613)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125916 reads in 10.211 CPU sec, 10.185 real sec\n",
            "[M::process] read 125892 sequences (10000157 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (217, 52217, 10, 225)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 174, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 488)\n",
            "[M::mem_pestat] mean and std.dev: (183.00, 79.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.68, 84.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (151, 496, 1571)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4411)\n",
            "[M::mem_pestat] mean and std.dev: (736.50, 708.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5831)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 193, 252)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 500)\n",
            "[M::mem_pestat] mean and std.dev: (190.47, 80.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 624)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 9.740 CPU sec, 9.918 real sec\n",
            "[M::process] read 125896 sequences (10000064 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (238, 52167, 10, 243)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 184, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 457)\n",
            "[M::mem_pestat] mean and std.dev: (183.24, 78.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 567)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 580)\n",
            "[M::mem_pestat] mean and std.dev: (301.01, 85.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 692)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (156, 1070, 1478)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4122)\n",
            "[M::mem_pestat] mean and std.dev: (939.00, 825.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5444)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 181, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (186.75, 79.79)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 567)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125892 reads in 9.180 CPU sec, 9.051 real sec\n",
            "[M::process] read 125898 sequences (10000080 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (254, 52120, 9, 228)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 181, 249)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 505)\n",
            "[M::mem_pestat] mean and std.dev: (189.97, 84.09)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 633)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.97, 84.68)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (142, 182, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
            "[M::mem_pestat] mean and std.dev: (194.48, 82.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 546)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125896 reads in 10.737 CPU sec, 11.009 real sec\n",
            "[M::process] read 125902 sequences (10000156 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (243, 52284, 11, 246)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 180, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 488)\n",
            "[M::mem_pestat] mean and std.dev: (188.14, 85.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (243, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 579)\n",
            "[M::mem_pestat] mean and std.dev: (300.77, 85.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (94, 307, 1484)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4264)\n",
            "[M::mem_pestat] mean and std.dev: (770.00, 847.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5654)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 169, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 461)\n",
            "[M::mem_pestat] mean and std.dev: (179.46, 78.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 574)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 8.608 CPU sec, 8.431 real sec\n",
            "[M::process] read 125892 sequences (10000139 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (227, 52075, 13, 235)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (120, 176, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 480)\n",
            "[M::mem_pestat] mean and std.dev: (182.60, 79.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.36, 84.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 261, 1002)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2756)\n",
            "[M::mem_pestat] mean and std.dev: (369.08, 384.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3633)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (120, 176, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 468)\n",
            "[M::mem_pestat] mean and std.dev: (182.18, 83.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 584)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125902 reads in 11.610 CPU sec, 11.977 real sec\n",
            "[M::process] read 125904 sequences (10000157 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (265, 52170, 5, 242)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (140, 192, 257)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 491)\n",
            "[M::mem_pestat] mean and std.dev: (197.84, 84.86)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (302.74, 84.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 166, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 461)\n",
            "[M::mem_pestat] mean and std.dev: (176.56, 75.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 574)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125892 reads in 8.156 CPU sec, 7.899 real sec\n",
            "[M::process] read 125900 sequences (10000125 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (257, 52203, 6, 249)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 168, 230)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 444)\n",
            "[M::mem_pestat] mean and std.dev: (173.81, 76.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 583)\n",
            "[M::mem_pestat] mean and std.dev: (301.68, 85.77)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 696)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 195, 254)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 500)\n",
            "[M::mem_pestat] mean and std.dev: (193.62, 82.86)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 623)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 11.528 CPU sec, 11.680 real sec\n",
            "[M::process] read 125888 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (257, 52275, 10, 231)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (116, 165, 223)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 437)\n",
            "[M::mem_pestat] mean and std.dev: (172.37, 80.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (243, 297, 354)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 576)\n",
            "[M::mem_pestat] mean and std.dev: (300.28, 85.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 687)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (261, 482, 837)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1989)\n",
            "[M::mem_pestat] mean and std.dev: (537.00, 457.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2565)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (120, 177, 231)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 453)\n",
            "[M::mem_pestat] mean and std.dev: (180.40, 79.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125900 reads in 8.424 CPU sec, 8.484 real sec\n",
            "[M::process] read 125898 sequences (10000095 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (250, 52146, 12, 215)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (137, 193, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 455)\n",
            "[M::mem_pestat] mean and std.dev: (192.96, 82.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 561)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (302.03, 84.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (321, 510, 2003)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5367)\n",
            "[M::mem_pestat] mean and std.dev: (1371.25, 1824.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8670)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 183, 233)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 431)\n",
            "[M::mem_pestat] mean and std.dev: (182.42, 78.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 530)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125888 reads in 10.428 CPU sec, 10.349 real sec\n",
            "[M::process] read 125918 sequences (10000152 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (233, 52163, 9, 252)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 187, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 477)\n",
            "[M::mem_pestat] mean and std.dev: (191.03, 82.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 594)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.31, 84.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 184, 251)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 519)\n",
            "[M::mem_pestat] mean and std.dev: (189.19, 92.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 653)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 9.586 CPU sec, 9.701 real sec\n",
            "[M::process] read 125918 sequences (10000050 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (242, 52252, 8, 255)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 179, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (182.24, 84.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.04, 84.96)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (136, 187, 247)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 469)\n",
            "[M::mem_pestat] mean and std.dev: (189.72, 77.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125918 reads in 9.841 CPU sec, 10.929 real sec\n",
            "[M::process] read 125900 sequences (10000003 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (238, 52007, 12, 278)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 171, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 450)\n",
            "[M::mem_pestat] mean and std.dev: (186.79, 77.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 555)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 579)\n",
            "[M::mem_pestat] mean and std.dev: (303.00, 84.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 690)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (527, 982, 1301)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2849)\n",
            "[M::mem_pestat] mean and std.dev: (1023.42, 855.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4445)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 182, 245)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 483)\n",
            "[M::mem_pestat] mean and std.dev: (187.15, 86.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125918 reads in 12.169 CPU sec, 12.675 real sec\n",
            "[M::process] read 125904 sequences (10000084 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (254, 52246, 9, 246)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 182, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 436)\n",
            "[M::mem_pestat] mean and std.dev: (185.55, 75.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 537)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.93, 84.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 184, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 461)\n",
            "[M::mem_pestat] mean and std.dev: (191.49, 82.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125900 reads in 9.679 CPU sec, 9.946 real sec\n",
            "[M::process] read 125902 sequences (10000095 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (222, 52158, 4, 217)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 172, 230)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 446)\n",
            "[M::mem_pestat] mean and std.dev: (174.37, 75.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 554)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.40, 85.08)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 179, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 482)\n",
            "[M::mem_pestat] mean and std.dev: (186.29, 82.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 598)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 9.148 CPU sec, 9.057 real sec\n",
            "[M::process] read 125908 sequences (10000122 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (245, 52130, 11, 225)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 181, 225)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 425)\n",
            "[M::mem_pestat] mean and std.dev: (178.10, 69.96)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 525)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (302.01, 85.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (663, 1072, 1435)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2979)\n",
            "[M::mem_pestat] mean and std.dev: (920.80, 455.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3751)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 171, 234)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
            "[M::mem_pestat] mean and std.dev: (178.10, 74.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125902 reads in 10.914 CPU sec, 11.141 real sec\n",
            "[M::process] read 125906 sequences (10000055 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (212, 52008, 9, 264)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 183, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 455)\n",
            "[M::mem_pestat] mean and std.dev: (180.66, 71.00)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 565)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 300, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.65, 84.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (140, 185, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 437)\n",
            "[M::mem_pestat] mean and std.dev: (185.26, 74.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 536)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 8.193 CPU sec, 7.992 real sec\n",
            "[M::process] read 125890 sequences (10000064 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (253, 52308, 14, 243)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (120, 174, 229)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 447)\n",
            "[M::mem_pestat] mean and std.dev: (178.15, 75.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.03, 85.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (213, 327, 1496)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4062)\n",
            "[M::mem_pestat] mean and std.dev: (594.62, 558.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5345)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (130, 180, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 454)\n",
            "[M::mem_pestat] mean and std.dev: (181.44, 74.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125906 reads in 11.846 CPU sec, 12.296 real sec\n",
            "[M::process] read 125886 sequences (10000129 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (265, 52136, 13, 227)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 187, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
            "[M::mem_pestat] mean and std.dev: (184.61, 75.76)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 566)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.49, 84.60)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (433, 691, 1982)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5080)\n",
            "[M::mem_pestat] mean and std.dev: (1031.08, 803.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6629)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 167, 252)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 512)\n",
            "[M::mem_pestat] mean and std.dev: (181.54, 87.90)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 642)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125890 reads in 8.056 CPU sec, 7.811 real sec\n",
            "[M::process] read 125894 sequences (10000050 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (259, 52124, 13, 220)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 184, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 483)\n",
            "[M::mem_pestat] mean and std.dev: (187.37, 82.59)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 603)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.78, 85.02)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (784, 1398, 5235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14137)\n",
            "[M::mem_pestat] mean and std.dev: (2117.15, 1923.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18588)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 185, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 455)\n",
            "[M::mem_pestat] mean and std.dev: (180.88, 78.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125886 reads in 10.923 CPU sec, 10.976 real sec\n",
            "[M::process] read 125918 sequences (10000080 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (234, 52200, 9, 215)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 193, 254)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 500)\n",
            "[M::mem_pestat] mean and std.dev: (192.70, 81.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 623)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.34, 84.74)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 168, 236)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
            "[M::mem_pestat] mean and std.dev: (182.91, 76.91)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 566)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125894 reads in 9.104 CPU sec, 9.225 real sec\n",
            "[M::process] read 125904 sequences (10000039 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (263, 52291, 10, 267)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (119, 181, 249)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 509)\n",
            "[M::mem_pestat] mean and std.dev: (187.92, 87.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 639)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.92, 84.93)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (334, 1072, 1359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3409)\n",
            "[M::mem_pestat] mean and std.dev: (860.89, 749.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4434)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (118, 175, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (180.61, 82.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 594)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125918 reads in 9.487 CPU sec, 9.477 real sec\n",
            "[M::process] read 125904 sequences (10000092 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (224, 52197, 12, 234)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 187, 246)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
            "[M::mem_pestat] mean and std.dev: (188.55, 82.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.60, 84.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 1319, 5274)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15588)\n",
            "[M::mem_pestat] mean and std.dev: (2324.50, 2574.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 20745)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 185, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 462)\n",
            "[M::mem_pestat] mean and std.dev: (184.72, 79.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 10.491 CPU sec, 10.704 real sec\n",
            "[M::process] read 125894 sequences (10000004 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (277, 51989, 14, 252)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 186, 254)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 508)\n",
            "[M::mem_pestat] mean and std.dev: (189.57, 81.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 635)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.82, 84.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (1315, 1437, 1504)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (937, 1882)\n",
            "[M::mem_pestat] mean and std.dev: (1423.20, 195.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (641, 2205)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 186, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 447)\n",
            "[M::mem_pestat] mean and std.dev: (185.58, 79.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 8.720 CPU sec, 8.537 real sec\n",
            "[M::process] read 125898 sequences (10000040 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (232, 51928, 10, 250)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (124, 177, 228)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 436)\n",
            "[M::mem_pestat] mean and std.dev: (177.11, 76.91)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 358)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 582)\n",
            "[M::mem_pestat] mean and std.dev: (303.37, 84.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 560, 1362)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3832)\n",
            "[M::mem_pestat] mean and std.dev: (684.00, 754.92)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5067)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (135, 194, 250)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 480)\n",
            "[M::mem_pestat] mean and std.dev: (194.85, 83.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125894 reads in 12.366 CPU sec, 12.641 real sec\n",
            "[M::process] read 125908 sequences (10000028 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (259, 52272, 6, 255)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (134, 177, 235)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 437)\n",
            "[M::mem_pestat] mean and std.dev: (181.93, 74.74)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 538)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 577)\n",
            "[M::mem_pestat] mean and std.dev: (300.28, 84.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 688)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 180, 242)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 476)\n",
            "[M::mem_pestat] mean and std.dev: (182.21, 77.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125898 reads in 8.453 CPU sec, 8.266 real sec\n",
            "[M::process] read 125894 sequences (10000031 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (259, 52092, 14, 278)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (138, 194, 259)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 501)\n",
            "[M::mem_pestat] mean and std.dev: (195.26, 85.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 622)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.93, 84.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (287, 1277, 1384)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3578)\n",
            "[M::mem_pestat] mean and std.dev: (938.00, 811.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4675)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (129, 179, 249)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 489)\n",
            "[M::mem_pestat] mean and std.dev: (188.45, 82.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 11.980 CPU sec, 12.030 real sec\n",
            "[M::process] read 125916 sequences (10000036 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (272, 52278, 8, 259)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 177, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 451)\n",
            "[M::mem_pestat] mean and std.dev: (191.55, 84.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 557)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (246, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 576)\n",
            "[M::mem_pestat] mean and std.dev: (302.13, 84.68)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 686)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (126, 179, 254)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 510)\n",
            "[M::mem_pestat] mean and std.dev: (186.45, 84.09)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 638)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125894 reads in 8.588 CPU sec, 8.688 real sec\n",
            "[M::process] read 125918 sequences (10000036 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (246, 52047, 12, 241)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 182, 240)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
            "[M::mem_pestat] mean and std.dev: (188.54, 83.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 581)\n",
            "[M::mem_pestat] mean and std.dev: (302.50, 85.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 693)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (109, 415, 1379)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3919)\n",
            "[M::mem_pestat] mean and std.dev: (760.08, 847.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5189)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (127, 178, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 460)\n",
            "[M::mem_pestat] mean and std.dev: (184.77, 80.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 571)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125916 reads in 10.349 CPU sec, 10.316 real sec\n",
            "[M::process] read 125904 sequences (10000038 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (249, 52298, 9, 271)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (122, 181, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
            "[M::mem_pestat] mean and std.dev: (187.09, 82.50)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (243, 297, 354)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 576)\n",
            "[M::mem_pestat] mean and std.dev: (300.24, 84.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 687)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (121, 183, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 475)\n",
            "[M::mem_pestat] mean and std.dev: (184.13, 82.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125918 reads in 10.230 CPU sec, 10.535 real sec\n",
            "[M::process] read 125910 sequences (10000080 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (261, 52091, 12, 220)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 184, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 461)\n",
            "[M::mem_pestat] mean and std.dev: (189.97, 84.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 572)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.46, 84.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (562, 993, 1045)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2011)\n",
            "[M::mem_pestat] mean and std.dev: (751.09, 477.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2659)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (117, 183, 248)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 510)\n",
            "[M::mem_pestat] mean and std.dev: (185.56, 87.89)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 641)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125904 reads in 17.039 CPU sec, 20.095 real sec\n",
            "[M::process] read 125912 sequences (10000039 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (240, 52292, 2, 251)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (132, 192, 255)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 501)\n",
            "[M::mem_pestat] mean and std.dev: (193.76, 82.88)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 624)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 299, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (302.16, 84.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (133, 178, 237)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
            "[M::mem_pestat] mean and std.dev: (185.22, 76.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125910 reads in 10.943 CPU sec, 11.238 real sec\n",
            "[M::process] read 125908 sequences (10000007 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (258, 52234, 9, 278)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (116, 180, 228)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
            "[M::mem_pestat] mean and std.dev: (177.41, 78.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (244, 298, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 583)\n",
            "[M::mem_pestat] mean and std.dev: (301.94, 85.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 696)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 177, 231)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 447)\n",
            "[M::mem_pestat] mean and std.dev: (177.71, 81.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 555)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125912 reads in 9.868 CPU sec, 9.937 real sec\n",
            "[M::process] read 125906 sequences (10000074 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (240, 52360, 10, 227)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 185, 243)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
            "[M::mem_pestat] mean and std.dev: (187.38, 78.55)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 297, 355)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 575)\n",
            "[M::mem_pestat] mean and std.dev: (300.51, 84.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (85, 730, 955)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2695)\n",
            "[M::mem_pestat] mean and std.dev: (526.00, 509.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3565)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (125, 181, 239)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
            "[M::mem_pestat] mean and std.dev: (182.68, 81.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 581)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125908 reads in 11.467 CPU sec, 12.048 real sec\n",
            "[M::process] read 97784 sequences (7766494 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (239, 52143, 14, 239)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 180, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 486)\n",
            "[M::mem_pestat] mean and std.dev: (187.90, 86.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 607)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (245, 298, 356)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 578)\n",
            "[M::mem_pestat] mean and std.dev: (301.55, 84.71)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 689)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (458, 1366, 2041)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5207)\n",
            "[M::mem_pestat] mean and std.dev: (1120.91, 638.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6790)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (131, 189, 238)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
            "[M::mem_pestat] mean and std.dev: (190.00, 82.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 559)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 125906 reads in 12.370 CPU sec, 13.004 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (195, 40377, 7, 183)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (141, 195, 252)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 474)\n",
            "[M::mem_pestat] mean and std.dev: (195.65, 84.73)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 585)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (247, 299, 357)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 577)\n",
            "[M::mem_pestat] mean and std.dev: (302.68, 84.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 687)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (128, 183, 244)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 476)\n",
            "[M::mem_pestat] mean and std.dev: (185.51, 79.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 592)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 97784 reads in 6.709 CPU sec, 6.802 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem aeruginosa.fna clean_1.fastq clean_2.fastq\n",
            "[main] Real time: 992.761 sec; CPU: 979.486 sec\n",
            "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "#Установка bwa и samtools\n",
        "!apt-get install -y bwa samtools\n",
        "#Индексация референсного генома\n",
        "!bwa index aeruginosa.fna\n",
        "#Выравнивание ридов dв файл aligned.sam\n",
        "!bwa mem aeruginosa.fna clean_1.fastq clean_2.fastq > aligned.sam\n",
        "#Преобразование SAM → BAM → сортировка\n",
        "!samtools view -S -b aligned.sam > aligned.bam\n",
        "!samtools sort aligned.bam -o aligned_sorted.bam\n",
        "!samtools index aligned_sorted.bam\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "S9iTYBB76EQH",
        "outputId": "6f7b027e-3cc8-4626-c49e-8f3f25cf5216"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "12485700 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "12436062 + 0 primary\n",
            "0 + 0 secondary\n",
            "49638 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "0 + 0 primary duplicates\n",
            "11184781 + 0 mapped (89.58% : N/A)\n",
            "11135143 + 0 primary mapped (89.54% : N/A)\n",
            "12436062 + 0 paired in sequencing\n",
            "6218031 + 0 read1\n",
            "6218031 + 0 read2\n",
            "10600640 + 0 properly paired (85.24% : N/A)\n",
            "11031604 + 0 with itself and mate mapped\n",
            "103539 + 0 singletons (0.83% : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ],
      "source": [
        "#Проверка. Покажет количество выровненных ридов, процент покрытия и др\n",
        "!samtools flagstat aligned_sorted.bam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "gXSunYbryZ6q"
      },
      "source": [
        "Очень высокий процент выравнивания (~90%) — отличное качество данных!\n",
        "\n",
        " Более 89% ридов выровнены в парах правильно — тоже отлично.\n",
        "\n",
        " Singletons — один рид из пары выровнен, другой — нет. У нас таких всего ~0.83%.\n",
        "\n",
        " Данные качественные:\n",
        "\n",
        "Почти все риды выровнены.\n",
        "\n",
        "Высокий процент правильно спаренных ридов.\n",
        "\n",
        "Нет дубликатов или химерных пар."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mutYALC4H9FD",
        "outputId": "b965bca2-f5e9-4769-93c6-526e998ea267"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "SRR12905311.108195\t163\tNC_002516.2\t1\t60\t5S75M\t=\t395\t472\tCCCCGTTTAAAGAGACCGGCGATTCTAGTGAAATCGAAAGGGCAGGTCAATTTACAACCAGCGATGACGTAATAGATAGA\tAA/A///E/EE6EEEEE/EE/E/EEAEEE/E//EEEAE/AE/EE6EAEE<AEE/EEAEEE/</EEEEE/EEA//EE</AA\tNM:i:2\tMD:Z:33C14C26\tMC:Z:78M\tAS:i:65\tXS:i:0\n",
            "SRR12905311.120318\t99\tNC_002516.2\t1\t60\t1S79M\t=\t170\t248\tGTTTAAAGAGACCGGCGATTCTAGTGAAATAGAACGGGCAGGTCAATATCCAACCAGCGATGAAGTAATAGATAGATACA\tAAAAAEEEEE6E/E/EEEEEE/EEEEEAEE/EE/EEEAEEE/E6EE//EE/E/EAA/EEEEEE/EA/E///EEAAE/AA/\tNM:i:3\tMD:Z:29C16T15C16\tMC:Z:79M\tAS:i:64\tXS:i:0\n",
            "SRR12905311.459357\t163\tNC_002516.2\t1\t60\t4S76M\t=\t244\t322\tCCCGTTTAAAGAGACCGGCGATTCTAGTGAAATCGAACGGGCAGGTCAATTTCCAACCAGCGATGACGTAATAGATAGAT\tAAAAAEEEEEEEEEEEEAEEEEEEEEEEAEEAEEEEA/EEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEE<E/EEEAEEE\tNM:i:0\tMD:Z:76\tMC:Z:79M\tAS:i:76\tXS:i:0\n",
            "SRR12905311.466260\t2209\tNC_002516.2\t1\t60\t44H36M\t=\t120\t198\tTTTAAAGAGACCGGCGATTCTAGTGAAATCGAACGG\tEEEAE/EEE/E<EE/EEEE/EEEEEE/EAEEA/<AA\tNM:i:0\tMD:Z:36\tMC:Z:79M\tAS:i:36\tXS:i:0\tSA:Z:NC_002516.2,6264361,+,44M36S,60,1;\n",
            "SRR12905311.624367\t163\tNC_002516.2\t1\t60\t5S75M\t=\t200\t278\tCCCCGTTTAAAGAGACCGGCGATTCTAGTGAAATCGAACGGGCAGGTCAATTTCCAACCAGCGATGACGTAATAGATAGA\tAAAAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEAEE\tNM:i:0\tMD:Z:75\tMC:Z:79M\tAS:i:75\tXS:i:0\n",
            "SRR12905311.747808\t163\tNC_002516.2\t1\t60\t8S72M\t=\t302\t381\tTGACCCCGTTTAAAGAGACCGGCGATTCTAGTGAAATCGAACGGGCAGGTCAATTTCCAACCAGCGATGACGTAATAGAT\tAAAAAEEEEEEEEEEEE/EEEEEEEEEEEAAEEEEEEEAEEEEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEE/EEEEEE\tNM:i:0\tMD:Z:72\tMC:Z:80M\tAS:i:72\tXS:i:0\n",
            "SRR12905311.965083\t99\tNC_002516.2\t1\t60\t13S67M\t=\t243\t321\tCGCAGTGACCCCGTTTAAAGAGACCGGCGATTCTAGTGAAATCGAACGGGCAGGTCAATTTCCAACCAGCGATGACGTAA\tAAAAAEEEEEEEEEEEAEEEEEEEEEEEEEEEAE/EEEEEEEEEAAEEEEEEEAEEEEEEEEEEEEEEEEEEEEEEEEA/\tNM:i:0\tMD:Z:67\tMC:Z:79M\tAS:i:67\tXS:i:0\n",
            "SRR12905311.1130311\t99\tNC_002516.2\t1\t60\t4S76M\t=\t348\t427\tCCCGTTTAAAGAGACCGGCGATTCTAGTGAAATCGAACGGGCAGGTCAATTTCCAACCAGCGATGACGTAATAGATAGAT\tAAAAAEE6EEEEEEEEEEEEEEEEE6EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE</EEAEEEEEEE\tNM:i:0\tMD:Z:76\tMC:Z:80M\tAS:i:76\tXS:i:0\n",
            "SRR12905311.1254973\t2209\tNC_002516.2\t1\t60\t34H34M9H\t=\t3067740\t3067819\tTTTAAAGAGACCGGCGATTCTAGTGAAATCGAAC\tEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE\tNM:i:0\tMD:Z:34\tMC:Z:80M\tAS:i:34\tXS:i:0\tSA:Z:NC_002516.2,6264371,+,34M43S,60,0;\n",
            "SRR12905311.1321336\t99\tNC_002516.2\t1\t60\t29S51M\t=\t215\t293\tGTTGACGACTTGAGGTCGCAGTGACCCCGTTTAAAGAGACCGGCGATTCTAGTGAAATCGAACGGGCAGGTCAATTTCCA\tAAAAAEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEE/EEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEAEEEEEE\tNM:i:0\tMD:Z:51\tMC:Z:79M\tAS:i:51\tXS:i:0\n"
          ]
        }
      ],
      "source": [
        "!samtools view aligned_sorted.bam | head"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 27,
      "metadata": {
        "id": "ojpPLsJSKgLY"
      },
      "outputs": [],
      "source": [
        "# Более подробная статистика (глубина, GC-контент, распределение длин и т.д.).\n",
        "!samtools stats aligned_sorted.bam > alignment_stats.txt\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 28,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MfYdmgbrK4kh",
        "outputId": "d9284e63-584a-4578-e497-cf7bb447611c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "NC_002516.2\t6264404\t11184781\t103539\n",
            "*\t0\t0\t1197380\n"
          ]
        }
      ],
      "source": [
        "#Cтатистика по каждому контигу (или хромосоме): число выровненных и невыравненных ридов.\n",
        "!samtools idxstats aligned_sorted.bam\n"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "##Провести поиск SNP.\n",
        "\n",
        "Выбрать 3-4 выявленные мутации. Проверить по NCBI, каким генам они принадлежат"
      ],
      "metadata": {
        "id": "kqRjSJunztER"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get update\n",
        "!apt-get install -y bcftools\n",
        "# Вызов вариантов с bcftools\n",
        "!samtools faidx aeruginosa.fna\n",
        "!bcftools mpileup -f aeruginosa.fna aligned_sorted.bam | bcftools call -mv -Oz -o variants.vcf.gz\n",
        "!bcftools index variants.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "qxKu4JQiysGr",
        "outputId": "62d00f59-eb48-4f52-d092-9bc613c88f4a"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rHit:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "\r0% [Connecting to archive.ubuntu.com] [Waiting for headers] [Waiting for header\r                                                                               \rHit:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
            "Hit:3 http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
            "Hit:4 https://r2u.stat.illinois.edu/ubuntu jammy InRelease\n",
            "Hit:5 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Hit:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
            "Hit:7 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "bcftools is already the newest version (1.13-1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 34 not upgraded.\n",
            "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
            "[mpileup] 1 samples in 1 input files\n",
            "[mpileup] maximum number of reads per input file set to -d 250\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "# Установка Java\n",
        "!apt-get install -y openjdk-21-jdk\n",
        "\n",
        "# Загрузка и распаковка SnpEff\n",
        "!wget -O snpEff.zip https://snpeff.blob.core.windows.net/versions/snpEff_latest_core.zip\n",
        "!unzip -o snpEff.zip\n",
        "%cd snpEff\n",
        "\n",
        "# Создание директории для пользовательского генома PAO1\n",
        "!mkdir -p data/PAO1\n",
        "\n",
        "# Загрузка FASTA и GFF\n",
        "!wget -O data/PAO1/sequences.fa.gz https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Pseudomonas_aeruginosa/reference/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
        "!wget -O data/PAO1/genes.gff.gz     https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Pseudomonas_aeruginosa/reference/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.gff.gz\n",
        "\n",
        "# Распаковка файлов\n",
        "!gunzip -f data/PAO1/*.gz\n",
        "\n",
        "# Редактирование конфигурационного файла\n",
        "!echo \"PAO1.genome : Pseudomonas_aeruginosa_PAO1\" >> snpEff.config\n",
        "\n",
        "# Построение базы данных БЕЗ проверки белков\n",
        "!java -Xmx4g -jar snpEff.jar build -noCheckProtein -noCheckCds -gff3 -v PAO1\n",
        "\n",
        "# Аннотация файла variants.vcf.gz (должен быть в /content)\n",
        "%cd /content/snpEff\n",
        "!java -Xmx4g -jar snpEff.jar -noDownload -v PAO1 /content/snpEff/variants.vcf.gz > variants.ann.vcf\n",
        "\n"
      ],
      "metadata": {
        "id": "j0D6SBYWQdfC",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "outputId": "c5c8df60-71ba-4787-9340-82615ecd86b3"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "openjdk-21-jdk is already the newest version (21.0.7+6~us1-0ubuntu1~22.04).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 34 not upgraded.\n",
            "--2025-05-09 19:16:39--  https://snpeff.blob.core.windows.net/versions/snpEff_latest_core.zip\n",
            "Resolving snpeff.blob.core.windows.net (snpeff.blob.core.windows.net)... 20.150.95.164\n",
            "Connecting to snpeff.blob.core.windows.net (snpeff.blob.core.windows.net)|20.150.95.164|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 66477753 (63M) [application/zip]\n",
            "Saving to: ‘snpEff.zip’\n",
            "\n",
            "snpEff.zip          100%[===================>]  63.40M  10.9MB/s    in 5.6s    \n",
            "\n",
            "2025-05-09 19:16:45 (11.3 MB/s) - ‘snpEff.zip’ saved [66477753/66477753]\n",
            "\n",
            "Archive:  snpEff.zip\n",
            "  inflating: snpEff/LICENSE.md       \n",
            "  inflating: snpEff/snpEff.jar       \n",
            "  inflating: snpEff/SnpSift.jar      \n",
            "  inflating: snpEff/galaxy/snpSift_int.xml  \n",
            "  inflating: snpEff/galaxy/tool-data/snpEff_genomes.loc  \n",
            "  inflating: snpEff/galaxy/tool-data/snpEff_genomes.loc.sample  \n",
            "  inflating: snpEff/galaxy/snpEffWrapper.pl  \n",
            "  inflating: snpEff/galaxy/snpEff.xml  \n",
            "  inflating: snpEff/galaxy/tool_conf.xml  \n",
            "  inflating: snpEff/galaxy/snpSift_caseControl.xml  \n",
            "  inflating: snpEff/galaxy/snpSift_filter.xml  \n",
            "  inflating: snpEff/galaxy/snpSift_annotate.xml  \n",
            "  inflating: snpEff/galaxy/snpSiftWrapper.pl  \n",
            "  inflating: snpEff/galaxy/tool_dependencies.xml  \n",
            "  inflating: snpEff/galaxy/snpEff_download.xml  \n",
            "  inflating: snpEff/snpEff.config    \n",
            "  inflating: snpEff/examples/samples_cancer.txt  \n",
            "  inflating: snpEff/examples/example_motif.vcf  \n",
            "  inflating: snpEff/examples/cancer.eff.vcf  \n",
            "  inflating: snpEff/examples/examples.sh  \n",
            "  inflating: snpEff/examples/test.chr22.ann.txt  \n",
            "  inflating: snpEff/examples/test.1KG.vcf  \n",
            "  inflating: snpEff/examples/cancer.ann.vcf  \n",
            "  inflating: snpEff/examples/1kg.head_chr1.filtered.vcf.gz  \n",
            " extracting: snpEff/examples/test.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense_first.vcf  \n",
            "  inflating: snpEff/examples/test.1KG.ann_reg.vcf  \n",
            "  inflating: snpEff/examples/cancer.vcf  \n",
            "  inflating: snpEff/examples/my_annotations.bed  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense_any_TRMT2A.vcf  \n",
            "  inflating: snpEff/examples/cancer_pedigree.ann.vcf  \n",
            "  inflating: snpEff/examples/variants_2.vcf  \n",
            "  inflating: snpEff/examples/test.ann.vcf  \n",
            "  inflating: snpEff/examples/intervals.bed  \n",
            "  inflating: snpEff/examples/test.1KG.ann_encode.vcf  \n",
            "  inflating: snpEff/examples/variants_1.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.vcf  \n",
            "  inflating: snpEff/examples/file.vcf  \n",
            "  inflating: snpEff/examples/variants_1.ann.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.vcf  \n",
            "  inflating: snpEff/examples/1kg.head_chr1.vcf.gz  \n",
            "  inflating: snpEff/examples/samples_cancer_one.txt  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense_any.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.one_per_line.txt  \n",
            "  inflating: snpEff/examples/variants_2.ann.vcf  \n",
            "  inflating: snpEff/examples/cancer_pedigree.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense.vcf  \n",
            "  inflating: snpEff/scripts/gsa/bayesFactor_correction_scoreCount_max10.sh  \n",
            "  inflating: snpEff/scripts/gsa/geneSetOverlap.sort.txt  \n",
            "  inflating: snpEff/scripts/gsa/create_sets.bds  \n",
            "  inflating: snpEff/scripts/gsa/bayesFactor_correction_scoreCount.r  \n",
            "  inflating: snpEff/scripts/gsa/geneSetsGtex.py  \n",
            "  inflating: snpEff/scripts/gsa/pvalue_correction_scoreCount.sh  \n",
            "  inflating: snpEff/scripts/gsa/pvalue_correction_scoreCount.r  \n",
            "  inflating: snpEff/scripts/gsa/checkGeneNames.py  \n",
            "  inflating: snpEff/scripts/gsa/geneSetOverlap.py  \n",
            "  inflating: snpEff/scripts/gsa/pvalue_correction_scoreCount_min10.sh  \n",
            "  inflating: snpEff/scripts/gsa/bayesFactor_correction_scoreCount.sh  \n",
            "  inflating: snpEff/scripts/make_dbNSFP.sh  \n",
            "  inflating: snpEff/scripts/snpEff   \n",
            "  inflating: snpEff/scripts/snpSift  \n",
            "  inflating: snpEff/scripts/_OLD/vcfBareBones.pl  \n",
            "  inflating: snpEff/scripts/_OLD/cgShore.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfInfoOnePerLine.pl  \n",
            "  inflating: snpEff/scripts/_OLD/sam2fastq.pl  \n",
            "  inflating: snpEff/scripts/_OLD/queue.pl  \n",
            "  inflating: snpEff/scripts/_OLD/extractSequences.pl  \n",
            "  inflating: snpEff/scripts/_OLD/isutf8.py  \n",
            "  inflating: snpEff/scripts/_OLD/db.pl  \n",
            "  inflating: snpEff/scripts/_OLD/annotate_demo.sh  \n",
            "  inflating: snpEff/scripts/_OLD/filterBy.py  \n",
            "  inflating: snpEff/scripts/_OLD/countColumns.py  \n",
            "  inflating: snpEff/scripts/_OLD/plotSmoothScatter.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plot.pl  \n",
            "  inflating: snpEff/scripts/_OLD/ped2vcf.py  \n",
            "  inflating: snpEff/scripts/_OLD/uniqCut.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfEffOnePerLine.pl  \n",
            "  inflating: snpEff/scripts/_OLD/joinSnpEff.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfEffHighest.ORI.py  \n",
            "  inflating: snpEff/scripts/_OLD/fastqSplit.pl  \n",
            "  inflating: snpEff/scripts/_OLD/splitChr.pl  \n",
            "  inflating: snpEff/scripts/_OLD/make_and_commit_docs.sh  \n",
            "  inflating: snpEff/scripts/_OLD/join.pl  \n",
            "  inflating: snpEff/scripts/_OLD/txt2fa.pl  \n",
            "  inflating: snpEff/scripts/_OLD/nextProt_filter.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfOnlyAlts.pl  \n",
            "  inflating: snpEff/scripts/_OLD/annotate_demo_GATK.sh  \n",
            "  inflating: snpEff/scripts/_OLD/fastaSplit.pl  \n",
            "  inflating: snpEff/scripts/_OLD/transpose.pl  \n",
            "  inflating: snpEff/scripts/_OLD/sortLine.py  \n",
            "  inflating: snpEff/scripts/_OLD/vcfAnnFirst.py  \n",
            "  inflating: snpEff/scripts/_OLD/cgShore.sh  \n",
            "  inflating: snpEff/scripts/_OLD/fasta2tab.pl  \n",
            "  inflating: snpEff/scripts/_OLD/txt2vcf.py  \n",
            "  inflating: snpEff/scripts/_OLD/snpSift_filter_sample_to_number.pl  \n",
            "  inflating: snpEff/scripts/_OLD/uniqCount.pl  \n",
            "  inflating: snpEff/scripts/_OLD/swapCols.pl  \n",
            "  inflating: snpEff/scripts/_OLD/fastaSample.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotQQ.pl  \n",
            "  inflating: snpEff/scripts/_OLD/bedEffOnePerLine.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfRefCorrect.py  \n",
            "  inflating: snpEff/scripts/_OLD/plotQQsubsample.pl  \n",
            "  inflating: snpEff/scripts/_OLD/sample.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfFilterSamples.pl  \n",
            "  inflating: snpEff/scripts/_OLD/1kg.sh  \n",
            "  inflating: snpEff/scripts/_OLD/wigSplit.pl  \n",
            "  inflating: snpEff/scripts/_OLD/gffRemovePhase.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotLabel.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfReduceGenotypes.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotMA.pl  \n",
            "  inflating: snpEff/scripts/_OLD/statsNum.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotHistogram.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotXY.pl  \n",
            "  inflating: snpEff/scripts/buildDbNcbi.sh  \n",
            "  inflating: snpEff/exec/snpeff      \n",
            "  inflating: snpEff/exec/snpsift     \n",
            "/content/snpEff/snpEff\n",
            "--2025-05-09 19:16:47--  https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Pseudomonas_aeruginosa/reference/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.fna.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.31, 130.14.250.7, 130.14.250.10, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.31|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1787220 (1.7M) [application/x-gzip]\n",
            "Saving to: ‘data/PAO1/sequences.fa.gz’\n",
            "\n",
            "data/PAO1/sequences 100%[===================>]   1.70M  2.87MB/s    in 0.6s    \n",
            "\n",
            "2025-05-09 19:16:48 (2.87 MB/s) - ‘data/PAO1/sequences.fa.gz’ saved [1787220/1787220]\n",
            "\n",
            "--2025-05-09 19:16:49--  https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/Pseudomonas_aeruginosa/reference/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.gff.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.31, 130.14.250.7, 130.14.250.10, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.31|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 365531 (357K) [application/x-gzip]\n",
            "Saving to: ‘data/PAO1/genes.gff.gz’\n",
            "\n",
            "data/PAO1/genes.gff 100%[===================>] 356.96K  1.17MB/s    in 0.3s    \n",
            "\n",
            "2025-05-09 19:16:49 (1.17 MB/s) - ‘data/PAO1/genes.gff.gz’ saved [365531/365531]\n",
            "\n",
            "00:00:00 SnpEff version SnpEff 5.2f (build 2025-02-07 08:36), by Pablo Cingolani\n",
            "00:00:00 Command: 'build'\n",
            "00:00:00 Building database for 'PAO1'\n",
            "00:00:00 Reading configuration file 'snpEff.config'. Genome: 'PAO1'\n",
            "00:00:00 Looking for config file: '/content/snpEff/snpEff/snpEff.config'\n",
            "00:00:00 Reading config file: /content/snpEff/snpEff/snpEff.config\n",
            "00:00:02 done\n",
            "00:00:02 Reading GFF3 data file  : '/content/snpEff/snpEff/./data/PAO1/genes.gff'\n",
            "00:00:02 Reading file '/content/snpEff/snpEff/./data/PAO1/genes.gff'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0001' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0001' for NC_002516.2\tRefSeq\tCDS\t482\t2026\t+\n",
            "\tdbxref : GenBank:NP_064721.1,GeneID:878417\n",
            "\tgbkey : CDS\n",
            "\tgene : dnaA\n",
            "\tid : cds-NP_064721.1\n",
            "\tlocus_tag : PA0001\n",
            "\tname : NP_064721.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0001\n",
            "\tproduct : chromosome replication initiator DnaA\n",
            "\tprotein_id : NP_064721.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 10\t'NC_002516.2\tRefSeq\tCDS\t483\t2027\t.\t+\t0\tID=cds-NP_064721.1;Parent=gene-PA0001;Dbxref=GenBank:NP_064721.1,GeneID:878417;Name=NP_064721.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=dnaA;locus_tag=PA0001;product=chromosome replication initiator DnaA;protein_id=NP_064721.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0002' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0002' for NC_002516.2\tRefSeq\tCDS\t2055\t3158\t+\n",
            "\tdbxref : GenBank:NP_064722.1,GeneID:879244\n",
            "\tgbkey : CDS\n",
            "\tgene : dnaN\n",
            "\tid : cds-NP_064722.1\n",
            "\tlocus_tag : PA0002\n",
            "\tname : NP_064722.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0002\n",
            "\tproduct : DNA polymerase III subunit beta\n",
            "\tprotein_id : NP_064722.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 12\t'NC_002516.2\tRefSeq\tCDS\t2056\t3159\t.\t+\t0\tID=cds-NP_064722.1;Parent=gene-PA0002;Dbxref=GenBank:NP_064722.1,GeneID:879244;Name=NP_064722.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=dnaN;locus_tag=PA0002;product=DNA polymerase III subunit beta;protein_id=NP_064722.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0003' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0003' for NC_002516.2\tRefSeq\tCDS\t3168\t4277\t+\n",
            "\tdbxref : GenBank:NP_064723.1,GeneID:879229\n",
            "\tgbkey : CDS\n",
            "\tgene : recF\n",
            "\tid : cds-NP_064723.1\n",
            "\tlocus_tag : PA0003\n",
            "\tname : NP_064723.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0003\n",
            "\tproduct : DNA replication and repair protein RecF\n",
            "\tprotein_id : NP_064723.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 14\t'NC_002516.2\tRefSeq\tCDS\t3169\t4278\t.\t+\t0\tID=cds-NP_064723.1;Parent=gene-PA0003;Dbxref=GenBank:NP_064723.1,GeneID:879229;Name=NP_064723.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=recF;locus_tag=PA0003;product=DNA replication and repair protein RecF;protein_id=NP_064723.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0004' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0004' for NC_002516.2\tRefSeq\tCDS\t4274\t6694\t+\n",
            "\tdbxref : GenBank:NP_064724.1,GeneID:879230\n",
            "\tgbkey : CDS\n",
            "\tgene : gyrB\n",
            "\tid : cds-NP_064724.1\n",
            "\tlocus_tag : PA0004\n",
            "\tname : NP_064724.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0004\n",
            "\tproduct : DNA gyrase subunit B\n",
            "\tprotein_id : NP_064724.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 16\t'NC_002516.2\tRefSeq\tCDS\t4275\t6695\t.\t+\t0\tID=cds-NP_064724.1;Parent=gene-PA0004;Dbxref=GenBank:NP_064724.1,GeneID:879230;Name=NP_064724.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=gyrB;locus_tag=PA0004;product=DNA gyrase subunit B;protein_id=NP_064724.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0005' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0005' for NC_002516.2\tRefSeq\tCDS\t7017\t7790\t-\n",
            "\tdbxref : GenBank:NP_064725.1,GeneID:877576\n",
            "\tgbkey : CDS\n",
            "\tgene : lptA\n",
            "\tid : cds-NP_064725.1\n",
            "\tlocus_tag : PA0005\n",
            "\tname : NP_064725.1\n",
            "\tnote : Product name confidence: class 1 (Function experimentally demonstrated in P. aeruginosa)\n",
            "\tparent : gene-PA0005\n",
            "\tproduct : lysophosphatidic acid acyltransferase\n",
            "\tprotein_id : NP_064725.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 18\t'NC_002516.2\tRefSeq\tCDS\t7018\t7791\t.\t-\t0\tID=cds-NP_064725.1;Parent=gene-PA0005;Dbxref=GenBank:NP_064725.1,GeneID:877576;Name=NP_064725.1;Note=Product name confidence: class 1 (Function experimentally demonstrated in P. aeruginosa);gbkey=CDS;gene=lptA;locus_tag=PA0005;product=lysophosphatidic acid acyltransferase;protein_id=NP_064725.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0006' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0006' for NC_002516.2\tRefSeq\tCDS\t7802\t8338\t-\n",
            "\tdbxref : GenBank:NP_064726.1,GeneID:879245\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_064726.1\n",
            "\tlocus_tag : PA0006\n",
            "\tname : NP_064726.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0006\n",
            "\tproduct : D-glycero-beta-D-manno-heptose-1%2C7-bisphosphate 7-phosphatase\n",
            "\tprotein_id : NP_064726.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 20\t'NC_002516.2\tRefSeq\tCDS\t7803\t8339\t.\t-\t0\tID=cds-NP_064726.1;Parent=gene-PA0006;Dbxref=GenBank:NP_064726.1,GeneID:879245;Name=NP_064726.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0006;product=D-glycero-beta-D-manno-heptose-1%2C7-bisphosphate 7-phosphatase;protein_id=NP_064726.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0007' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0007' for NC_002516.2\tRefSeq\tCDS\t8670\t10376\t+\n",
            "\tdbxref : GenBank:NP_064727.1,GeneID:877575\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_064727.1\n",
            "\tlocus_tag : PA0007\n",
            "\tname : NP_064727.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0007\n",
            "\tproduct : hypothetical protein\n",
            "\tprotein_id : NP_064727.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 22\t'NC_002516.2\tRefSeq\tCDS\t8671\t10377\t.\t+\t0\tID=cds-NP_064727.1;Parent=gene-PA0007;Dbxref=GenBank:NP_064727.1,GeneID:877575;Name=NP_064727.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0007;product=hypothetical protein;protein_id=NP_064727.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0008' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0008' for NC_002516.2\tRefSeq\tCDS\t10433\t12487\t-\n",
            "\tdbxref : GenBank:NP_064728.1,GeneID:878214\n",
            "\tgbkey : CDS\n",
            "\tgene : glyS\n",
            "\tid : cds-NP_064728.1\n",
            "\tlocus_tag : PA0008\n",
            "\tname : NP_064728.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0008\n",
            "\tproduct : glycine--tRNA ligase subunit beta\n",
            "\tprotein_id : NP_064728.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 24\t'NC_002516.2\tRefSeq\tCDS\t10434\t12488\t.\t-\t0\tID=cds-NP_064728.1;Parent=gene-PA0008;Dbxref=GenBank:NP_064728.1,GeneID:878214;Name=NP_064728.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=glyS;locus_tag=PA0008;product=glycine--tRNA ligase subunit beta;protein_id=NP_064728.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0009' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0009' for NC_002516.2\tRefSeq\tCDS\t12487\t13434\t-\n",
            "\tdbxref : GenBank:NP_064729.1,GeneID:878095\n",
            "\tgbkey : CDS\n",
            "\tgene : glyQ\n",
            "\tid : cds-NP_064729.1\n",
            "\tlocus_tag : PA0009\n",
            "\tname : NP_064729.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0009\n",
            "\tproduct : glycine--tRNA ligase subunit alpha\n",
            "\tprotein_id : NP_064729.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 26\t'NC_002516.2\tRefSeq\tCDS\t12488\t13435\t.\t-\t0\tID=cds-NP_064729.1;Parent=gene-PA0009;Dbxref=GenBank:NP_064729.1,GeneID:878095;Name=NP_064729.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=glyQ;locus_tag=PA0009;product=glycine--tRNA ligase subunit alpha;protein_id=NP_064729.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0010' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0010' for NC_002516.2\tRefSeq\tCDS\t13539\t14090\t+\n",
            "\tdbxref : GenBank:NP_248700.1,GeneID:878076\n",
            "\tgbkey : CDS\n",
            "\tgene : tag\n",
            "\tid : cds-NP_248700.1\n",
            "\tlocus_tag : PA0010\n",
            "\tname : NP_248700.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0010\n",
            "\tproduct : DNA-3-methyladenine glycosidase I\n",
            "\tprotein_id : NP_248700.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 28\t'NC_002516.2\tRefSeq\tCDS\t13540\t14091\t.\t+\t0\tID=cds-NP_248700.1;Parent=gene-PA0010;Dbxref=GenBank:NP_248700.1,GeneID:878076;Name=NP_248700.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=tag;locus_tag=PA0010;product=DNA-3-methyladenine glycosidase I;protein_id=NP_248700.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0011' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0011' for NC_002516.2\tRefSeq\tCDS\t14234\t15121\t+\n",
            "\tdbxref : GenBank:NP_248701.1,GeneID:877921\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248701.1\n",
            "\tlocus_tag : PA0011\n",
            "\tname : NP_248701.1\n",
            "\tnote : 'Product name confidence: class 3 (Function proposed based on presence of conserved amino acid motif%2C structural feature or limited sequence similarity to an experimentally studied gene)'\n",
            "\tparent : gene-PA0011\n",
            "\tproduct : lipid A biosynthesis lauroyl acyltransferase\n",
            "\tprotein_id : NP_248701.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 30\t'NC_002516.2\tRefSeq\tCDS\t14235\t15122\t.\t+\t0\tID=cds-NP_248701.1;Parent=gene-PA0011;Dbxref=GenBank:NP_248701.1,GeneID:877921;Name=NP_248701.1;Note='Product name confidence: class 3 (Function proposed based on presence of conserved amino acid motif%2C structural feature or limited sequence similarity to an experimentally studied gene)';gbkey=CDS;locus_tag=PA0011;product=lipid A biosynthesis lauroyl acyltransferase;protein_id=NP_248701.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0012' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0012' for NC_002516.2\tRefSeq\tCDS\t15206\t15472\t+\n",
            "\tdbxref : GenBank:NP_248702.1,GeneID:877795\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248702.1\n",
            "\tlocus_tag : PA0012\n",
            "\tname : NP_248702.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0012\n",
            "\tproduct : hypothetical protein\n",
            "\tprotein_id : NP_248702.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 32\t'NC_002516.2\tRefSeq\tCDS\t15207\t15473\t.\t+\t0\tID=cds-NP_248702.1;Parent=gene-PA0012;Dbxref=GenBank:NP_248702.1,GeneID:877795;Name=NP_248702.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0012;product=hypothetical protein;protein_id=NP_248702.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0013' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0013' for NC_002516.2\tRefSeq\tCDS\t15619\t16272\t+\n",
            "\tdbxref : GenBank:NP_248703.1,GeneID:877634\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248703.1\n",
            "\tlocus_tag : PA0013\n",
            "\tname : NP_248703.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0013\n",
            "\tproduct : hypothetical protein\n",
            "\tprotein_id : NP_248703.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 34\t'NC_002516.2\tRefSeq\tCDS\t15620\t16273\t.\t+\t0\tID=cds-NP_248703.1;Parent=gene-PA0013;Dbxref=GenBank:NP_248703.1,GeneID:877634;Name=NP_248703.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0013;product=hypothetical protein;protein_id=NP_248703.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0014' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0014' for NC_002516.2\tRefSeq\tCDS\t16334\t16606\t-\n",
            "\tdbxref : GenBank:NP_248704.1,GeneID:879294\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248704.1\n",
            "\tlocus_tag : PA0014\n",
            "\tname : NP_248704.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0014\n",
            "\tproduct : hypothetical protein\n",
            "\tprotein_id : NP_248704.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 36\t'NC_002516.2\tRefSeq\tCDS\t16335\t16607\t.\t-\t0\tID=cds-NP_248704.1;Parent=gene-PA0014;Dbxref=GenBank:NP_248704.1,GeneID:879294;Name=NP_248704.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0014;product=hypothetical protein;protein_id=NP_248704.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0015' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0015' for NC_002516.2\tRefSeq\tCDS\t16899\t17216\t-\n",
            "\tdbxref : GenBank:NP_248705.1,GeneID:879251\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248705.1\n",
            "\tlocus_tag : PA0015\n",
            "\tname : NP_248705.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0015\n",
            "\tproduct : hypothetical protein\n",
            "\tprotein_id : NP_248705.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 38\t'NC_002516.2\tRefSeq\tCDS\t16900\t17217\t.\t-\t0\tID=cds-NP_248705.1;Parent=gene-PA0015;Dbxref=GenBank:NP_248705.1,GeneID:879251;Name=NP_248705.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0015;product=hypothetical protein;protein_id=NP_248705.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0016' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0016' for NC_002516.2\tRefSeq\tCDS\t17365\t18738\t-\n",
            "\tdbxref : GenBank:NP_248706.1,GeneID:879255\n",
            "\tgbkey : CDS\n",
            "\tgene : trkA\n",
            "\tid : cds-NP_248706.1\n",
            "\tlocus_tag : PA0016\n",
            "\tname : NP_248706.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0016\n",
            "\tproduct : potassium transporter inner membrane associated protein\n",
            "\tprotein_id : NP_248706.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 40\t'NC_002516.2\tRefSeq\tCDS\t17366\t18739\t.\t-\t0\tID=cds-NP_248706.1;Parent=gene-PA0016;Dbxref=GenBank:NP_248706.1,GeneID:879255;Name=NP_248706.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=trkA;locus_tag=PA0016;product=potassium transporter inner membrane associated protein;protein_id=NP_248706.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0017' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0017' for NC_002516.2\tRefSeq\tCDS\t18766\t20070\t-\n",
            "\tdbxref : GenBank:NP_248707.1,GeneID:879297\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248707.1\n",
            "\tlocus_tag : PA0017\n",
            "\tname : NP_248707.1\n",
            "\tparent : gene-PA0017\n",
            "\tproduct : Ribosomal RNA small subunit methyltransferase B\n",
            "\tprotein_id : NP_248707.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 42\t'NC_002516.2\tRefSeq\tCDS\t18767\t20071\t.\t-\t0\tID=cds-NP_248707.1;Parent=gene-PA0017;Dbxref=GenBank:NP_248707.1,GeneID:879297;Name=NP_248707.1;gbkey=CDS;locus_tag=PA0017;product=Ribosomal RNA small subunit methyltransferase B;protein_id=NP_248707.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0018' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0018' for NC_002516.2\tRefSeq\tCDS\t20067\t21011\t-\n",
            "\tdbxref : GenBank:NP_248708.1,GeneID:879258\n",
            "\tgbkey : CDS\n",
            "\tgene : fmt\n",
            "\tid : cds-NP_248708.1\n",
            "\tlocus_tag : PA0018\n",
            "\tname : NP_248708.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0018\n",
            "\tproduct : methionyl-tRNA formyltransferase\n",
            "\tprotein_id : NP_248708.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 44\t'NC_002516.2\tRefSeq\tCDS\t20068\t21012\t.\t-\t0\tID=cds-NP_248708.1;Parent=gene-PA0018;Dbxref=GenBank:NP_248708.1,GeneID:879258;Name=NP_248708.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=fmt;locus_tag=PA0018;product=methionyl-tRNA formyltransferase;protein_id=NP_248708.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0019' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0019' for NC_002516.2\tRefSeq\tCDS\t21066\t21572\t-\n",
            "\tdbxref : GenBank:NP_248709.1,GeneID:879306\n",
            "\tgbkey : CDS\n",
            "\tgene : def\n",
            "\tid : cds-NP_248709.1\n",
            "\tlocus_tag : PA0019\n",
            "\tname : NP_248709.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0019\n",
            "\tproduct : peptide deformylase\n",
            "\tprotein_id : NP_248709.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 46\t'NC_002516.2\tRefSeq\tCDS\t21067\t21573\t.\t-\t0\tID=cds-NP_248709.1;Parent=gene-PA0019;Dbxref=GenBank:NP_248709.1,GeneID:879306;Name=NP_248709.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=def;locus_tag=PA0019;product=peptide deformylase;protein_id=NP_248709.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0020' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0020' for NC_002516.2\tRefSeq\tCDS\t21711\t22736\t+\n",
            "\tdbxref : GenBank:NP_248710.1,GeneID:879309\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248710.1\n",
            "\tlocus_tag : PA0020\n",
            "\tname : NP_248710.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0020\n",
            "\tproduct : hypothetical protein\n",
            "\tprotein_id : NP_248710.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 48\t'NC_002516.2\tRefSeq\tCDS\t21712\t22737\t.\t+\t0\tID=cds-NP_248710.1;Parent=gene-PA0020;Dbxref=GenBank:NP_248710.1,GeneID:879309;Name=NP_248710.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0020;product=hypothetical protein;protein_id=NP_248710.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0021' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0021' for NC_002516.2\tRefSeq\tCDS\t22871\t23959\t+\n",
            "\tdbxref : GenBank:NP_248711.1,GeneID:880730\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248711.1\n",
            "\tlocus_tag : PA0021\n",
            "\tname : NP_248711.1\n",
            "\tnote : 'Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)'\n",
            "\tparent : gene-PA0021\n",
            "\tproduct : hypothetical protein\n",
            "\tprotein_id : NP_248711.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 50\t'NC_002516.2\tRefSeq\tCDS\t22872\t23960\t.\t+\t0\tID=cds-NP_248711.1;Parent=gene-PA0021;Dbxref=GenBank:NP_248711.1,GeneID:880730;Name=NP_248711.1;Note='Product name confidence: class 4 (Homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)';gbkey=CDS;locus_tag=PA0021;product=hypothetical protein;protein_id=NP_248711.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0022' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0022' for NC_002516.2\tRefSeq\tCDS\t24000\t24557\t+\n",
            "\tdbxref : GenBank:NP_248712.1,GeneID:880711\n",
            "\tgbkey : CDS\n",
            "\tid : cds-NP_248712.1\n",
            "\tlocus_tag : PA0022\n",
            "\tname : NP_248712.1\n",
            "\tparent : gene-PA0022\n",
            "\tproduct : threonylcarbamoyl-AMP synthase\n",
            "\tprotein_id : NP_248712.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 52\t'NC_002516.2\tRefSeq\tCDS\t24001\t24558\t.\t+\t0\tID=cds-NP_248712.1;Parent=gene-PA0022;Dbxref=GenBank:NP_248712.1,GeneID:880711;Name=NP_248712.1;gbkey=CDS;locus_tag=PA0022;product=threonylcarbamoyl-AMP synthase;protein_id=NP_248712.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0023' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0023' for NC_002516.2\tRefSeq\tCDS\t24567\t25544\t-\n",
            "\tdbxref : GenBank:NP_248713.1,GeneID:880685\n",
            "\tgbkey : CDS\n",
            "\tgene : qor\n",
            "\tid : cds-NP_248713.1\n",
            "\tlocus_tag : PA0023\n",
            "\tname : NP_248713.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0023\n",
            "\tproduct : oxygen-dependent quinone oxidoreductase\n",
            "\tprotein_id : NP_248713.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 54\t'NC_002516.2\tRefSeq\tCDS\t24568\t25545\t.\t-\t0\tID=cds-NP_248713.1;Parent=gene-PA0023;Dbxref=GenBank:NP_248713.1,GeneID:880685;Name=NP_248713.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=qor;locus_tag=PA0023;product=oxygen-dependent quinone oxidoreductase;protein_id=NP_248713.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0024' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0024' for NC_002516.2\tRefSeq\tCDS\t25735\t26652\t+\n",
            "\tdbxref : GenBank:NP_248714.1,GeneID:880901\n",
            "\tgbkey : CDS\n",
            "\tgene : hemF\n",
            "\tid : cds-NP_248714.1\n",
            "\tlocus_tag : PA0024\n",
            "\tname : NP_248714.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0024\n",
            "\tproduct : coproporphyrinogen III oxidase\n",
            "\tprotein_id : NP_248714.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 56\t'NC_002516.2\tRefSeq\tCDS\t25736\t26653\t.\t+\t0\tID=cds-NP_248714.1;Parent=gene-PA0024;Dbxref=GenBank:NP_248714.1,GeneID:880901;Name=NP_248714.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=hemF;locus_tag=PA0024;product=coproporphyrinogen III oxidase;protein_id=NP_248714.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0025' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0025' for NC_002516.2\tRefSeq\tCDS\t26710\t27534\t+\n",
            "\tdbxref : GenBank:NP_248715.1,GeneID:880894\n",
            "\tgbkey : CDS\n",
            "\tgene : aroE\n",
            "\tid : cds-NP_248715.1\n",
            "\tlocus_tag : PA0025\n",
            "\tname : NP_248715.1\n",
            "\tnote : Product name confidence: class 2 (High similarity to functionally studied protein)\n",
            "\tparent : gene-PA0025\n",
            "\tproduct : shikimate dehydrogenase\n",
            "\tprotein_id : NP_248715.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 58\t'NC_002516.2\tRefSeq\tCDS\t26711\t27535\t.\t+\t0\tID=cds-NP_248715.1;Parent=gene-PA0025;Dbxref=GenBank:NP_248715.1,GeneID:880894;Name=NP_248715.1;Note=Product name confidence: class 2 (High similarity to functionally studied protein);gbkey=CDS;gene=aroE;locus_tag=PA0025;product=shikimate dehydrogenase;protein_id=NP_248715.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Exon's parent 'gene-PA0026' is a Gene instead of a transcript. Created transcript 'TRANSCRIPT_gene-PA0026' for NC_002516.2\tRefSeq\tCDS\t27645\t28631\t+\n",
            "\tdbxref : GenBank:NP_248716.1,GeneID:880904\n",
            "\tgbkey : CDS\n",
            "\tgene : plcB\n",
            "\tid : cds-NP_248716.1\n",
            "\tlocus_tag : PA0026\n",
            "\tname : NP_248716.1\n",
            "\tnote : Product name confidence: class 1 (Function experimentally demonstrated in P. aeruginosa)\n",
            "\tparent : gene-PA0026\n",
            "\tproduct : phospholipase C\n",
            "\tprotein_id : NP_248716.1\n",
            "\tsource : RefSeq\n",
            "\ttransl_table : 11\n",
            "\ttype : CDS\n",
            ". File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 61\t'NC_002516.2\tRefSeq\tCDS\t27646\t28632\t.\t+\t0\tID=cds-NP_248716.1;Parent=gene-PA0026;Dbxref=GenBank:NP_248716.1,GeneID:880904;Name=NP_248716.1;Note=Product name confidence: class 1 (Function experimentally demonstrated in P. aeruginosa);gbkey=CDS;gene=plcB;locus_tag=PA0026;product=phospholipase C;protein_id=NP_248716.1;transl_table=11'\n",
            "WARNING_TRANSCRIPT_NOT_FOUND: Too many 'WARNING_TRANSCRIPT_NOT_FOUND' warnings, no further warnings will be shown.\n",
            "...WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA0369' (NC_002516.2:413364-413930). Created new gene 'null.2' (NC_002516.2:413364-413930). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 781\t'NC_002516.2\tRefSeq\tpseudogene\t413364\t413930\t.\t-\t.\tID=gene-PA0369;Dbxref=GeneID:882043;Name=PA0369;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA0369;pseudo=true;pseudogene=unknown'\n",
            "....WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA0748' (NC_002516.2:816105-817025). Created new gene 'null.3' (NC_002516.2:816105-817025). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 1586\t'NC_002516.2\tRefSeq\tpseudogene\t816105\t817025\t.\t+\t.\tID=gene-PA0748;Dbxref=GeneID:879324,PseudoCap:PA0748;Name=PA0748;Note=transcriptional regulator;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA0748;pseudo=true'\n",
            "...WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA1029' (NC_002516.2:1116133-1116500). Created new gene 'null.4' (NC_002516.2:1116133-1116500). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 2198\t'NC_002516.2\tRefSeq\tpseudogene\t1116133\t1116500\t.\t+\t.\tID=gene-PA1029;Dbxref=GeneID:880268;Name=PA1029;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA1029;pseudo=true;pseudogene=unknown'\n",
            ".WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA1122' (NC_002516.2:1215284-1215822). Created new gene 'null.5' (NC_002516.2:1215284-1215822). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 2406\t'NC_002516.2\tRefSeq\tpseudogene\t1215284\t1215822\t.\t+\t.\tID=gene-PA1122;Dbxref=GeneID:878310;Name=PA1122;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA1122;pseudo=true;pseudogene=unknown'\n",
            ".....WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA1559' (NC_002516.2:1697188-1698344). Created new gene 'null.6' (NC_002516.2:1697188-1698344). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 3320\t'NC_002516.2\tRefSeq\tpseudogene\t1697188\t1698344\t.\t+\t.\tID=gene-PA1559;Dbxref=GeneID:883111;Name=PA1559;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA1559;pseudo=true;pseudogene=unknown'\n",
            ".....WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA2046' (NC_002516.2:2238860-2240214). Created new gene 'null.7' (NC_002516.2:2238860-2240214). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 4378\t'NC_002516.2\tRefSeq\tpseudogene\t2238860\t2240214\t.\t-\t.\tID=gene-PA2046;Dbxref=GeneID:878090;Name=PA2046;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA2046;pseudo=true;pseudogene=unknown'\n",
            ".WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA2139' (NC_002516.2:2355684-2355898). Created new gene 'null.8' (NC_002516.2:2355684-2355898). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 4577\t'NC_002516.2\tRefSeq\tpseudogene\t2355684\t2355898\t.\t+\t.\tID=gene-PA2139;Dbxref=GeneID:880420;Name=PA2139;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA2139;pseudo=true;pseudogene=unknown'\n",
            "...WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA2451' (NC_002516.2:2752866-2754445). Created new gene 'null.9' (NC_002516.2:2752866-2754445). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 5231\t'NC_002516.2\tRefSeq\tpseudogene\t2752866\t2754445\t.\t-\t.\tID=gene-PA2451;Dbxref=GeneID:882922;Name=PA2451;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA2451;pseudo=true;pseudogene=unknown'\n",
            "..WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA2668' (NC_002516.2:3016675-3016970). Created new gene 'null.10' (NC_002516.2:3016675-3016970). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 5687\t'NC_002516.2\tRefSeq\tpseudogene\t3016675\t3016970\t.\t-\t.\tID=gene-PA2668;Dbxref=GeneID:882377;Name=PA2668;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA2668;pseudo=true;pseudogene=unknown'\n",
            ".WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA2727' (NC_002516.2:3079850-3086145). Created new gene 'null.11' (NC_002516.2:3079850-3086145). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 5807\t'NC_002516.2\tRefSeq\tpseudogene\t3079850\t3086145\t.\t+\t.\tID=gene-PA2727;Dbxref=GeneID:882576;Name=PA2727;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA2727;pseudo=true;pseudogene=unknown'\n",
            ".....WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA3164' (NC_002516.2:3550746-3552986). Created new gene 'null.12' (NC_002516.2:3550746-3552986). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 6782\t'NC_002516.2\tRefSeq\tpseudogene\t3550746\t3552986\t.\t-\t.\tID=gene-PA3164;Dbxref=GeneID:878959,PseudoCap:PA3164;Name=PA3164;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA3164;pseudo=true;pseudogene=unknown'\n",
            "...WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA3503' (NC_002516.2:3918796-3919760). Created new gene 'null.13' (NC_002516.2:3918796-3919760). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 7522\t'NC_002516.2\tRefSeq\tpseudogene\t3918796\t3919760\t.\t-\t.\tID=gene-PA3503;Dbxref=GeneID:879168;Name=PA3503;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA3503;pseudo=true;pseudogene=unknown'\n",
            "......WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA4058' (NC_002516.2:4539156-4539817). Created new gene 'null.14' (NC_002516.2:4539156-4539817). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 8705\t'NC_002516.2\tRefSeq\tpseudogene\t4539156\t4539817\t.\t+\t.\tID=gene-PA4058;Dbxref=GeneID:878920;Name=PA4058;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA4058;pseudo=true;pseudogene=unknown'\n",
            "...WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA4360' (NC_002516.2:4887722-4888557). Created new gene 'null.15' (NC_002516.2:4887722-4888557). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 9376\t'NC_002516.2\tRefSeq\tpseudogene\t4887722\t4888557\t.\t-\t.\tID=gene-PA4360;Dbxref=GeneID:881406;Name=PA4360;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA4360;pseudo=true;pseudogene=unknown'\n",
            "...WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA4641' (NC_002516.2:5210037-5210450). Created new gene 'null.16' (NC_002516.2:5210037-5210450). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 10003\t'NC_002516.2\tRefSeq\tpseudogene\t5210037\t5210450\t.\t-\t.\tID=gene-PA4641;Dbxref=GeneID:881250;Name=PA4641;Note='product name confidence: class 4 (homologs of previously reported genes of unknown function%2C or no similarity to any previously reported sequences)%3B this gene contains a frameshift that has been confirmed in the final sequence assembly. Contains high homology to Eschericha coli PhnB';gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA4641;pseudo=true'\n",
            "...WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA4875' (NC_002516.2:5472197-5472735). Created new gene 'null.17' (NC_002516.2:5472197-5472735). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 10547\t'NC_002516.2\tRefSeq\tpseudogene\t5472197\t5472735\t.\t+\t.\tID=gene-PA4875;Dbxref=GeneID:879706;Name=PA4875;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA4875;pseudo=true;pseudogene=unknown'\n",
            ".......WARNING_GENE_NOT_FOUND: Gene 'null' (NC_002516.2:69056-69526) does not include 'gene-PA5532' (NC_002516.2:6226018-6226979). Created new gene 'null.18' (NC_002516.2:6226018-6226979). File '/content/snpEff/snpEff/./data/PAO1/genes.gff' line 11945\t'NC_002516.2\tRefSeq\tpseudogene\t6226018\t6226979\t.\t-\t.\tID=gene-PA5532;Dbxref=GeneID:878139;Name=PA5532;gbkey=Gene;gene_biotype=pseudogene;locus_tag=PA5532;pseudo=true;pseudogene=unknown'\n",
            ".00:00:04 \n",
            "\tTotal: 5907 markers added.\n",
            "00:00:04 Create exons from CDS (if needed): \n",
            ".....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................00:00:04 Exons created for 5572 transcripts.\n",
            "00:00:04 Deleting redundant exons (if needed): \n",
            "00:00:04 \tTotal transcripts with deleted exons: 0\n",
            "00:00:04 Collapsing zero length introns (if needed): \n",
            "00:00:04 \tTotal collapsed transcripts: 0\n",
            "00:00:04 \tReading sequences   :\n",
            "00:00:04 \tFASTA file: '/content/snpEff/snpEff/./data/genomes/PAO1.fa' not found.\n",
            "00:00:04 \tReading FASTA file: '/content/snpEff/snpEff/./data/PAO1/sequences.fa'\n",
            "00:00:04 \t\tReading sequence 'NC_002516.2', length: 6264404\n",
            "00:00:04 \t\tAdding genomic sequences to genes: \n",
            "00:00:05 \tDone (4779 sequences added).\n",
            "00:00:05 \t\tAdding genomic sequences to exons: \n",
            "00:00:05 \tDone (5679 sequences added, 0 ignored).\n",
            "00:00:05 \tTotal: 5679 sequences added, 0 sequences ignored.\n",
            "00:00:05 Finishing up genome\n",
            "00:00:05 Adjusting transcripts: \n",
            "00:00:05 Adjusting genes: \n",
            "00:00:05 Adjusting chromosomes lengths: \n",
            "00:00:05 Ranking exons: \n",
            ".00:00:05 Create UTRs from CDS (if needed): \n",
            "00:00:05 Correcting exons based on frame information.\n",
            "\t\n",
            "00:00:06 \n",
            "00:00:06 Remove empty chromosomes: \n",
            "00:00:06 Marking as 'coding' from CDS information: \n",
            "00:00:06 Done: 0 transcripts marked\n",
            "00:00:06 \n",
            "00:00:06 #-----------------------------------------------\n",
            "# Genome name                : 'Pseudomonas_aeruginosa_PAO1'\n",
            "# Genome version             : 'PAO1'\n",
            "# Genome ID                  : 'PAO1[0]'\n",
            "# Has protein coding info    : true\n",
            "# Has Tr. Support Level info : true\n",
            "# Genes                      : 5699\n",
            "# Protein coding genes       : 5635\n",
            "#-----------------------------------------------\n",
            "# Transcripts                : 5697\n",
            "# Avg. transcripts per gene  : 1.00\n",
            "# TSL transcripts            : 0\n",
            "#-----------------------------------------------\n",
            "# Checked transcripts        : \n",
            "#               AA sequences :      0 ( 0.00% )\n",
            "#              DNA sequences :      0 ( 0.00% )\n",
            "#-----------------------------------------------\n",
            "# Protein coding transcripts : 5635\n",
            "#              Length errors :     55 ( 0.98% )\n",
            "#  STOP codons in CDS errors :     11 ( 0.20% )\n",
            "#         START codon errors :    587 ( 10.42% )\n",
            "#        STOP codon warnings :      8 ( 0.14% )\n",
            "#              UTR sequences :      0 ( 0.00% )\n",
            "#               Total Errors :    587 ( 10.42% )\n",
            "# WARNING                    : No protein coding transcript has UTR\n",
            "#-----------------------------------------------\n",
            "# Cds                        : 5573\n",
            "# Exons                      : 5679\n",
            "# Exons with sequence        : 5679\n",
            "# Exons without sequence     : 0\n",
            "# Avg. exons per transcript  : 1.00\n",
            "#-----------------------------------------------\n",
            "# Number of chromosomes      : 1\n",
            "# Chromosomes                : Format 'chromo_name size codon_table'\n",
            "#\t\t'NC_002516.2'\t6264404\tStandard\n",
            "#-----------------------------------------------\n",
            "\n",
            "00:00:07 Caracterizing exons by splicing (stage 1) : \n",
            "\t.....\n",
            "00:00:07 Caracterizing exons by splicing (stage 2) : \n",
            "\t.....00:00:07 done.\n",
            "00:00:07 [Optional] Rare amino acid annotations\n",
            "WARNING_FILE_NOT_FOUND: Rare Amino Acid analysis: Cannot read protein sequence file '/content/snpEff/snpEff/./data/PAO1/protein.fa', nothing done.\n",
            "00:00:07 Saving database\n",
            "00:00:09 Saving sequences for chromosome 'NC_002516.2' to file '/content/snpEff/snpEff/./data/PAO1/sequence.NC_002516.2.bin'\n",
            "00:00:11 [Optional] Reading regulation elements: GFF\n",
            "WARNING_FILE_NOT_FOUND: Cannot read optional regulation file '/content/snpEff/snpEff/./data/PAO1/regulation.gff', nothing done.\n",
            "00:00:11 [Optional] Reading regulation elements: BED \n",
            "00:00:11 Cannot find optional regulation dir '/content/snpEff/snpEff/./data/PAO1/regulation.bed/', nothing done.\n",
            "00:00:11 [Optional] Reading motifs: GFF\n",
            "WARNING_FILE_NOT_FOUND: Cannot open PWMs file /content/snpEff/snpEff/./data/PAO1/pwms.bin. Nothing done\n",
            "00:00:11 Done\n",
            "00:00:11 Logging\n",
            "00:00:12 Checking for updates...\n",
            "00:00:14 Done.\n",
            "/content/snpEff\n",
            "00:00:00 SnpEff version SnpEff 5.2f (build 2025-02-07 08:36), by Pablo Cingolani\n",
            "00:00:00 Command: 'ann'\n",
            "00:00:00 Reading configuration file 'snpEff.config'. Genome: 'PAO1'\n",
            "00:00:00 Looking for config file: '/content/snpEff/snpEff.config'\n",
            "00:00:00 Reading config file: /content/snpEff/snpEff.config\n",
            "00:00:03 done\n",
            "00:00:03 Reading database for genome version 'PAO1' from file '/content/snpEff/./data/PAO1/snpEffectPredictor.bin' (this might take a while)\n",
            "00:00:05 done\n",
            "00:00:05 Loading Motifs and PWMs\n",
            "00:00:05 Building interval forest\n",
            "00:00:06 done.\n",
            "00:00:06 Genome stats :\n",
            "#-----------------------------------------------\n",
            "# Genome name                : 'Pseudomonas_aeruginosa_PAO1'\n",
            "# Genome version             : 'PAO1'\n",
            "# Genome ID                  : 'PAO1[0]'\n",
            "# Has protein coding info    : true\n",
            "# Has Tr. Support Level info : true\n",
            "# Genes                      : 5699\n",
            "# Protein coding genes       : 5635\n",
            "#-----------------------------------------------\n",
            "# Transcripts                : 5697\n",
            "# Avg. transcripts per gene  : 1.00\n",
            "# TSL transcripts            : 0\n",
            "#-----------------------------------------------\n",
            "# Checked transcripts        : \n",
            "#               AA sequences :      0 ( 0.00% )\n",
            "#              DNA sequences :      0 ( 0.00% )\n",
            "#-----------------------------------------------\n",
            "# Protein coding transcripts : 5635\n",
            "#              Length errors :     55 ( 0.98% )\n",
            "#  STOP codons in CDS errors :     11 ( 0.20% )\n",
            "#         START codon errors :    587 ( 10.42% )\n",
            "#        STOP codon warnings :      8 ( 0.14% )\n",
            "#              UTR sequences :      0 ( 0.00% )\n",
            "#               Total Errors :    587 ( 10.42% )\n",
            "# WARNING                    : No protein coding transcript has UTR\n",
            "#-----------------------------------------------\n",
            "# Cds                        : 5573\n",
            "# Exons                      : 5679\n",
            "# Exons with sequence        : 5679\n",
            "# Exons without sequence     : 0\n",
            "# Avg. exons per transcript  : 1.00\n",
            "#-----------------------------------------------\n",
            "# Number of chromosomes      : 1\n",
            "# Chromosomes                : Format 'chromo_name size codon_table'\n",
            "#\t\t'NC_002516.2'\t6264404\tStandard\n",
            "#-----------------------------------------------\n",
            "\n",
            "00:00:06 Predicting variants\n",
            "00:00:07 Loading sequences for chromosome 'NC_002516.2' from file '/content/snpEff/./data/PAO1/sequence.NC_002516.2.bin'\n",
            "00:00:07 Building sequence tree for chromosome 'NC_002516.2'\n",
            "00:00:07 Done. Loaded 4780 sequences.\n",
            "00:00:13 \t10000 variants (1627 variants per second), 9999 VCF entries\n",
            "00:00:17 \t20000 variants (1897 variants per second), 19998 VCF entries\n",
            "00:00:22 \t30000 variants (1926 variants per second), 29997 VCF entries\n",
            "00:00:26 \t40000 variants (2016 variants per second), 39993 VCF entries\n",
            "00:00:30 \t50000 variants (2154 variants per second), 49993 VCF entries\n",
            "00:00:32 \t60000 variants (2386 variants per second), 59991 VCF entries\n",
            "\n",
            "WARNINGS: Some warning were detected\n",
            "Warning type\tNumber of warnings\n",
            "INFO_REALIGN_3_PRIME\t2\n",
            "WARNING_TRANSCRIPT_INCOMPLETE\t4320\n",
            "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t1503\n",
            "WARNING_TRANSCRIPT_NO_START_CODON\t61336\n",
            "\n",
            "\n",
            "00:00:33 Creating summary file: snpEff_summary.html\n",
            "00:00:34 Creating genes file: snpEff_genes.txt\n",
            "00:00:35 done.\n",
            "00:00:35 Logging\n",
            "00:00:36 Checking for updates...\n",
            "00:00:37 Done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "print('Chromosome      SNP     Gene')\n",
        "print('_____________________________')\n",
        "!awk -F '\\t' '/^#/ { next } { split($8, info, \";\"); for (i in info) if (info[i] ~ /^ANN=/) { split(info[i], ann, \"\\\\|\"); print $1\"\\t\"$2\"\\t\"ann[4]; count++; if (count >= 4) exit } }' variants.ann.vcf\n",
        "\n"
      ],
      "metadata": {
        "id": "I9-_xzujQpQA",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6e3a9180-ee93-4099-839d-e7f7800d7804"
      },
      "execution_count": 69,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Chromosome      SNP     Gene\n",
            "_____________________________\n",
            "NC_002516.2\t857\tdnaA\n",
            "NC_002516.2\t899\tdnaA\n",
            "NC_002516.2\t947\tdnaA\n",
            "NC_002516.2\t1148\tdnaA\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "SNP находятся в гене dnaA, который кодирует инициатор репликации ДНК — белок DnaA, запускающий процесс репликации хромосомы у бактерий, включая Pseudomonas aeruginosa.\n",
        "\n",
        "Чем это опасно для бактерии?\n",
        "Нарушения в DnaA = проблемы с репликацией ДНК\n",
        "\n",
        "Это может:\n",
        "\n",
        "замедлить деление\n",
        "\n",
        "привести к нестабильности генома\n",
        "\n",
        "сделать бактерию более чувствительной к стрессу или антибиотикам"
      ],
      "metadata": {
        "id": "86Y3AtpgoXrj"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "1\tNC_002516.2\tИдентификатор хромосомы/контега — это референс последовательность\n",
        "\n",
        "2\t857\tПозиция SNP на хромосоме (1-based позиция из VCF)\n",
        "\n",
        "3\tdnaA\tГен, в котором находится данный SNP (получено из поля ANN в VCF)"
      ],
      "metadata": {
        "id": "e0YtRQtuiF0G"
      }
    }
  ],
  "metadata": {
    "accelerator": "TPU",
    "colab": {
      "gpuType": "V28",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}